## Make rbcL and matK OTUs

```{r}
#stop on errors
knitr::opts_chunk$set(error=FALSE,tidy=TRUE)
```

### Load libraries
```{r}
# installed from https://github.com/sherrillmix/dnar 
library(dnar)
packageVersion('dnar')
library(parallel)
packageVersion('parallel')
source('functions.R')
```
### Check software versions
```{r}
suppressWarnings(system('mafft --help 2>&1',intern=TRUE))
system('test -x dpkg && dpkg -s fasttree 2>&1',intern=TRUE)
suppressWarnings(system('swarm --version 2>&1',intern=TRUE))
```

### Primers used
```{r primers}
primerSeqs<-list('rbcl'=c('rbcL634F'='ATGCGTTGGAGAGACCGTTTC','rbcLbR'='TCGGTYAGAGCRGGCATRTGCCA'),'matk'=c('matK472F'='CCCRTYCATCTGGAAATCTTGGTTC','matK1248R'='GCTRTRATAATGAGAAAGATTTCTGC'))
```

### Find fastqs
```{r listFastqs}
fastqs<-list.files('data/','_R[12]_.*\\.fastq\\.gz$',recursive=TRUE,full.names=TRUE)
fastqs<-fastqs[!grepl('Undetermined',fastqs)]
primers<-sub('.*(matK|rbcL).*_R([0-9]+)_.*','\\1\\2',basename(fastqs))
primerBases<-sub('[0-9]$','',primers)
```


### Run swarm, mafft and fasttree
```{r swarmAlign, cache=TRUE}
for(primerBase in unique(primerBases)){
  message('Working on ',primerBase)
  if(!dir.exists('work/swarmPair'))dir.create('work/swarmPair',recursive=TRUE)
  outMat<-sprintf('work/swarmPair/%s.Rdat',primerBase)
  outFa<-sprintf('work/swarmPair/%s.fa.gz',primerBase)
  outAlign<-sprintf('work/swarmPair/%s_align.fa.gz',primerBase)
  outTree<-sprintf('work/swarmPair/%s_align.tre',primerBase)
  message('Trimming primers off the start of reads')
  trimReads<-lapply(sprintf('%s%d',primerBase,1:2),function(ii){
    thisPrimer<-primerSeqs[[sub('[12]$','',tolower(ii))]][as.numeric(substring(ii,nchar(ii)))]
    thisFiles<-fastqs[primers==ii]
    reads<-mclapply(thisFiles,function(xx){library(dnar);cat('.');read.fastq(xx)},mc.cores=20,mc.preschedule=FALSE)
    if(mean(unlist(lapply(reads,function(xx)substring(xx$seq,1,nchar(thisPrimer))))%in% expandAmbiguous(thisPrimer)[[1]])<.75)stop(simpleError('Expected primer does not match read start'))
    trimReads<-lapply(reads,function(xx){xx$seq<-substring(xx$seq,nchar(thisPrimer)+1);return(xx)})
    names(trimReads)<-thisFiles
    return(trimReads)
  })
  if(any(names(trimReads[[1]])!=sub('_R2_','_R1_',names(trimReads[[2]]))))stop('Read 1 vs reads 2 file name mismatch')
  readCounts<-as.data.frame(sapply(trimReads,function(xx)sapply(xx,nrow)))
  colnames(readCounts)<-c('raw1','raw2')
  message('Discarding reads with >1 expected error in left or right read and concatenating left-right')
  trimReads<-mcmapply(function(left,right,...){
    cat('.')
    if(any(sub(' .*$','',left$name)!=sub(' .*$','',right$name)))stop('Read 1 vs read 2 name mismatch')
    #last base always low qual so ignore
    q1<-sapply(qualToInts(substring(left$qual,1,nchar(left$qual)-1)),function(xx)sum(10^(-xx/10)))
    q2<-sapply(qualToInts(substring(right$qual,1,nchar(right$qual)-1)),function(xx)sum(10^(-xx/10)))
    #less than 1 expected error in both reads
    selector<-q1<1&q2<1 & !grepl('[^ACTG]',left$seq)&!grepl('[^ACTG]',right$seq)
    seqs<-paste(left[selector,'seq'],revComp(right[selector,'seq']),sep='')
    return(seqs)
  },trimReads[[1]],trimReads[[2]],mc.cores=10,SIMPLIFY=FALSE)
  readCounts$filter<-sapply(trimReads,length)
  message('Read counts')
  print(readCounts)
  samples<-rep(basename(names(trimReads)),sapply(trimReads,length))
  message('Running swarm')
  otus<-runSwarm(unlist(trimReads),swarmArgs='-f -t 40')
  swarmOtus<-as.data.frame.matrix(table(samples,otus[['otus']]))
  write.fa(otus[['seqs']]$name,otus[['seqs']]$seq,outFa)
  save(swarmOtus,file=outMat)
  load(outMat)
  tmpFile<-tempfile()
  message('Discarding singleton OTUs')
  swarmOtus<-swarmOtus[,apply(swarmOtus,2,sum)>1]
  seqs<-otus[['seqs']]
  rownames(seqs)<-seqs$name
  write.fa(colnames(swarmOtus),seqs[colnames(swarmOtus),'seq'],tmpFile)
  message('Aligning with mafft')
  cmd<-sprintf('mafft --thread 50 %s|gzip>%s',tmpFile,outAlign)
  message(cmd)
  system(cmd)
  message('Creating tree with fasttree')
  cmd<-sprintf('zcat %s|fasttree -gtr -nt>%s',outAlign,outTree)
  message(cmd)
  system(cmd)
  message('Output files in:')
  message('    ',outMat)
  message('    ',outFa)
  message('    ',outAlign)
  message('    ',outTree)
}
```

