<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
  </style>
  <style type="text/css">
  /*! normalize.css v2.1.3 | MIT License | git.io/normalize */
  
  /* ==========================================================================
     HTML5 display definitions
     ========================================================================== */
  
  /**
   * Correct `block` display not defined in IE 8/9.
   */
  
  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  main,
  nav,
  section,
  summary {
      display: block;
  }
  
  /**
   * Correct `inline-block` display not defined in IE 8/9.
   */
  
  audio,
  canvas,
  video {
      display: inline-block;
  }
  
  /**
   * Prevent modern browsers from displaying `audio` without controls.
   * Remove excess height in iOS 5 devices.
   */
  
  audio:not([controls]) {
      display: none;
      height: 0;
  }
  
  /**
   * Address `[hidden]` styling not present in IE 8/9.
   * Hide the `template` element in IE, Safari, and Firefox < 22.
   */
  
  [hidden],
  template {
      display: none;
  }
  
  /* ==========================================================================
     Base
     ========================================================================== */
  
  /**
   * 1. Set default font family to sans-serif.
   * 2. Prevent iOS text size adjust after orientation change, without disabling
   *    user zoom.
   */
  
  html {
      font-family: sans-serif; /* 1 */
      -ms-text-size-adjust: 100%; /* 2 */
      -webkit-text-size-adjust: 100%; /* 2 */
  }
  
  /**
   * Remove default margin.
   */
  
  body {
      margin: 0;
  }
  
  /* ==========================================================================
     Links
     ========================================================================== */
  
  /**
   * Remove the gray background color from active links in IE 10.
   */
  
  a {
      background: transparent;
  }
  
  /**
   * Address `outline` inconsistency between Chrome and other browsers.
   */
  
  a:focus {
      outline: thin dotted;
  }
  
  /**
   * Improve readability when focused and also mouse hovered in all browsers.
   */
  
  a:active,
  a:hover {
      outline: 0;
  }
  
  /* ==========================================================================
     Typography
     ========================================================================== */
  
  /**
   * Address variable `h1` font-size and margin within `section` and `article`
   * contexts in Firefox 4+, Safari 5, and Chrome.
   */
  
  h1 {
      font-size: 2em;
      margin: 0.67em 0;
  }
  
  /**
   * Address styling not present in IE 8/9, Safari 5, and Chrome.
   */
  
  abbr[title] {
      border-bottom: 1px dotted;
  }
  
  /**
   * Address style set to `bolder` in Firefox 4+, Safari 5, and Chrome.
   */
  
  b,
  strong {
      font-weight: bold;
  }
  
  /**
   * Address styling not present in Safari 5 and Chrome.
   */
  
  dfn {
      font-style: italic;
  }
  
  /**
   * Address differences between Firefox and other browsers.
   */
  
  hr {
      -moz-box-sizing: content-box;
      box-sizing: content-box;
      height: 0;
  }
  
  /**
   * Address styling not present in IE 8/9.
   */
  
  mark {
      background: #ff0;
      color: #000;
  }
  
  /**
   * Correct font family set oddly in Safari 5 and Chrome.
   */
  
  code,
  kbd,
  pre,
  samp {
      font-family: monospace, serif;
      font-size: 1em;
  }
  
  /**
   * Improve readability of pre-formatted text in all browsers.
   */
  
  pre {
      white-space: pre-wrap;
  }
  
  /**
   * Set consistent quote types.
   */
  
  q {
      quotes: "\201C" "\201D" "\2018" "\2019";
  }
  
  /**
   * Address inconsistent and variable font size in all browsers.
   */
  
  small {
      font-size: 80%;
  }
  
  /**
   * Prevent `sub` and `sup` affecting `line-height` in all browsers.
   */
  
  sub,
  sup {
      font-size: 75%;
      line-height: 0;
      position: relative;
      vertical-align: baseline;
  }
  
  sup {
      top: -0.5em;
  }
  
  sub {
      bottom: -0.25em;
  }
  
  /* ==========================================================================
     Embedded content
     ========================================================================== */
  
  /**
   * Remove border when inside `a` element in IE 8/9.
   */
  
  img {
      border: 0;
  }
  
  /**
   * Correct overflow displayed oddly in IE 9.
   */
  
  svg:not(:root) {
      overflow: hidden;
  }
  
  /* ==========================================================================
     Figures
     ========================================================================== */
  
  /**
   * Address margin not present in IE 8/9 and Safari 5.
   */
  
  figure {
      margin: 0;
  }
  
  /* ==========================================================================
     Forms
     ========================================================================== */
  
  /**
   * Define consistent border, margin, and padding.
   */
  
  fieldset {
      border: 1px solid #c0c0c0;
      margin: 0 2px;
      padding: 0.35em 0.625em 0.75em;
  }
  
  /**
   * 1. Correct `color` not being inherited in IE 8/9.
   * 2. Remove padding so people aren't caught out if they zero out fieldsets.
   */
  
  legend {
      border: 0; /* 1 */
      padding: 0; /* 2 */
  }
  
  /**
   * 1. Correct font family not being inherited in all browsers.
   * 2. Correct font size not being inherited in all browsers.
   * 3. Address margins set differently in Firefox 4+, Safari 5, and Chrome.
   */
  
  button,
  input,
  select,
  textarea {
      font-family: inherit; /* 1 */
      font-size: 100%; /* 2 */
      margin: 0; /* 3 */
  }
  
  /**
   * Address Firefox 4+ setting `line-height` on `input` using `!important` in
   * the UA stylesheet.
   */
  
  button,
  input {
      line-height: normal;
  }
  
  /**
   * Address inconsistent `text-transform` inheritance for `button` and `select`.
   * All other form control elements do not inherit `text-transform` values.
   * Correct `button` style inheritance in Chrome, Safari 5+, and IE 8+.
   * Correct `select` style inheritance in Firefox 4+ and Opera.
   */
  
  button,
  select {
      text-transform: none;
  }
  
  /**
   * 1. Avoid the WebKit bug in Android 4.0.* where (2) destroys native `audio`
   *    and `video` controls.
   * 2. Correct inability to style clickable `input` types in iOS.
   * 3. Improve usability and consistency of cursor style between image-type
   *    `input` and others.
   */
  
  button,
  html input[type="button"], /* 1 */
  input[type="reset"],
  input[type="submit"] {
      -webkit-appearance: button; /* 2 */
      cursor: pointer; /* 3 */
  }
  
  /**
   * Re-set default cursor for disabled elements.
   */
  
  button[disabled],
  html input[disabled] {
      cursor: default;
  }
  
  /**
   * 1. Address box sizing set to `content-box` in IE 8/9/10.
   * 2. Remove excess padding in IE 8/9/10.
   */
  
  input[type="checkbox"],
  input[type="radio"] {
      box-sizing: border-box; /* 1 */
      padding: 0; /* 2 */
  }
  
  /**
   * 1. Address `appearance` set to `searchfield` in Safari 5 and Chrome.
   * 2. Address `box-sizing` set to `border-box` in Safari 5 and Chrome
   *    (include `-moz` to future-proof).
   */
  
  input[type="search"] {
      -webkit-appearance: textfield; /* 1 */
      -moz-box-sizing: content-box;
      -webkit-box-sizing: content-box; /* 2 */
      box-sizing: content-box;
  }
  
  /**
   * Remove inner padding and search cancel button in Safari 5 and Chrome
   * on OS X.
   */
  
  input[type="search"]::-webkit-search-cancel-button,
  input[type="search"]::-webkit-search-decoration {
      -webkit-appearance: none;
  }
  
  /**
   * Remove inner padding and border in Firefox 4+.
   */
  
  button::-moz-focus-inner,
  input::-moz-focus-inner {
      border: 0;
      padding: 0;
  }
  
  /**
   * 1. Remove default vertical scrollbar in IE 8/9.
   * 2. Improve readability and alignment in all browsers.
   */
  
  textarea {
      overflow: auto; /* 1 */
      vertical-align: top; /* 2 */
  }
  
  /* ==========================================================================
     Tables
     ========================================================================== */
  
  /**
   * Remove most spacing between table cells.
   */
  
  table {
      border-collapse: collapse;
      border-spacing: 0;
  }
  
  .go-top {
  position: fixed;
  bottom: 2em;
  right: 2em;
  text-decoration: none;
  background-color: #E0E0E0;
  font-size: 12px;
  padding: 1em;
  display: inline;
  }
  
  /* Github css */
  
  html,body{        margin: auto;
      padding-right: 1em;
      padding-left: 1em;
      max-width: 66em; color:black;}*:not('#mkdbuttons'){margin:0;padding:0}body{font:13.34px helvetica,arial,freesans,clean,sans-serif;-webkit-font-smoothing:subpixel-antialiased;line-height:1.4;padding:3px;background:#fff;border-radius:3px;-moz-border-radius:3px;-webkit-border-radius:3px}p{margin:1em 0}a{color:#4183c4;text-decoration:none}body{background-color:#fff;padding:30px;margin:15px;font-size:14px;line-height:1.6}body>*:first-child{margin-top:0!important}body>*:last-child{margin-bottom:0!important}@media screen{body{box-shadow:0 0 0 1px #cacaca,0 0 0 4px #eee}}h1,h2,h3,h4,h5,h6{margin:20px 0 10px;padding:0;font-weight:bold;-webkit-font-smoothing:subpixel-antialiased;cursor:text}h1{font-size:28px;color:#000}h2{font-size:24px;border-bottom:1px solid #ccc;color:#000}h3{font-size:18px;color:#333}h4{font-size:16px;color:#333}h5{font-size:14px;color:#333}h6{color:#777;font-size:14px}p,blockquote,table,pre{margin:15px 0}ul{padding-left:30px}ol{padding-left:30px}ol li ul:first-of-type{margin-top:0}hr{background:transparent url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAECAYAAACtBE5DAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyJpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNSBNYWNpbnRvc2giIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OENDRjNBN0E2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OENDRjNBN0I2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo4Q0NGM0E3ODY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo4Q0NGM0E3OTY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PqqezsUAAAAfSURBVHjaYmRABcYwBiM2QSA4y4hNEKYDQxAEAAIMAHNGAzhkPOlYAAAAAElFTkSuQmCC) repeat-x 0 0;border:0 none;color:#ccc;height:4px;padding:0}body>h2:first-child{margin-top:0;padding-top:0}body>h1:first-child{margin-top:0;padding-top:0}body>h1:first-child+h2{margin-top:0;padding-top:0}body>h3:first-child,body>h4:first-child,body>h5:first-child,body>h6:first-child{margin-top:0;padding-top:0}a:first-child h1,a:first-child h2,a:first-child h3,a:first-child h4,a:first-child h5,a:first-child h6{margin-top:0;padding-top:0}h1+p,h2+p,h3+p,h4+p,h5+p,h6+p,ul li>:first-child,ol li>:first-child{margin-top:0}dl{padding:0}dl dt{font-size:14px;font-weight:bold;font-style:italic;padding:0;margin:15px 0 5px}dl dt:first-child{padding:0}dl dt>:first-child{margin-top:0}dl dt>:last-child{margin-bottom:0}dl dd{margin:0 0 15px;padding:0 15px}dl dd>:first-child{margin-top:0}dl dd>:last-child{margin-bottom:0}blockquote{border-left:4px solid #DDD;padding:0 15px;color:#777}blockquote>:first-child{margin-top:0}blockquote>:last-child{margin-bottom:0}table{border-collapse:collapse;border-spacing:0;font-size:100%;font:inherit}table th{font-weight:bold;border:1px solid #ccc;padding:6px 13px}table td{border:1px solid #ccc;padding:6px 13px}table tr{border-top:1px solid #ccc;background-color:#fff}table tr:nth-child(2n){background-color:#f8f8f8}img{max-width:100%}code,tt{margin:0 2px;padding:0 5px;white-space:nowrap;border:1px solid #eaeaea;background-color:#f8f8f8;border-radius:3px;font-family:Consolas,'Liberation Mono',Courier,monospace;font-size:12px;color:#333}pre>code{margin:0;padding:0;white-space:pre;border:0;background:transparent}.highlight pre{background-color:#f8f8f8;border:1px solid #ccc;font-size:13px;line-height:19px;overflow:auto;padding:6px 10px;border-radius:3px}pre{background-color:#f8f8f8;border:1px solid #ccc;font-size:13px;line-height:19px;overflow:auto;padding:6px 10px;border-radius:3px}pre code,pre tt{background-color:transparent;border:0}.poetry pre{font-family:Georgia,Garamond,serif!important;font-style:italic;font-size:110%!important;line-height:1.6em;display:block;margin-left:1em}.poetry pre code{font-family:Georgia,Garamond,serif!important;word-break:break-all;word-break:break-word;-webkit-hyphens:auto;-moz-hyphens:auto;hyphens:auto;white-space:pre-wrap}sup,sub,a.footnote{font-size:1.4ex;height:0;line-height:1;vertical-align:super;position:relative}sub{vertical-align:sub;top:-1px}@media print{body{background:#fff}img,pre,blockquote,table,figure{page-break-inside:avoid}body{background:#fff;border:0}code{background-color:#fff;color:#333!important;padding:0 .2em;border:1px solid #dedede}pre{background:#fff}pre code{background-color:white!important;overflow:visible}}@media screen{body.inverted{color:#eee!important;border-color:#555;box-shadow:none}.inverted body,.inverted hr .inverted p,.inverted td,.inverted li,.inverted h1,.inverted h2,.inverted h3,.inverted h4,.inverted h5,.inverted h6,.inverted th,.inverted .math,.inverted caption,.inverted dd,.inverted dt,.inverted blockquote{color:#eee!important;border-color:#555;box-shadow:none}.inverted td,.inverted th{background:#333}.inverted h2{border-color:#555}.inverted hr{border-color:#777;border-width:1px!important}::selection{background:rgba(157,193,200,0.5)}h1::selection{background-color:rgba(45,156,208,0.3)}h2::selection{background-color:rgba(90,182,224,0.3)}h3::selection,h4::selection,h5::selection,h6::selection,li::selection,ol::selection{background-color:rgba(133,201,232,0.3)}code::selection{background-color:rgba(0,0,0,0.7);color:#eee}code span::selection{background-color:rgba(0,0,0,0.7)!important;color:#eee!important}a::selection{background-color:rgba(255,230,102,0.2)}.inverted a::selection{background-color:rgba(255,230,102,0.6)}td::selection,th::selection,caption::selection{background-color:rgba(180,237,95,0.5)}.inverted{background:#0b2531;background:#252a2a}.inverted body{background:#252a2a}.inverted a{color:#acd1d5}}.highlight .c{color:#998;font-style:italic}.highlight .err{color:#a61717;background-color:#e3d2d2}.highlight .k,.highlight .o{font-weight:bold}.highlight .cm{color:#998;font-style:italic}.highlight .cp{color:#999;font-weight:bold}.highlight .c1{color:#998;font-style:italic}.highlight .cs{color:#999;font-weight:bold;font-style:italic}.highlight .gd{color:#000;background-color:#fdd}.highlight .gd .x{color:#000;background-color:#faa}.highlight .ge{font-style:italic}.highlight .gr{color:#a00}.highlight .gh{color:#999}.highlight .gi{color:#000;background-color:#dfd}.highlight .gi .x{color:#000;background-color:#afa}.highlight .go{color:#888}.highlight .gp{color:#555}.highlight .gs{font-weight:bold}.highlight .gu{color:#800080;font-weight:bold}.highlight .gt{color:#a00}.highlight .kc,.highlight .kd,.highlight .kn,.highlight .kp,.highlight .kr{font-weight:bold}.highlight .kt{color:#458;font-weight:bold}.highlight .m{color:#099}.highlight .s{color:#d14}.highlight .na{color:#008080}.highlight .nb{color:#0086b3}.highlight .nc{color:#458;font-weight:bold}.highlight .no{color:#008080}.highlight .ni{color:#800080}.highlight .ne,.highlight .nf{color:#900;font-weight:bold}.highlight .nn{color:#555}.highlight .nt{color:#000080}.highlight .nv{color:#008080}.highlight .ow{font-weight:bold}.highlight .w{color:#bbb}.highlight .mf,.highlight .mh,.highlight .mi,.highlight .mo{color:#099}.highlight .sb,.highlight .sc,.highlight .sd,.highlight .s2,.highlight .se,.highlight .sh,.highlight .si,.highlight .sx{color:#d14}.highlight .sr{color:#009926}.highlight .s1{color:#d14}.highlight .ss{color:#990073}.highlight .bp{color:#999}.highlight .vc,.highlight .vg,.highlight .vi{color:#008080}.highlight .il{color:#099}.highlight .gc{color:#999;background-color:#eaf2f5}.type-csharp .highlight .k,.type-csharp .highlight .kt{color:#00F}.type-csharp .highlight .nf{color:#000;font-weight:normal}.type-csharp .highlight .nc{color:#2b91af}.type-csharp .highlight .nn{color:#000}.type-csharp .highlight .s,.type-csharp .highlight .sc{color:#a31515}
  </style>
</head>
<body>
<div id="TOC">
<ul>
<li><a href="#analysis-code-for-geographically-restricted-malaria-infections-of-wild-bonobos-include-a-new-laverania-species">Analysis code for “Geographically restricted malaria infections of wild bonobos include a new Laverania species”</a><ul>
<li><a href="#download-data-from-sra">Download data from SRA</a><ul>
<li><a href="#software-versions">Software versions</a></li>
<li><a href="#read-in-sra-run-table">Read in SRA run table</a></li>
<li><a href="#download-data-from-sra-1">Download data from SRA</a></li>
<li><a href="#rename-into-convenient-form">Rename into convenient form</a></li>
</ul></li>
<li><a href="#make-rbcl-and-matk-otus">Make rbcL and matK OTUs</a><ul>
<li><a href="#load-libraries">Load libraries</a></li>
<li><a href="#check-software-versions">Check software versions</a></li>
<li><a href="#primers-used">Primers used</a></li>
<li><a href="#find-fastqs">Find fastqs</a></li>
<li><a href="#run-swarm-mafft-and-fasttree">Run swarm, mafft and fasttree</a></li>
</ul></li>
<li><a href="#software-versions-1">Software versions</a></li>
<li><a href="#prepare-blast-databases">Prepare blast databases</a></li>
<li><a href="#blast-swarm-otus-against-rbclmatk-databases">Blast swarm OTUs against rbcl/matk databases</a></li>
<li><a href="#load-libraries-1">Load libraries</a></li>
<li><a href="#download-ncbi-accession-numbers-and-taxonomy">Download NCBI accession numbers and taxonomy</a></li>
<li><a href="#prepare-database-of-accession-numbers-and-taxonomy">Prepare database of accession numbers and taxonomy</a></li>
<li><a href="#assign-taxonomy-to-blast-hits">Assign taxonomy to blast hits</a></li>
<li><a href="#load-libraries-2">Load libraries</a></li>
<li><a href="#load-data">Load data</a></li>
<li><a href="#load-libraries-3">Load libraries</a></li>
<li><a href="#load-data-1">Load data</a><ul>
<li><a href="#make-heatmap-of-otu-abundances">Make heatmap of OTU abundances</a></li>
</ul></li>
<li><a href="#software-versions-2">Software versions</a></li>
<li><a href="#merge-reads">Merge reads</a></li>
<li><a href="#read-in-all-sequences">Read in all sequences</a></li>
<li><a href="#run-qiime">Run qiime</a></li>
<li><a href="#load-libraries-4">Load libraries</a></li>
<li><a href="#load-the-data">Load the data</a></li>
<li><a href="#set-up-grouping-variables">Set up grouping variables</a></li>
<li><a href="#set-up-plotting-parameters">Set up plotting parameters</a></li>
<li><a href="#calculate-shannon-diversities">Calculate Shannon diversities</a></li>
<li><a href="#calculate-wilcoxon-test-p-values">Calculate Wilcoxon test p-values</a></li>
<li><a href="#generate-distance-comparison-figure">Generate distance comparison figure</a></li>
<li><a href="#load-the-data-1">Load the data</a></li>
<li><a href="#setup-plotting-parameters">Setup plotting parameters</a></li>
<li><a href="#plot-heatmap-of-bacterial-proportions">Plot heatmap of bacterial proportions</a></li>
<li><a href="#calculate-p-values">Calculate p-values</a></li>
<li><a href="#plot-otus-that-different-between-tl-2-plasmodium-positive-and-negative-bonobos">Plot OTUs that different between TL-2 plasmodium positive and negative bonobos</a></li>
<li><a href="#load-libraries-5">Load libraries</a></li>
<li><a href="#load-the-data-2">Load the data</a></li>
<li><a href="#calculate-distance-matrices-from-otu-table">Calculate distance matrices from OTU table</a></li>
<li><a href="#set-up-plotting-parameters-1">Set up plotting parameters</a></li>
<li><a href="#plot-16s-pcoa">Plot 16s PCoA</a></li>
<li><a href="#plot-16s-t-sne">Plot 16s t-SNE</a></li>
<li><a href="#load-data-2">Load data</a></li>
<li><a href="#convenience-function-for-getting-distances-from-matrix">Convenience function for getting distances from matrix</a></li>
<li><a href="#pull-out-distances-for-comparisons">Pull out distances for comparisons</a></li>
<li><a href="#calculate-wilcoxon-test-p-values-between-groups-of-distances">Calculate Wilcoxon test p-values between groups of distances</a></li>
<li><a href="#setup-plotting-parameters-1">Setup plotting parameters</a></li>
<li><a href="#plot-comparison-of-distances-between-groups">Plot comparison of distances between groups</a></li>
<li><a href="#calculate-permanova-statistics-of-distances-and-covariates">Calculate PERMANOVA statistics of distances and covariates</a></li>
</ul></li>
</ul>
</div>
<h1 id="analysis-code-for-geographically-restricted-malaria-infections-of-wild-bonobos-include-a-new-laverania-species">Analysis code for “Geographically restricted malaria infections of wild bonobos include a new Laverania species”</h1>
<h2 id="download-data-from-sra">Download data from SRA</h2>
<h3 id="software-versions">Software versions</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;fastq-dump --version&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] &quot;&quot;                   &quot;fastq-dump : 2.8.0&quot; &quot;&quot;</code></pre>
<h3 id="read-in-sra-run-table">Read in SRA run table</h3>
<p>The SRA run table was downloaded from <a href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP108776" class="uri">https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP108776</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samples&lt;-<span class="kw">read.table</span>(<span class="st">&#39;SraRunTable.txt&#39;</span>,<span class="dt">sep=</span><span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>,<span class="dt">header=</span><span class="ot">TRUE</span>,<span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
<span class="kw">rownames</span>(samples)&lt;-samples$Run_s
samples[,<span class="kw">c</span>(<span class="st">&#39;sample&#39;</span>,<span class="st">&#39;primer&#39;</span>)]&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">strsplit</span>(samples$Library_Name_s,<span class="st">&#39;_&#39;</span>))</code></pre></div>
<h3 id="download-data-from-sra-1">Download data from SRA</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;fastq-dump --version 2&gt;&amp;1&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] &quot;&quot;                   &quot;fastq-dump : 2.8.0&quot; &quot;&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">if(!<span class="kw">dir.exists</span>(<span class="st">&#39;sra&#39;</span>))<span class="kw">dir.create</span>(<span class="st">&#39;sra&#39;</span>)
for(ii in samples$Run_s){
  outFiles&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;sra/%s_%d.fastq.gz&#39;</span>,ii,<span class="dv">1</span>:<span class="dv">2</span>)
  cmd&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;fastq-dump --gzip --split-files --outdir data %s&#39;</span>,ii)
  <span class="kw">message</span>(cmd)
  if(<span class="kw">any</span>(!<span class="kw">file.exists</span>(outFiles)))<span class="kw">system</span>(cmd)
  if(<span class="kw">any</span>(!<span class="kw">file.exists</span>(outFiles)))<span class="kw">stop</span>(<span class="st">&#39;fastq-dump unsucessful&#39;</span>)
}</code></pre></div>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656515</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656723</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656724</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656721</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656722</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656520</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656517</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656719</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656720</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656717</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656514</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656521</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656581</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656582</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656620</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656621</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656646</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656647</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656699</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656700</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656701</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656702</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656710</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656711</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656712</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656713</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656714</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656715</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656716</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656718</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656432</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656433</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656479</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656604</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656642</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656643</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656644</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656645</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656648</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656649</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656652</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656654</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656655</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656662</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656664</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656665</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656666</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656667</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656674</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656675</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656686</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656687</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656688</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656689</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656690</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656691</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656692</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656693</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656694</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656695</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656696</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656697</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656698</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656703</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656704</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656705</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656706</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656707</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656708</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656709</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656431</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656434</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656440</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656441</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656442</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656451</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656462</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656463</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656464</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656465</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656483</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656488</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656489</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656490</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656491</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656492</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656493</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656497</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656498</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656499</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656500</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656501</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656512</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656513</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656524</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656525</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656528</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656529</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656535</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656544</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656545</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656546</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656548</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656566</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656568</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656579</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656601</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656602</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656603</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656612</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656619</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656629</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656630</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656631</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656632</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656633</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656634</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656635</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656636</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656637</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656638</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656639</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656640</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656641</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656650</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656651</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656653</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656656</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656657</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656658</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656659</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656660</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656661</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656663</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656668</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656669</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656670</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656671</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656672</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656673</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656676</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656677</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656678</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656679</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656680</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656681</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656682</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656683</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656684</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656685</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656435</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656436</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656437</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656438</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656439</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656443</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656444</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656445</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656446</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656447</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656448</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656449</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656450</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656452</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656453</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656454</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656455</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656456</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656457</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656458</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656459</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656460</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656461</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656466</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656467</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656468</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656469</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656470</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656471</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656472</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656473</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656474</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656475</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656476</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656477</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656478</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656480</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656481</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656482</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656484</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656485</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656486</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656487</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656494</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656495</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656496</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656502</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656503</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656504</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656505</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656506</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656507</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656508</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656509</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656510</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656511</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656516</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656518</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656519</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656522</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656523</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656526</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656527</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656530</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656531</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656532</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656533</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656534</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656536</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656537</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656538</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656539</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656540</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656541</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656542</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656543</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656547</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656549</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656550</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656551</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656552</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656553</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656554</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656555</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656556</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656557</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656558</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656559</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656560</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656561</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656562</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656563</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656564</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656565</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656567</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656569</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656570</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656571</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656572</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656573</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656574</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656575</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656576</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656577</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656578</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656580</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656583</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656584</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656585</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656586</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656587</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656588</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656589</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656590</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656591</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656592</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656593</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656594</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656595</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656596</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656597</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656598</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656599</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656600</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656605</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656606</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656607</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656608</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656609</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656610</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656611</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656613</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656614</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656615</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656616</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656617</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656618</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656622</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656623</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656624</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656625</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656626</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656627</code></pre>
<pre><code>## fastq-dump --gzip --split-files --outdir data SRR5656628</code></pre>
<h3 id="rename-into-convenient-form">Rename into convenient form</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">if(!<span class="kw">dir.exists</span>(<span class="st">&#39;16s/data&#39;</span>))<span class="kw">dir.create</span>(<span class="st">&#39;16s/data&#39;</span>)
if(!<span class="kw">dir.exists</span>(<span class="st">&#39;data&#39;</span>))<span class="kw">dir.create</span>(<span class="st">&#39;data&#39;</span>)
for(ii in samples$Run_s){
  sraNames&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;sra/%s_%d.fastq.gz&#39;</span>,ii,<span class="dv">1</span>:<span class="dv">2</span>)
  newNames&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;%s/%s_%s_R%d_.fastq.gz&#39;</span>, <span class="kw">ifelse</span>(samples[ii,<span class="st">&#39;primer&#39;</span>]==<span class="st">&#39;16s&#39;</span>,<span class="st">&#39;16s/data&#39;</span>,<span class="st">&#39;data&#39;</span>),
    samples[ii,<span class="st">&#39;sample&#39;</span>], samples[ii,<span class="st">&#39;primer&#39;</span>], <span class="dv">1</span>:<span class="dv">2</span>)
  <span class="kw">message</span>(<span class="st">&#39;Moving &#39;</span>,<span class="kw">paste</span>(sraNames,<span class="dt">collapse=</span><span class="st">&#39; &#39;</span>),<span class="st">&#39; to &#39;</span>,<span class="kw">paste</span>(newNames,<span class="dt">collapse=</span><span class="st">&#39; &#39;</span>))
  <span class="kw">file.rename</span>(sraNames,newNames)
}</code></pre></div>
<pre><code>## Moving sra/SRR5656515_1.fastq.gz sra/SRR5656515_2.fastq.gz to 16s/data/TL3814_16s_R1_.fastq.gz 16s/data/TL3814_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656723_1.fastq.gz sra/SRR5656723_2.fastq.gz to data/TL3814_matK_R1_.fastq.gz data/TL3814_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656724_1.fastq.gz sra/SRR5656724_2.fastq.gz to data/TL3814_rbcL_R1_.fastq.gz data/TL3814_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656721_1.fastq.gz sra/SRR5656721_2.fastq.gz to data/TL3816_matK_R1_.fastq.gz data/TL3816_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656722_1.fastq.gz sra/SRR5656722_2.fastq.gz to data/TL3816_rbcL_R1_.fastq.gz data/TL3816_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656520_1.fastq.gz sra/SRR5656520_2.fastq.gz to 16s/data/TL3816_16s_R1_.fastq.gz 16s/data/TL3816_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656517_1.fastq.gz sra/SRR5656517_2.fastq.gz to 16s/data/TL3793_16s_R1_.fastq.gz 16s/data/TL3793_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656719_1.fastq.gz sra/SRR5656719_2.fastq.gz to data/TL3793_matK_R1_.fastq.gz data/TL3793_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656720_1.fastq.gz sra/SRR5656720_2.fastq.gz to data/TL3793_rbcL_R1_.fastq.gz data/TL3793_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656717_1.fastq.gz sra/SRR5656717_2.fastq.gz to data/TL3797_matK_R1_.fastq.gz data/TL3797_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656514_1.fastq.gz sra/SRR5656514_2.fastq.gz to 16s/data/TL3797_16s_R1_.fastq.gz 16s/data/TL3797_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656521_1.fastq.gz sra/SRR5656521_2.fastq.gz to 16s/data/TL3820_16s_R1_.fastq.gz 16s/data/TL3820_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656581_1.fastq.gz sra/SRR5656581_2.fastq.gz to data/TL3943_rbcL_R1_.fastq.gz data/TL3943_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656582_1.fastq.gz sra/SRR5656582_2.fastq.gz to data/TL3943_matK_R1_.fastq.gz data/TL3943_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656620_1.fastq.gz sra/SRR5656620_2.fastq.gz to data/TL3944_matK_R1_.fastq.gz data/TL3944_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656621_1.fastq.gz sra/SRR5656621_2.fastq.gz to data/TL3944_rbcL_R1_.fastq.gz data/TL3944_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656646_1.fastq.gz sra/SRR5656646_2.fastq.gz to data/IK3358_rbcL_R1_.fastq.gz data/IK3358_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656647_1.fastq.gz sra/SRR5656647_2.fastq.gz to data/IK3358_matK_R1_.fastq.gz data/IK3358_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656699_1.fastq.gz sra/SRR5656699_2.fastq.gz to data/TL3946_matK_R1_.fastq.gz data/TL3946_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656700_1.fastq.gz sra/SRR5656700_2.fastq.gz to data/TL3946_rbcL_R1_.fastq.gz data/TL3946_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656701_1.fastq.gz sra/SRR5656701_2.fastq.gz to data/TL3945_matK_R1_.fastq.gz data/TL3945_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656702_1.fastq.gz sra/SRR5656702_2.fastq.gz to data/TL3945_rbcL_R1_.fastq.gz data/TL3945_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656710_1.fastq.gz sra/SRR5656710_2.fastq.gz to 16s/data/IK3358_16s_R1_.fastq.gz 16s/data/IK3358_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656711_1.fastq.gz sra/SRR5656711_2.fastq.gz to 16s/data/TL3944_16s_R1_.fastq.gz 16s/data/TL3944_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656712_1.fastq.gz sra/SRR5656712_2.fastq.gz to 16s/data/TL3943_16s_R1_.fastq.gz 16s/data/TL3943_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656713_1.fastq.gz sra/SRR5656713_2.fastq.gz to 16s/data/TL3946_16s_R1_.fastq.gz 16s/data/TL3946_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656714_1.fastq.gz sra/SRR5656714_2.fastq.gz to 16s/data/TL3945_16s_R1_.fastq.gz 16s/data/TL3945_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656715_1.fastq.gz sra/SRR5656715_2.fastq.gz to data/TL3820_matK_R1_.fastq.gz data/TL3820_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656716_1.fastq.gz sra/SRR5656716_2.fastq.gz to data/TL3820_rbcL_R1_.fastq.gz data/TL3820_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656718_1.fastq.gz sra/SRR5656718_2.fastq.gz to data/TL3797_rbcL_R1_.fastq.gz data/TL3797_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656432_1.fastq.gz sra/SRR5656432_2.fastq.gz to 16s/data/BI0260_16s_R1_.fastq.gz 16s/data/BI0260_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656433_1.fastq.gz sra/SRR5656433_2.fastq.gz to 16s/data/BI2414_16s_R1_.fastq.gz 16s/data/BI2414_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656479_1.fastq.gz sra/SRR5656479_2.fastq.gz to 16s/data/IK4184_16s_R1_.fastq.gz 16s/data/IK4184_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656604_1.fastq.gz sra/SRR5656604_2.fastq.gz to data/IK4184_matK_R1_.fastq.gz data/IK4184_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656642_1.fastq.gz sra/SRR5656642_2.fastq.gz to data/IK3158_rbcL_R1_.fastq.gz data/IK3158_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656643_1.fastq.gz sra/SRR5656643_2.fastq.gz to data/IK3158_matK_R1_.fastq.gz data/IK3158_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656644_1.fastq.gz sra/SRR5656644_2.fastq.gz to data/IK3276_rbcL_R1_.fastq.gz data/IK3276_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656645_1.fastq.gz sra/SRR5656645_2.fastq.gz to data/IK3276_matK_R1_.fastq.gz data/IK3276_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656648_1.fastq.gz sra/SRR5656648_2.fastq.gz to data/IK3469_rbcL_R1_.fastq.gz data/IK3469_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656649_1.fastq.gz sra/SRR5656649_2.fastq.gz to data/IK3469_matK_R1_.fastq.gz data/IK3469_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656652_1.fastq.gz sra/SRR5656652_2.fastq.gz to data/UB1452_rbcL_R1_.fastq.gz data/UB1452_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656654_1.fastq.gz sra/SRR5656654_2.fastq.gz to data/UB1454_rbcL_R1_.fastq.gz data/UB1454_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656655_1.fastq.gz sra/SRR5656655_2.fastq.gz to data/UB1454_matK_R1_.fastq.gz data/UB1454_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656662_1.fastq.gz sra/SRR5656662_2.fastq.gz to 16s/data/BI2415_16s_R1_.fastq.gz 16s/data/BI2415_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656664_1.fastq.gz sra/SRR5656664_2.fastq.gz to data/UB2037_rbcL_R1_.fastq.gz data/UB2037_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656665_1.fastq.gz sra/SRR5656665_2.fastq.gz to data/UB2037_matK_R1_.fastq.gz data/UB2037_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656666_1.fastq.gz sra/SRR5656666_2.fastq.gz to 16s/data/BI0248_16s_R1_.fastq.gz 16s/data/BI0248_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656667_1.fastq.gz sra/SRR5656667_2.fastq.gz to 16s/data/BI0257_16s_R1_.fastq.gz 16s/data/BI0257_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656674_1.fastq.gz sra/SRR5656674_2.fastq.gz to data/BI0054_matK_R1_.fastq.gz data/BI0054_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656675_1.fastq.gz sra/SRR5656675_2.fastq.gz to data/BI0054_rbcL_R1_.fastq.gz data/BI0054_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656686_1.fastq.gz sra/SRR5656686_2.fastq.gz to 16s/data/UB1454_16s_R1_.fastq.gz 16s/data/UB1454_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656687_1.fastq.gz sra/SRR5656687_2.fastq.gz to 16s/data/UB2037_16s_R1_.fastq.gz 16s/data/UB2037_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656688_1.fastq.gz sra/SRR5656688_2.fastq.gz to data/BI2415_matK_R1_.fastq.gz data/BI2415_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656689_1.fastq.gz sra/SRR5656689_2.fastq.gz to data/BI2415_rbcL_R1_.fastq.gz data/BI2415_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656690_1.fastq.gz sra/SRR5656690_2.fastq.gz to data/BI0260_matK_R1_.fastq.gz data/BI0260_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656691_1.fastq.gz sra/SRR5656691_2.fastq.gz to data/BI0260_rbcL_R1_.fastq.gz data/BI0260_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656692_1.fastq.gz sra/SRR5656692_2.fastq.gz to data/BI2414_matK_R1_.fastq.gz data/BI2414_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656693_1.fastq.gz sra/SRR5656693_2.fastq.gz to data/BI2414_rbcL_R1_.fastq.gz data/BI2414_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656694_1.fastq.gz sra/SRR5656694_2.fastq.gz to data/BI0248_matK_R1_.fastq.gz data/BI0248_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656695_1.fastq.gz sra/SRR5656695_2.fastq.gz to data/BI0248_rbcL_R1_.fastq.gz data/BI0248_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656696_1.fastq.gz sra/SRR5656696_2.fastq.gz to data/BI0257_matK_R1_.fastq.gz data/BI0257_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656697_1.fastq.gz sra/SRR5656697_2.fastq.gz to data/BI0257_rbcL_R1_.fastq.gz data/BI0257_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656698_1.fastq.gz sra/SRR5656698_2.fastq.gz to data/IK4184_rbcL_R1_.fastq.gz data/IK4184_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656703_1.fastq.gz sra/SRR5656703_2.fastq.gz to 16s/data/IK3276_16s_R1_.fastq.gz 16s/data/IK3276_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656704_1.fastq.gz sra/SRR5656704_2.fastq.gz to 16s/data/IK3158_16s_R1_.fastq.gz 16s/data/IK3158_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656705_1.fastq.gz sra/SRR5656705_2.fastq.gz to data/TL3948_matK_R1_.fastq.gz data/TL3948_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656706_1.fastq.gz sra/SRR5656706_2.fastq.gz to data/TL3948_rbcL_R1_.fastq.gz data/TL3948_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656707_1.fastq.gz sra/SRR5656707_2.fastq.gz to 16s/data/BI0054_16s_R1_.fastq.gz 16s/data/BI0054_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656708_1.fastq.gz sra/SRR5656708_2.fastq.gz to 16s/data/TL3948_16s_R1_.fastq.gz 16s/data/TL3948_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656709_1.fastq.gz sra/SRR5656709_2.fastq.gz to 16s/data/IK3469_16s_R1_.fastq.gz 16s/data/IK3469_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656431_1.fastq.gz sra/SRR5656431_2.fastq.gz to 16s/data/PA1038_16s_R1_.fastq.gz 16s/data/PA1038_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656434_1.fastq.gz sra/SRR5656434_2.fastq.gz to 16s/data/PA1059_16s_R1_.fastq.gz 16s/data/PA1059_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656440_1.fastq.gz sra/SRR5656440_2.fastq.gz to data/LK682_rbcL_R1_.fastq.gz data/LK682_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656441_1.fastq.gz sra/SRR5656441_2.fastq.gz to data/LK682_matK_R1_.fastq.gz data/LK682_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656442_1.fastq.gz sra/SRR5656442_2.fastq.gz to data/BI0097_matK_R1_.fastq.gz data/BI0097_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656451_1.fastq.gz sra/SRR5656451_2.fastq.gz to 16s/data/KR57_16s_R1_.fastq.gz 16s/data/KR57_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656462_1.fastq.gz sra/SRR5656462_2.fastq.gz to data/BI0055_matK_R1_.fastq.gz data/BI0055_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656463_1.fastq.gz sra/SRR5656463_2.fastq.gz to data/TL3842_rbcL_R1_.fastq.gz data/TL3842_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656464_1.fastq.gz sra/SRR5656464_2.fastq.gz to data/TL3842_matK_R1_.fastq.gz data/TL3842_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656465_1.fastq.gz sra/SRR5656465_2.fastq.gz to data/BI0055_rbcL_R1_.fastq.gz data/BI0055_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656483_1.fastq.gz sra/SRR5656483_2.fastq.gz to 16s/data/IK3513_16s_R1_.fastq.gz 16s/data/IK3513_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656488_1.fastq.gz sra/SRR5656488_2.fastq.gz to data/TL3889_matK_R1_.fastq.gz data/TL3889_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656489_1.fastq.gz sra/SRR5656489_2.fastq.gz to data/TL3889_rbcL_R1_.fastq.gz data/TL3889_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656490_1.fastq.gz sra/SRR5656490_2.fastq.gz to data/TL3856_matK_R1_.fastq.gz data/TL3856_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656491_1.fastq.gz sra/SRR5656491_2.fastq.gz to data/TL3856_rbcL_R1_.fastq.gz data/TL3856_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656492_1.fastq.gz sra/SRR5656492_2.fastq.gz to data/TL3862_matK_R1_.fastq.gz data/TL3862_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656493_1.fastq.gz sra/SRR5656493_2.fastq.gz to data/TL3862_rbcL_R1_.fastq.gz data/TL3862_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656497_1.fastq.gz sra/SRR5656497_2.fastq.gz to 16s/data/PA1065_16s_R1_.fastq.gz 16s/data/PA1065_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656498_1.fastq.gz sra/SRR5656498_2.fastq.gz to data/TL3905_matK_R1_.fastq.gz data/TL3905_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656499_1.fastq.gz sra/SRR5656499_2.fastq.gz to data/TL3905_rbcL_R1_.fastq.gz data/TL3905_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656500_1.fastq.gz sra/SRR5656500_2.fastq.gz to 16s/data/PA1044_16s_R1_.fastq.gz 16s/data/PA1044_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656501_1.fastq.gz sra/SRR5656501_2.fastq.gz to 16s/data/PA1039_16s_R1_.fastq.gz 16s/data/PA1039_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656512_1.fastq.gz sra/SRR5656512_2.fastq.gz to data/PA1038_rbcL_R1_.fastq.gz data/PA1038_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656513_1.fastq.gz sra/SRR5656513_2.fastq.gz to data/PA0367_matK_R1_.fastq.gz data/PA0367_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656524_1.fastq.gz sra/SRR5656524_2.fastq.gz to data/TL3910_rbcL_R1_.fastq.gz data/TL3910_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656525_1.fastq.gz sra/SRR5656525_2.fastq.gz to data/TL3910_matK_R1_.fastq.gz data/TL3910_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656528_1.fastq.gz sra/SRR5656528_2.fastq.gz to data/TL3915_rbcL_R1_.fastq.gz data/TL3915_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656529_1.fastq.gz sra/SRR5656529_2.fastq.gz to data/TL3915_matK_R1_.fastq.gz data/TL3915_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656535_1.fastq.gz sra/SRR5656535_2.fastq.gz to data/PA0367_rbcL_R1_.fastq.gz data/PA0367_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656544_1.fastq.gz sra/SRR5656544_2.fastq.gz to 16s/data/UB1435_16s_R1_.fastq.gz 16s/data/UB1435_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656545_1.fastq.gz sra/SRR5656545_2.fastq.gz to data/KR57_rbcL_R1_.fastq.gz data/KR57_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656546_1.fastq.gz sra/SRR5656546_2.fastq.gz to data/LK685_matK_R1_.fastq.gz data/LK685_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656548_1.fastq.gz sra/SRR5656548_2.fastq.gz to data/PA0368_matK_R1_.fastq.gz data/PA0368_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656566_1.fastq.gz sra/SRR5656566_2.fastq.gz to 16s/data/PA1049_16s_R1_.fastq.gz 16s/data/PA1049_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656568_1.fastq.gz sra/SRR5656568_2.fastq.gz to data/PA0368_rbcL_R1_.fastq.gz data/PA0368_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656579_1.fastq.gz sra/SRR5656579_2.fastq.gz to data/LK685_rbcL_R1_.fastq.gz data/LK685_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656601_1.fastq.gz sra/SRR5656601_2.fastq.gz to data/PA0370_matK_R1_.fastq.gz data/PA0370_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656602_1.fastq.gz sra/SRR5656602_2.fastq.gz to data/PA0456_rbcL_R1_.fastq.gz data/PA0456_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656603_1.fastq.gz sra/SRR5656603_2.fastq.gz to data/PA0370_rbcL_R1_.fastq.gz data/PA0370_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656612_1.fastq.gz sra/SRR5656612_2.fastq.gz to 16s/data/PA0456_16s_R1_.fastq.gz 16s/data/PA0456_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656619_1.fastq.gz sra/SRR5656619_2.fastq.gz to 16s/data/BI0097_16s_R1_.fastq.gz 16s/data/BI0097_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656629_1.fastq.gz sra/SRR5656629_2.fastq.gz to 16s/data/TL3862_16s_R1_.fastq.gz 16s/data/TL3862_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656630_1.fastq.gz sra/SRR5656630_2.fastq.gz to 16s/data/TL3856_16s_R1_.fastq.gz 16s/data/TL3856_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656631_1.fastq.gz sra/SRR5656631_2.fastq.gz to 16s/data/TL3842_16s_R1_.fastq.gz 16s/data/TL3842_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656632_1.fastq.gz sra/SRR5656632_2.fastq.gz to 16s/data/LK685_16s_R1_.fastq.gz 16s/data/LK685_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656633_1.fastq.gz sra/SRR5656633_2.fastq.gz to 16s/data/TL3910_16s_R1_.fastq.gz 16s/data/TL3910_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656634_1.fastq.gz sra/SRR5656634_2.fastq.gz to 16s/data/TL3905_16s_R1_.fastq.gz 16s/data/TL3905_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656635_1.fastq.gz sra/SRR5656635_2.fastq.gz to 16s/data/TL3889_16s_R1_.fastq.gz 16s/data/TL3889_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656636_1.fastq.gz sra/SRR5656636_2.fastq.gz to 16s/data/LK682_16s_R1_.fastq.gz 16s/data/LK682_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656637_1.fastq.gz sra/SRR5656637_2.fastq.gz to 16s/data/TL3915_16s_R1_.fastq.gz 16s/data/TL3915_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656638_1.fastq.gz sra/SRR5656638_2.fastq.gz to 16s/data/BI0093_16s_R1_.fastq.gz 16s/data/BI0093_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656639_1.fastq.gz sra/SRR5656639_2.fastq.gz to 16s/data/BI0055_16s_R1_.fastq.gz 16s/data/BI0055_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656640_1.fastq.gz sra/SRR5656640_2.fastq.gz to data/PA0456_matK_R1_.fastq.gz data/PA0456_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656641_1.fastq.gz sra/SRR5656641_2.fastq.gz to data/PA1038_matK_R1_.fastq.gz data/PA1038_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656650_1.fastq.gz sra/SRR5656650_2.fastq.gz to data/IK3513_rbcL_R1_.fastq.gz data/IK3513_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656651_1.fastq.gz sra/SRR5656651_2.fastq.gz to data/IK3513_matK_R1_.fastq.gz data/IK3513_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656653_1.fastq.gz sra/SRR5656653_2.fastq.gz to data/UB1452_matK_R1_.fastq.gz data/UB1452_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656656_1.fastq.gz sra/SRR5656656_2.fastq.gz to data/UB1435_rbcL_R1_.fastq.gz data/UB1435_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656657_1.fastq.gz sra/SRR5656657_2.fastq.gz to data/UB1435_matK_R1_.fastq.gz data/UB1435_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656658_1.fastq.gz sra/SRR5656658_2.fastq.gz to data/UB1446_rbcL_R1_.fastq.gz data/UB1446_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656659_1.fastq.gz sra/SRR5656659_2.fastq.gz to data/UB1446_matK_R1_.fastq.gz data/UB1446_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656660_1.fastq.gz sra/SRR5656660_2.fastq.gz to 16s/data/PA0368_16s_R1_.fastq.gz 16s/data/PA0368_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656661_1.fastq.gz sra/SRR5656661_2.fastq.gz to 16s/data/PA0370_16s_R1_.fastq.gz 16s/data/PA0370_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656663_1.fastq.gz sra/SRR5656663_2.fastq.gz to 16s/data/PA0367_16s_R1_.fastq.gz 16s/data/PA0367_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656668_1.fastq.gz sra/SRR5656668_2.fastq.gz to data/KR57_matK_R1_.fastq.gz data/KR57_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656669_1.fastq.gz sra/SRR5656669_2.fastq.gz to data/BI0097_rbcL_R1_.fastq.gz data/BI0097_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656670_1.fastq.gz sra/SRR5656670_2.fastq.gz to data/BI0093_matK_R1_.fastq.gz data/BI0093_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656671_1.fastq.gz sra/SRR5656671_2.fastq.gz to data/BI0093_rbcL_R1_.fastq.gz data/BI0093_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656672_1.fastq.gz sra/SRR5656672_2.fastq.gz to data/PA1065_rbcL_R1_.fastq.gz data/PA1065_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656673_1.fastq.gz sra/SRR5656673_2.fastq.gz to data/PA1065_matK_R1_.fastq.gz data/PA1065_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656676_1.fastq.gz sra/SRR5656676_2.fastq.gz to data/PA1049_rbcL_R1_.fastq.gz data/PA1049_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656677_1.fastq.gz sra/SRR5656677_2.fastq.gz to data/PA1049_matK_R1_.fastq.gz data/PA1049_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656678_1.fastq.gz sra/SRR5656678_2.fastq.gz to data/PA1059_rbcL_R1_.fastq.gz data/PA1059_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656679_1.fastq.gz sra/SRR5656679_2.fastq.gz to data/PA1059_matK_R1_.fastq.gz data/PA1059_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656680_1.fastq.gz sra/SRR5656680_2.fastq.gz to data/PA1039_rbcL_R1_.fastq.gz data/PA1039_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656681_1.fastq.gz sra/SRR5656681_2.fastq.gz to data/PA1039_matK_R1_.fastq.gz data/PA1039_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656682_1.fastq.gz sra/SRR5656682_2.fastq.gz to data/PA1044_rbcL_R1_.fastq.gz data/PA1044_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656683_1.fastq.gz sra/SRR5656683_2.fastq.gz to data/PA1044_matK_R1_.fastq.gz data/PA1044_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656684_1.fastq.gz sra/SRR5656684_2.fastq.gz to 16s/data/UB1446_16s_R1_.fastq.gz 16s/data/UB1446_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656685_1.fastq.gz sra/SRR5656685_2.fastq.gz to 16s/data/UB1452_16s_R1_.fastq.gz 16s/data/UB1452_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656435_1.fastq.gz sra/SRR5656435_2.fastq.gz to data/BI0246_rbcL_R1_.fastq.gz data/BI0246_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656436_1.fastq.gz sra/SRR5656436_2.fastq.gz to data/BI0246_matK_R1_.fastq.gz data/BI0246_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656437_1.fastq.gz sra/SRR5656437_2.fastq.gz to 16s/data/TL3916_16s_R1_.fastq.gz 16s/data/TL3916_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656438_1.fastq.gz sra/SRR5656438_2.fastq.gz to 16s/data/LG4314_16s_R1_.fastq.gz 16s/data/LG4314_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656439_1.fastq.gz sra/SRR5656439_2.fastq.gz to 16s/data/LG4322_16s_R1_.fastq.gz 16s/data/LG4322_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656443_1.fastq.gz sra/SRR5656443_2.fastq.gz to data/LK670_matK_R1_.fastq.gz data/LK670_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656444_1.fastq.gz sra/SRR5656444_2.fastq.gz to data/LK668_rbcL_R1_.fastq.gz data/LK668_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656445_1.fastq.gz sra/SRR5656445_2.fastq.gz to data/LK668_matK_R1_.fastq.gz data/LK668_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656446_1.fastq.gz sra/SRR5656446_2.fastq.gz to 16s/data/KR12_16s_R1_.fastq.gz 16s/data/KR12_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656447_1.fastq.gz sra/SRR5656447_2.fastq.gz to 16s/data/KR21_16s_R1_.fastq.gz 16s/data/KR21_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656448_1.fastq.gz sra/SRR5656448_2.fastq.gz to 16s/data/KR33_16s_R1_.fastq.gz 16s/data/KR33_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656449_1.fastq.gz sra/SRR5656449_2.fastq.gz to 16s/data/KR35_16s_R1_.fastq.gz 16s/data/KR35_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656450_1.fastq.gz sra/SRR5656450_2.fastq.gz to 16s/data/KR52_16s_R1_.fastq.gz 16s/data/KR52_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656452_1.fastq.gz sra/SRR5656452_2.fastq.gz to 16s/data/KR67_16s_R1_.fastq.gz 16s/data/KR67_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656453_1.fastq.gz sra/SRR5656453_2.fastq.gz to 16s/data/LG4300_16s_R1_.fastq.gz 16s/data/LG4300_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656454_1.fastq.gz sra/SRR5656454_2.fastq.gz to data/TL3838_rbcL_R1_.fastq.gz data/TL3838_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656455_1.fastq.gz sra/SRR5656455_2.fastq.gz to data/TL3838_matK_R1_.fastq.gz data/TL3838_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656456_1.fastq.gz sra/SRR5656456_2.fastq.gz to data/TL3826_rbcL_R1_.fastq.gz data/TL3826_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656457_1.fastq.gz sra/SRR5656457_2.fastq.gz to data/TL3826_matK_R1_.fastq.gz data/TL3826_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656458_1.fastq.gz sra/SRR5656458_2.fastq.gz to data/TL3824_rbcL_R1_.fastq.gz data/TL3824_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656459_1.fastq.gz sra/SRR5656459_2.fastq.gz to data/TL3824_matK_R1_.fastq.gz data/TL3824_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656460_1.fastq.gz sra/SRR5656460_2.fastq.gz to data/TL3821_rbcL_R1_.fastq.gz data/TL3821_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656461_1.fastq.gz sra/SRR5656461_2.fastq.gz to data/TL3821_matK_R1_.fastq.gz data/TL3821_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656466_1.fastq.gz sra/SRR5656466_2.fastq.gz to data/LK665_rbcL_R1_.fastq.gz data/LK665_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656467_1.fastq.gz sra/SRR5656467_2.fastq.gz to data/LK647_matK_R1_.fastq.gz data/LK647_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656468_1.fastq.gz sra/SRR5656468_2.fastq.gz to data/LK647_rbcL_R1_.fastq.gz data/LK647_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656469_1.fastq.gz sra/SRR5656469_2.fastq.gz to data/LK645_matK_R1_.fastq.gz data/LK645_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656470_1.fastq.gz sra/SRR5656470_2.fastq.gz to data/LK645_rbcL_R1_.fastq.gz data/LK645_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656471_1.fastq.gz sra/SRR5656471_2.fastq.gz to data/LK661_matK_R1_.fastq.gz data/LK661_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656472_1.fastq.gz sra/SRR5656472_2.fastq.gz to data/LK661_rbcL_R1_.fastq.gz data/LK661_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656473_1.fastq.gz sra/SRR5656473_2.fastq.gz to data/LK653_matK_R1_.fastq.gz data/LK653_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656474_1.fastq.gz sra/SRR5656474_2.fastq.gz to data/LK653_rbcL_R1_.fastq.gz data/LK653_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656475_1.fastq.gz sra/SRR5656475_2.fastq.gz to data/LK665_matK_R1_.fastq.gz data/LK665_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656476_1.fastq.gz sra/SRR5656476_2.fastq.gz to 16s/data/KR05_16s_R1_.fastq.gz 16s/data/KR05_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656477_1.fastq.gz sra/SRR5656477_2.fastq.gz to 16s/data/KR02_16s_R1_.fastq.gz 16s/data/KR02_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656478_1.fastq.gz sra/SRR5656478_2.fastq.gz to 16s/data/IK4214_16s_R1_.fastq.gz 16s/data/IK4214_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656480_1.fastq.gz sra/SRR5656480_2.fastq.gz to 16s/data/IK3777_16s_R1_.fastq.gz 16s/data/IK3777_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656481_1.fastq.gz sra/SRR5656481_2.fastq.gz to 16s/data/IK3701_16s_R1_.fastq.gz 16s/data/IK3701_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656482_1.fastq.gz sra/SRR5656482_2.fastq.gz to 16s/data/IK3650_16s_R1_.fastq.gz 16s/data/IK3650_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656484_1.fastq.gz sra/SRR5656484_2.fastq.gz to 16s/data/KR10_16s_R1_.fastq.gz 16s/data/KR10_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656485_1.fastq.gz sra/SRR5656485_2.fastq.gz to 16s/data/KR07_16s_R1_.fastq.gz 16s/data/KR07_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656486_1.fastq.gz sra/SRR5656486_2.fastq.gz to data/TL3882_matK_R1_.fastq.gz data/TL3882_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656487_1.fastq.gz sra/SRR5656487_2.fastq.gz to data/TL3882_rbcL_R1_.fastq.gz data/TL3882_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656494_1.fastq.gz sra/SRR5656494_2.fastq.gz to 16s/data/UB0599_16s_R1_.fastq.gz 16s/data/UB0599_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656495_1.fastq.gz sra/SRR5656495_2.fastq.gz to 16s/data/UB0445_16s_R1_.fastq.gz 16s/data/UB0445_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656496_1.fastq.gz sra/SRR5656496_2.fastq.gz to 16s/data/UB0439_16s_R1_.fastq.gz 16s/data/UB0439_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656502_1.fastq.gz sra/SRR5656502_2.fastq.gz to data/LG4314_rbcL_R1_.fastq.gz data/LG4314_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656503_1.fastq.gz sra/SRR5656503_2.fastq.gz to data/LG4314_matK_R1_.fastq.gz data/LG4314_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656504_1.fastq.gz sra/SRR5656504_2.fastq.gz to data/LG4322_rbcL_R1_.fastq.gz data/LG4322_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656505_1.fastq.gz sra/SRR5656505_2.fastq.gz to data/LG4322_matK_R1_.fastq.gz data/LG4322_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656506_1.fastq.gz sra/SRR5656506_2.fastq.gz to data/KR67_rbcL_R1_.fastq.gz data/KR67_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656507_1.fastq.gz sra/SRR5656507_2.fastq.gz to data/KR67_matK_R1_.fastq.gz data/KR67_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656508_1.fastq.gz sra/SRR5656508_2.fastq.gz to data/LG4300_rbcL_R1_.fastq.gz data/LG4300_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656509_1.fastq.gz sra/SRR5656509_2.fastq.gz to data/LG4300_matK_R1_.fastq.gz data/LG4300_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656510_1.fastq.gz sra/SRR5656510_2.fastq.gz to data/LG4327_rbcL_R1_.fastq.gz data/LG4327_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656511_1.fastq.gz sra/SRR5656511_2.fastq.gz to data/LG4327_matK_R1_.fastq.gz data/LG4327_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656516_1.fastq.gz sra/SRR5656516_2.fastq.gz to 16s/data/LK686_16s_R1_.fastq.gz 16s/data/LK686_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656518_1.fastq.gz sra/SRR5656518_2.fastq.gz to 16s/data/TL3821_16s_R1_.fastq.gz 16s/data/TL3821_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656519_1.fastq.gz sra/SRR5656519_2.fastq.gz to 16s/data/TL3824_16s_R1_.fastq.gz 16s/data/TL3824_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656522_1.fastq.gz sra/SRR5656522_2.fastq.gz to 16s/data/TL3826_16s_R1_.fastq.gz 16s/data/TL3826_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656523_1.fastq.gz sra/SRR5656523_2.fastq.gz to 16s/data/TL3838_16s_R1_.fastq.gz 16s/data/TL3838_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656526_1.fastq.gz sra/SRR5656526_2.fastq.gz to data/TL3911_rbcL_R1_.fastq.gz data/TL3911_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656527_1.fastq.gz sra/SRR5656527_2.fastq.gz to data/TL3911_matK_R1_.fastq.gz data/TL3911_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656530_1.fastq.gz sra/SRR5656530_2.fastq.gz to data/TL3916_rbcL_R1_.fastq.gz data/TL3916_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656531_1.fastq.gz sra/SRR5656531_2.fastq.gz to data/TL3916_matK_R1_.fastq.gz data/TL3916_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656532_1.fastq.gz sra/SRR5656532_2.fastq.gz to data/TL3918_rbcL_R1_.fastq.gz data/TL3918_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656533_1.fastq.gz sra/SRR5656533_2.fastq.gz to data/TL3918_matK_R1_.fastq.gz data/TL3918_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656534_1.fastq.gz sra/SRR5656534_2.fastq.gz to 16s/data/TL3911_16s_R1_.fastq.gz 16s/data/TL3911_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656536_1.fastq.gz sra/SRR5656536_2.fastq.gz to data/KR21_matK_R1_.fastq.gz data/KR21_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656537_1.fastq.gz sra/SRR5656537_2.fastq.gz to data/KR21_rbcL_R1_.fastq.gz data/KR21_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656538_1.fastq.gz sra/SRR5656538_2.fastq.gz to data/KR33_matK_R1_.fastq.gz data/KR33_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656539_1.fastq.gz sra/SRR5656539_2.fastq.gz to data/KR33_rbcL_R1_.fastq.gz data/KR33_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656540_1.fastq.gz sra/SRR5656540_2.fastq.gz to data/KR35_matK_R1_.fastq.gz data/KR35_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656541_1.fastq.gz sra/SRR5656541_2.fastq.gz to data/KR35_rbcL_R1_.fastq.gz data/KR35_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656542_1.fastq.gz sra/SRR5656542_2.fastq.gz to data/KR52_matK_R1_.fastq.gz data/KR52_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656543_1.fastq.gz sra/SRR5656543_2.fastq.gz to data/KR52_rbcL_R1_.fastq.gz data/KR52_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656547_1.fastq.gz sra/SRR5656547_2.fastq.gz to 16s/data/UB1430_16s_R1_.fastq.gz 16s/data/UB1430_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656549_1.fastq.gz sra/SRR5656549_2.fastq.gz to data/LK670_rbcL_R1_.fastq.gz data/LK670_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656550_1.fastq.gz sra/SRR5656550_2.fastq.gz to 16s/data/LK645_16s_R1_.fastq.gz 16s/data/LK645_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656551_1.fastq.gz sra/SRR5656551_2.fastq.gz to 16s/data/LG4327_16s_R1_.fastq.gz 16s/data/LG4327_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656552_1.fastq.gz sra/SRR5656552_2.fastq.gz to 16s/data/LK653_16s_R1_.fastq.gz 16s/data/LK653_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656553_1.fastq.gz sra/SRR5656553_2.fastq.gz to 16s/data/LK647_16s_R1_.fastq.gz 16s/data/LK647_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656554_1.fastq.gz sra/SRR5656554_2.fastq.gz to 16s/data/LK665_16s_R1_.fastq.gz 16s/data/LK665_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656555_1.fastq.gz sra/SRR5656555_2.fastq.gz to 16s/data/LK661_16s_R1_.fastq.gz 16s/data/LK661_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656556_1.fastq.gz sra/SRR5656556_2.fastq.gz to data/TL3932_matK_R1_.fastq.gz data/TL3932_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656557_1.fastq.gz sra/SRR5656557_2.fastq.gz to data/TL3932_rbcL_R1_.fastq.gz data/TL3932_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656558_1.fastq.gz sra/SRR5656558_2.fastq.gz to data/TL3929_matK_R1_.fastq.gz data/TL3929_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656559_1.fastq.gz sra/SRR5656559_2.fastq.gz to data/TL3929_rbcL_R1_.fastq.gz data/TL3929_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656560_1.fastq.gz sra/SRR5656560_2.fastq.gz to data/TL3927_matK_R1_.fastq.gz data/TL3927_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656561_1.fastq.gz sra/SRR5656561_2.fastq.gz to data/TL3927_rbcL_R1_.fastq.gz data/TL3927_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656562_1.fastq.gz sra/SRR5656562_2.fastq.gz to data/TL3926_matK_R1_.fastq.gz data/TL3926_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656563_1.fastq.gz sra/SRR5656563_2.fastq.gz to data/TL3926_rbcL_R1_.fastq.gz data/TL3926_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656564_1.fastq.gz sra/SRR5656564_2.fastq.gz to data/TL3925_matK_R1_.fastq.gz data/TL3925_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656565_1.fastq.gz sra/SRR5656565_2.fastq.gz to data/TL3925_rbcL_R1_.fastq.gz data/TL3925_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656567_1.fastq.gz sra/SRR5656567_2.fastq.gz to 16s/data/TL3939_16s_R1_.fastq.gz 16s/data/TL3939_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656569_1.fastq.gz sra/SRR5656569_2.fastq.gz to data/KR05_rbcL_R1_.fastq.gz data/KR05_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656570_1.fastq.gz sra/SRR5656570_2.fastq.gz to data/KR05_matK_R1_.fastq.gz data/KR05_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656571_1.fastq.gz sra/SRR5656571_2.fastq.gz to data/KR02_rbcL_R1_.fastq.gz data/KR02_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656572_1.fastq.gz sra/SRR5656572_2.fastq.gz to data/KR02_matK_R1_.fastq.gz data/KR02_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656573_1.fastq.gz sra/SRR5656573_2.fastq.gz to data/KR10_rbcL_R1_.fastq.gz data/KR10_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656574_1.fastq.gz sra/SRR5656574_2.fastq.gz to data/KR10_matK_R1_.fastq.gz data/KR10_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656575_1.fastq.gz sra/SRR5656575_2.fastq.gz to data/KR07_rbcL_R1_.fastq.gz data/KR07_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656576_1.fastq.gz sra/SRR5656576_2.fastq.gz to data/KR07_matK_R1_.fastq.gz data/KR07_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656577_1.fastq.gz sra/SRR5656577_2.fastq.gz to data/KR12_rbcL_R1_.fastq.gz data/KR12_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656578_1.fastq.gz sra/SRR5656578_2.fastq.gz to data/KR12_matK_R1_.fastq.gz data/KR12_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656580_1.fastq.gz sra/SRR5656580_2.fastq.gz to data/LK686_matK_R1_.fastq.gz data/LK686_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656583_1.fastq.gz sra/SRR5656583_2.fastq.gz to data/TL3939_rbcL_R1_.fastq.gz data/TL3939_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656584_1.fastq.gz sra/SRR5656584_2.fastq.gz to data/TL3939_matK_R1_.fastq.gz data/TL3939_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656585_1.fastq.gz sra/SRR5656585_2.fastq.gz to data/TL3936_rbcL_R1_.fastq.gz data/TL3936_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656586_1.fastq.gz sra/SRR5656586_2.fastq.gz to data/TL3936_matK_R1_.fastq.gz data/TL3936_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656587_1.fastq.gz sra/SRR5656587_2.fastq.gz to data/TL3942_rbcL_R1_.fastq.gz data/TL3942_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656588_1.fastq.gz sra/SRR5656588_2.fastq.gz to data/TL3942_matK_R1_.fastq.gz data/TL3942_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656589_1.fastq.gz sra/SRR5656589_2.fastq.gz to data/TL3940_rbcL_R1_.fastq.gz data/TL3940_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656590_1.fastq.gz sra/SRR5656590_2.fastq.gz to data/TL3940_matK_R1_.fastq.gz data/TL3940_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656591_1.fastq.gz sra/SRR5656591_2.fastq.gz to 16s/data/TL3929_16s_R1_.fastq.gz 16s/data/TL3929_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656592_1.fastq.gz sra/SRR5656592_2.fastq.gz to 16s/data/TL3932_16s_R1_.fastq.gz 16s/data/TL3932_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656593_1.fastq.gz sra/SRR5656593_2.fastq.gz to 16s/data/TL3936_16s_R1_.fastq.gz 16s/data/TL3936_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656594_1.fastq.gz sra/SRR5656594_2.fastq.gz to 16s/data/LK670_16s_R1_.fastq.gz 16s/data/LK670_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656595_1.fastq.gz sra/SRR5656595_2.fastq.gz to 16s/data/TL3918_16s_R1_.fastq.gz 16s/data/TL3918_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656596_1.fastq.gz sra/SRR5656596_2.fastq.gz to 16s/data/TL3925_16s_R1_.fastq.gz 16s/data/TL3925_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656597_1.fastq.gz sra/SRR5656597_2.fastq.gz to 16s/data/TL3926_16s_R1_.fastq.gz 16s/data/TL3926_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656598_1.fastq.gz sra/SRR5656598_2.fastq.gz to 16s/data/TL3927_16s_R1_.fastq.gz 16s/data/TL3927_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656599_1.fastq.gz sra/SRR5656599_2.fastq.gz to 16s/data/TL3940_16s_R1_.fastq.gz 16s/data/TL3940_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656600_1.fastq.gz sra/SRR5656600_2.fastq.gz to 16s/data/TL3942_16s_R1_.fastq.gz 16s/data/TL3942_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656605_1.fastq.gz sra/SRR5656605_2.fastq.gz to 16s/data/LK668_16s_R1_.fastq.gz 16s/data/LK668_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656606_1.fastq.gz sra/SRR5656606_2.fastq.gz to data/IK3777_matK_R1_.fastq.gz data/IK3777_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656607_1.fastq.gz sra/SRR5656607_2.fastq.gz to data/IK3777_rbcL_R1_.fastq.gz data/IK3777_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656608_1.fastq.gz sra/SRR5656608_2.fastq.gz to data/IK3701_matK_R1_.fastq.gz data/IK3701_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656609_1.fastq.gz sra/SRR5656609_2.fastq.gz to data/IK3701_rbcL_R1_.fastq.gz data/IK3701_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656610_1.fastq.gz sra/SRR5656610_2.fastq.gz to data/IK3650_matK_R1_.fastq.gz data/IK3650_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656611_1.fastq.gz sra/SRR5656611_2.fastq.gz to data/IK3650_rbcL_R1_.fastq.gz data/IK3650_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656613_1.fastq.gz sra/SRR5656613_2.fastq.gz to 16s/data/BI0246_16s_R1_.fastq.gz 16s/data/BI0246_16s_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656614_1.fastq.gz sra/SRR5656614_2.fastq.gz to data/LK686_rbcL_R1_.fastq.gz data/LK686_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656615_1.fastq.gz sra/SRR5656615_2.fastq.gz to data/IK4214_matK_R1_.fastq.gz data/IK4214_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656616_1.fastq.gz sra/SRR5656616_2.fastq.gz to data/IK4214_rbcL_R1_.fastq.gz data/IK4214_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656617_1.fastq.gz sra/SRR5656617_2.fastq.gz to data/UB1430_matK_R1_.fastq.gz data/UB1430_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656618_1.fastq.gz sra/SRR5656618_2.fastq.gz to data/UB1430_rbcL_R1_.fastq.gz data/UB1430_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656622_1.fastq.gz sra/SRR5656622_2.fastq.gz to data/UB0439_matK_R1_.fastq.gz data/UB0439_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656623_1.fastq.gz sra/SRR5656623_2.fastq.gz to data/UB0439_rbcL_R1_.fastq.gz data/UB0439_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656624_1.fastq.gz sra/SRR5656624_2.fastq.gz to data/UB0445_matK_R1_.fastq.gz data/UB0445_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656625_1.fastq.gz sra/SRR5656625_2.fastq.gz to data/UB0445_rbcL_R1_.fastq.gz data/UB0445_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656626_1.fastq.gz sra/SRR5656626_2.fastq.gz to data/UB0599_matK_R1_.fastq.gz data/UB0599_matK_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656627_1.fastq.gz sra/SRR5656627_2.fastq.gz to data/UB0599_rbcL_R1_.fastq.gz data/UB0599_rbcL_R2_.fastq.gz</code></pre>
<pre><code>## Moving sra/SRR5656628_1.fastq.gz sra/SRR5656628_2.fastq.gz to 16s/data/TL3882_16s_R1_.fastq.gz 16s/data/TL3882_16s_R2_.fastq.gz</code></pre>
<h2 id="make-rbcl-and-matk-otus">Make rbcL and matK OTUs</h2>
<h3 id="load-libraries">Load libraries</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># installed from https://github.com/sherrillmix/dnar </span>
<span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(parallel)
<span class="kw">packageVersion</span>(<span class="st">&#39;parallel&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;3.4.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;functions.R&#39;</span>)</code></pre></div>
<h3 id="check-software-versions">Check software versions</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressWarnings</span>(<span class="kw">system</span>(<span class="st">&#39;mafft --help 2&gt;&amp;1&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>))</code></pre></div>
<pre><code>##  [1] &quot;&quot;                                                                               
##  [2] &quot;------------------------------------------------------------------------------&quot; 
##  [3] &quot;  MAFFT v7.310 (2017/Mar/17)&quot;                                                   
##  [4] &quot;  http://mafft.cbrc.jp/alignment/software/&quot;                                     
##  [5] &quot;  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)&quot;                               
##  [6] &quot;------------------------------------------------------------------------------&quot; 
##  [7] &quot;High speed:&quot;                                                                    
##  [8] &quot;  % mafft in &gt; out&quot;                                                             
##  [9] &quot;  % mafft --retree 1 in &gt; out (fast)&quot;                                           
## [10] &quot;&quot;                                                                               
## [11] &quot;High accuracy (for &lt;~200 sequences x &lt;~2,000 aa/nt):&quot;                           
## [12] &quot;  % mafft --maxiterate 1000 --localpair  in &gt; out (% linsi in &gt; out is also ok)&quot;
## [13] &quot;  % mafft --maxiterate 1000 --genafpair  in &gt; out (% einsi in &gt; out)&quot;           
## [14] &quot;  % mafft --maxiterate 1000 --globalpair in &gt; out (% ginsi in &gt; out)&quot;           
## [15] &quot;&quot;                                                                               
## [16] &quot;If unsure which option to use:&quot;                                                 
## [17] &quot;  % mafft --auto in &gt; out&quot;                                                      
## [18] &quot;&quot;                                                                               
## [19] &quot;--op # :         Gap opening penalty, default: 1.53&quot;                            
## [20] &quot;--ep # :         Offset (works like gap extension penalty), default: 0.0&quot;       
## [21] &quot;--maxiterate # : Maximum number of iterative refinement, default: 0&quot;            
## [22] &quot;--clustalout :   Output: clustal format, default: fasta&quot;                        
## [23] &quot;--reorder :      Outorder: aligned, default: input order&quot;                       
## [24] &quot;--quiet :        Do not report progress&quot;                                        
## [25] &quot;--thread # :     Number of threads (if unsure, --thread -1)&quot;                    
## attr(,&quot;status&quot;)
## [1] 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;dpkg -s fasttree 2&gt;&amp;1&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>##  [1] &quot;Package: fasttree&quot;                                                                            
##  [2] &quot;Status: install ok installed&quot;                                                                 
##  [3] &quot;Priority: optional&quot;                                                                           
##  [4] &quot;Section: science&quot;                                                                             
##  [5] &quot;Installed-Size: 432&quot;                                                                          
##  [6] &quot;Maintainer: Ubuntu Developers &lt;ubuntu-devel-discuss@lists.ubuntu.com&gt;&quot;                        
##  [7] &quot;Architecture: amd64&quot;                                                                          
##  [8] &quot;Version: 2.1.8-2&quot;                                                                             
##  [9] &quot;Depends: libc6 (&gt;= 2.14), libgomp1 (&gt;= 4.9)&quot;                                                  
## [10] &quot;Description: phylogenetic trees from alignments of nucleotide or protein sequences&quot;           
## [11] &quot; FastTree infers approximately-maximum-likelihood phylogenetic trees from&quot;                    
## [12] &quot; alignments of nucleotide or protein sequences. It handles alignments&quot;                        
## [13] &quot; with up to a million of sequences in a reasonable amount of time and&quot;                        
## [14] &quot; memory. For large alignments, FastTree is 100-1,000 times faster than&quot;                       
## [15] &quot; PhyML 3.0 or RAxML 7.&quot;                                                                       
## [16] &quot; .&quot;                                                                                           
## [17] &quot; FastTree is more accurate than PhyML 3 with default settings, and much&quot;                      
## [18] &quot; more accurate than the distance-matrix methods that are traditionally&quot;                       
## [19] &quot; used for large alignments. FastTree uses the Jukes-Cantor or generalized&quot;                    
## [20] &quot; time-reversible (GTR) models of nucleotide evolution and the JTT&quot;                            
## [21] &quot; (Jones-Taylor-Thornton 1992) model of amino acid evolution. To account&quot;                      
## [22] &quot; for the varying rates of evolution across sites, FastTree uses a single&quot;                     
## [23] &quot; rate for each site (the \&quot;CAT\&quot; approximation). To quickly estimate the&quot;                     
## [24] &quot; reliability of each split in the tree, FastTree computes local support&quot;                      
## [25] &quot; values with the Shimodaira-Hasegawa test (these are the same as PhyML 3&#39;s&quot;                   
## [26] &quot; \&quot;SH-like local supports\&quot;).&quot;                                                                
## [27] &quot; .&quot;                                                                                           
## [28] &quot; This package contains a single threaded version (fasttree) and a&quot;                            
## [29] &quot; parallel version which uses OpenMP (fasttreMP).&quot;                                             
## [30] &quot;Original-Maintainer: Debian Med Packaging Team &lt;debian-med-packaging@lists.alioth.debian.org&gt;&quot;
## [31] &quot;Homepage: http://www.microbesonline.org/fasttree/&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressWarnings</span>(<span class="kw">system</span>(<span class="st">&#39;swarm --version 2&gt;&amp;1&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>))</code></pre></div>
<pre><code>## [1] &quot;Swarm 1.2.19 [Mar  1 2016 23:41:10]&quot;                                    
## [2] &quot;Copyright (C) 2012-2014 Torbjorn Rognes and Frederic Mahe&quot;              
## [3] &quot;https://github.com/torognes/swarm&quot;                                      
## [4] &quot;&quot;                                                                       
## [5] &quot;Please cite: Mahe F, Rognes T, Quince C, de Vargas C, Dunthorn M (2014)&quot;
## [6] &quot;Swarm: robust and fast clustering method for amplicon-based studies.&quot;   
## [7] &quot;PeerJ 2:e593 http://dx.doi.org/10.7717/peerj.593&quot;                       
## [8] &quot;&quot;                                                                       
## attr(,&quot;status&quot;)
## [1] 1</code></pre>
<h3 id="primers-used">Primers used</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primerSeqs&lt;-<span class="kw">list</span>(<span class="st">&#39;rbcl&#39;</span>=<span class="kw">c</span>(<span class="st">&#39;rbcL634F&#39;</span>=<span class="st">&#39;ATGCGTTGGAGAGACCGTTTC&#39;</span>,<span class="st">&#39;rbcLbR&#39;</span>=<span class="st">&#39;TCGGTYAGAGCRGGCATRTGCCA&#39;</span>),<span class="st">&#39;matk&#39;</span>=<span class="kw">c</span>(<span class="st">&#39;matK472F&#39;</span>=<span class="st">&#39;CCCRTYCATCTGGAAATCTTGGTTC&#39;</span>,<span class="st">&#39;matK1248R&#39;</span>=<span class="st">&#39;GCTRTRATAATGAGAAAGATTTCTGC&#39;</span>))</code></pre></div>
<h3 id="find-fastqs">Find fastqs</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fastqs&lt;-<span class="kw">list.files</span>(<span class="st">&#39;data/&#39;</span>,<span class="st">&#39;_R[12]_.*</span><span class="ch">\\</span><span class="st">.fastq</span><span class="ch">\\</span><span class="st">.gz$&#39;</span>,<span class="dt">recursive=</span><span class="ot">TRUE</span>,<span class="dt">full.names=</span><span class="ot">TRUE</span>)
fastqs&lt;-fastqs[!<span class="kw">grepl</span>(<span class="st">&#39;Undetermined&#39;</span>,fastqs)]
primers&lt;-<span class="kw">sub</span>(<span class="st">&#39;.*(matK|rbcL).*_R([0-9]+)_.*&#39;</span>,<span class="st">&#39;</span><span class="ch">\\</span><span class="st">1</span><span class="ch">\\</span><span class="st">2&#39;</span>,<span class="kw">basename</span>(fastqs))
primerBases&lt;-<span class="kw">sub</span>(<span class="st">&#39;[0-9]$&#39;</span>,<span class="st">&#39;&#39;</span>,primers)</code></pre></div>
<h3 id="run-swarm-mafft-and-fasttree">Run swarm, mafft and fasttree</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">for(primerBase in <span class="kw">unique</span>(primerBases)){
  <span class="kw">message</span>(<span class="st">&#39;Working on &#39;</span>,primerBase)
  if(!<span class="kw">dir.exists</span>(<span class="st">&#39;work/swarmPair&#39;</span>))<span class="kw">dir.create</span>(<span class="st">&#39;work/swarmPair&#39;</span>,<span class="dt">recursive=</span><span class="ot">TRUE</span>)
  outMat&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;work/swarmPair/%s.Rdat&#39;</span>,primerBase)
  outFa&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;work/swarmPair/%s.fa.gz&#39;</span>,primerBase)
  outAlign&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;work/swarmPair/%s_align.fa.gz&#39;</span>,primerBase)
  outTree&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;work/swarmPair/%s_align.tre&#39;</span>,primerBase)
  <span class="kw">message</span>(<span class="st">&#39;Trimming primers off the start of reads&#39;</span>)
  trimReads&lt;-<span class="kw">lapply</span>(<span class="kw">sprintf</span>(<span class="st">&#39;%s%d&#39;</span>,primerBase,<span class="dv">1</span>:<span class="dv">2</span>),function(ii){
    thisPrimer&lt;-primerSeqs[[<span class="kw">sub</span>(<span class="st">&#39;[12]$&#39;</span>,<span class="st">&#39;&#39;</span>,<span class="kw">tolower</span>(ii))]][<span class="kw">as.numeric</span>(<span class="kw">substring</span>(ii,<span class="kw">nchar</span>(ii)))]
    thisFiles&lt;-fastqs[primers==ii]
    reads&lt;-<span class="kw">mclapply</span>(thisFiles,function(xx){<span class="kw">library</span>(dnar);<span class="kw">cat</span>(<span class="st">&#39;.&#39;</span>);<span class="kw">read.fastq</span>(xx)},<span class="dt">mc.cores=</span><span class="dv">20</span>,<span class="dt">mc.preschedule=</span><span class="ot">FALSE</span>)
    if(<span class="kw">mean</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(reads,function(xx)<span class="kw">substring</span>(xx$seq,<span class="dv">1</span>,<span class="kw">nchar</span>(thisPrimer))))%in%<span class="st"> </span><span class="kw">expandAmbiguous</span>(thisPrimer)[[<span class="dv">1</span>]])&lt;.<span class="dv">75</span>)<span class="kw">stop</span>(<span class="kw">simpleError</span>(<span class="st">&#39;Expected primer does not match read start&#39;</span>))
    trimReads&lt;-<span class="kw">lapply</span>(reads,function(xx){xx$seq&lt;-<span class="kw">substring</span>(xx$seq,<span class="kw">nchar</span>(thisPrimer)+<span class="dv">1</span>);<span class="kw">return</span>(xx)})
    <span class="kw">names</span>(trimReads)&lt;-thisFiles
    <span class="kw">return</span>(trimReads)
  })
  if(<span class="kw">any</span>(<span class="kw">names</span>(trimReads[[<span class="dv">1</span>]])!=<span class="kw">sub</span>(<span class="st">&#39;_R2_&#39;</span>,<span class="st">&#39;_R1_&#39;</span>,<span class="kw">names</span>(trimReads[[<span class="dv">2</span>]]))))<span class="kw">stop</span>(<span class="st">&#39;Read 1 vs reads 2 file name mismatch&#39;</span>)
  readCounts&lt;-<span class="kw">as.data.frame</span>(<span class="kw">sapply</span>(trimReads,function(xx)<span class="kw">sapply</span>(xx,nrow)))
  <span class="kw">colnames</span>(readCounts)&lt;-<span class="kw">c</span>(<span class="st">&#39;raw1&#39;</span>,<span class="st">&#39;raw2&#39;</span>)
  <span class="kw">message</span>(<span class="st">&#39;Discarding reads with &gt;1 expected error in left or right read and concatenating left-right&#39;</span>)
  trimReads&lt;-<span class="kw">mcmapply</span>(function(left,right,...){
    <span class="kw">cat</span>(<span class="st">&#39;.&#39;</span>)
    if(<span class="kw">any</span>(<span class="kw">sub</span>(<span class="st">&#39; .*$&#39;</span>,<span class="st">&#39;&#39;</span>,left$name)!=<span class="kw">sub</span>(<span class="st">&#39; .*$&#39;</span>,<span class="st">&#39;&#39;</span>,right$name)))<span class="kw">stop</span>(<span class="st">&#39;Read 1 vs read 2 name mismatch&#39;</span>)
    <span class="co">#last base always low qual so ignore</span>
    q1&lt;-<span class="kw">sapply</span>(<span class="kw">qualToInts</span>(<span class="kw">substring</span>(left$qual,<span class="dv">1</span>,<span class="kw">nchar</span>(left$qual)-<span class="dv">1</span>)),function(xx)<span class="kw">sum</span>(<span class="dv">10</span>^(-xx/<span class="dv">10</span>)))
    q2&lt;-<span class="kw">sapply</span>(<span class="kw">qualToInts</span>(<span class="kw">substring</span>(right$qual,<span class="dv">1</span>,<span class="kw">nchar</span>(right$qual)-<span class="dv">1</span>)),function(xx)<span class="kw">sum</span>(<span class="dv">10</span>^(-xx/<span class="dv">10</span>)))
    <span class="co">#less than 1 expected error in both reads</span>
    selector&lt;-q1&lt;<span class="dv">1</span>&amp;q2&lt;<span class="dv">1</span> &amp;<span class="st"> </span>!<span class="kw">grepl</span>(<span class="st">&#39;[^ACTG]&#39;</span>,left$seq)&amp;!<span class="kw">grepl</span>(<span class="st">&#39;[^ACTG]&#39;</span>,right$seq)
    seqs&lt;-<span class="kw">paste</span>(left[selector,<span class="st">&#39;seq&#39;</span>],<span class="kw">revComp</span>(right[selector,<span class="st">&#39;seq&#39;</span>]),<span class="dt">sep=</span><span class="st">&#39;&#39;</span>)
    <span class="kw">return</span>(seqs)
  },trimReads[[<span class="dv">1</span>]],trimReads[[<span class="dv">2</span>]],<span class="dt">mc.cores=</span><span class="dv">10</span>,<span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)
  readCounts$filter&lt;-<span class="kw">sapply</span>(trimReads,length)
  <span class="kw">message</span>(<span class="st">&#39;Read counts&#39;</span>)
  <span class="kw">print</span>(readCounts)
  samples&lt;-<span class="kw">rep</span>(<span class="kw">basename</span>(<span class="kw">names</span>(trimReads)),<span class="kw">sapply</span>(trimReads,length))
  <span class="kw">message</span>(<span class="st">&#39;Running swarm&#39;</span>)
  otus&lt;-<span class="kw">runSwarm</span>(<span class="kw">unlist</span>(trimReads),<span class="st">&#39;~/installs/swarm/swarm&#39;</span>,<span class="dt">swarmArgs=</span><span class="st">&#39;-f -t 40&#39;</span>)
  swarmOtus&lt;-<span class="kw">as.data.frame.matrix</span>(<span class="kw">table</span>(samples,otus[[<span class="st">&#39;otus&#39;</span>]]))
  <span class="kw">write.fa</span>(otus[[<span class="st">&#39;seqs&#39;</span>]]$name,otus[[<span class="st">&#39;seqs&#39;</span>]]$seq,outFa)
  <span class="kw">save</span>(swarmOtus,<span class="dt">file=</span>outMat)
  <span class="kw">load</span>(outMat)
  tmpFile&lt;-<span class="kw">tempfile</span>()
  <span class="kw">message</span>(<span class="st">&#39;Discarding singleton OTUs&#39;</span>)
  swarmOtus&lt;-swarmOtus[,<span class="kw">apply</span>(swarmOtus,<span class="dv">2</span>,sum)&gt;<span class="dv">1</span>]
  seqs&lt;-otus[[<span class="st">&#39;seqs&#39;</span>]]
  <span class="kw">rownames</span>(seqs)&lt;-seqs$name
  <span class="kw">write.fa</span>(<span class="kw">colnames</span>(swarmOtus),seqs[<span class="kw">colnames</span>(swarmOtus),<span class="st">&#39;seq&#39;</span>],tmpFile)
  <span class="kw">message</span>(<span class="st">&#39;Aligning with mafft&#39;</span>)
  cmd&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;~/installs/mafft/bin/mafft --thread 50 %s|gzip&gt;%s&#39;</span>,tmpFile,outAlign)
  <span class="kw">message</span>(cmd)
  <span class="kw">system</span>(cmd)
  <span class="kw">message</span>(<span class="st">&#39;Creating tree with fasttree&#39;</span>)
  cmd&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;zcat %s|fasttree -gtr -nt&gt;%s&#39;</span>,outAlign,outTree)
  <span class="kw">message</span>(cmd)
  <span class="kw">system</span>(cmd)
  <span class="kw">message</span>(<span class="st">&#39;Output files in:&#39;</span>)
  <span class="kw">message</span>(<span class="st">&#39;    &#39;</span>,outMat)
  <span class="kw">message</span>(<span class="st">&#39;    &#39;</span>,outFa)
  <span class="kw">message</span>(<span class="st">&#39;    &#39;</span>,outAlign)
  <span class="kw">message</span>(<span class="st">&#39;    &#39;</span>,outTree)
}</code></pre></div>
<pre><code>## Working on matK</code></pre>
<pre><code>## Trimming primers off the start of reads</code></pre>
<pre><code>## Error in readChild(x): vector is too large</code></pre>
<h2 id="software-versions-1">Software versions</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;makeblastdb -version&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] &quot;makeblastdb: 2.2.31+&quot;                             
## [2] &quot;Package: blast 2.2.31, build Jan  7 2016 23:17:17&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;blastn -version&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] &quot;blastn: 2.2.31+&quot;                                  
## [2] &quot;Package: blast 2.2.31, build Jan  7 2016 23:17:17&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;parallel --version&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>##  [1] &quot;GNU parallel 20141022&quot;                                                                             
##  [2] &quot;Copyright (C) 2007,2008,2009,2010,2011,2012,2013,2014 Ole Tange and Free Software Foundation, Inc.&quot;
##  [3] &quot;License GPLv3+: GNU GPL version 3 or later &lt;http://gnu.org/licenses/gpl.html&gt;&quot;                     
##  [4] &quot;This is free software: you are free to change and redistribute it.&quot;                                
##  [5] &quot;GNU parallel comes with no warranty.&quot;                                                              
##  [6] &quot;&quot;                                                                                                  
##  [7] &quot;Web site: http://www.gnu.org/software/parallel&quot;                                                    
##  [8] &quot;&quot;                                                                                                  
##  [9] &quot;When using programs that use GNU Parallel to process data for publication please cite:&quot;            
## [10] &quot;&quot;                                                                                                  
## [11] &quot;O. Tange (2011): GNU Parallel - The Command-Line Power Tool, &quot;                                     
## [12] &quot;;login: The USENIX Magazine, February 2011:42-47.&quot;                                                 
## [13] &quot;&quot;                                                                                                  
## [14] &quot;Or you can get GNU Parallel without this requirement by paying 10000 EUR.&quot;</code></pre>
<h2 id="prepare-blast-databases">Prepare blast databases</h2>
<p>rbcL and matk data were downloaded manually from the EBI: * <a href="http://www.ebi.ac.uk/ena/data/search?query=rbcl" class="uri">http://www.ebi.ac.uk/ena/data/search?query=rbcl</a> * <a href="http://www.ebi.ac.uk/ena/data/search?query=matk" class="uri">http://www.ebi.ac.uk/ena/data/search?query=matk</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">if(!<span class="kw">file.exists</span>(<span class="st">&#39;ebi/rbcl.fa.gz&#39;</span>))<span class="kw">stop</span>(<span class="st">&#39;Please download rbcl data from http://www.ebi.ac.uk/ena/data/search?query=rbcl to ebi/rbcl.fa.gz&#39;</span>)
if(!<span class="kw">file.exists</span>(<span class="st">&#39;ebi/matk.fa.gz&#39;</span>))<span class="kw">stop</span>(<span class="st">&#39;Please download matk data from http://www.ebi.ac.uk/ena/data/search?query=matk to ebi/matk.fa.gz&#39;</span>)
<span class="kw">system</span>(<span class="st">&#39;zcat ebi/matk.fa.gz | makeblastdb -in - -title matk -dbtype nucl -out work/matk&#39;</span>)
<span class="kw">system</span>(<span class="st">&#39;zcat ebi/rbcl.fa.gz | makeblastdb -in - -title rbcl -dbtype nucl -out work/rbcl&#39;</span>)</code></pre></div>
<h2 id="blast-swarm-otus-against-rbclmatk-databases">Blast swarm OTUs against rbcl/matk databases</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">contigFa&lt;-<span class="kw">list.files</span>(<span class="st">&#39;work/swarmPair&#39;</span>,<span class="st">&#39;.fa.gz$&#39;</span>,<span class="dt">full.names=</span><span class="ot">TRUE</span>)
contigFa&lt;-contigFa[!<span class="kw">grepl</span>(<span class="st">&#39;_align&#39;</span>,contigFa)]
for(fasta in contigFa){
  outFile&lt;-<span class="kw">sub</span>(<span class="st">&#39;fa.gz$&#39;</span>,<span class="st">&#39;blast.gz&#39;</span>,fasta)
  if(<span class="kw">grepl</span>(<span class="st">&#39;rbcL&#39;</span>,fasta))gene&lt;-<span class="st">&#39;rbcl&#39;</span>
  else if(<span class="kw">grepl</span>(<span class="st">&#39;matK&#39;</span>,fasta))gene&lt;-<span class="st">&#39;matk&#39;</span>
  else <span class="kw">stop</span>(<span class="st">&#39;Unknown gene&#39;</span>)
  cmd&lt;-<span class="kw">sprintf</span>(<span class="st">&quot;zcat %s|parallel --block 1m --recstart &#39;&gt;&#39; -L2 -j 14 --pipe blastn -db work/%s -num_threads 5 -culling_limit 30 -outfmt 6 -query -|gzip &gt; %s&quot;</span>,fasta,gene,outFile)
  <span class="kw">message</span>(cmd)
  <span class="kw">system</span>(cmd)
}</code></pre></div>
<pre><code>## zcat work/swarmPair/matK.fa.gz|parallel --block 1m --recstart &#39;&gt;&#39; -L2 -j 14 --pipe blastn -db work/matk -num_threads 5 -culling_limit 30 -outfmt 6 -query -|gzip &gt; work/swarmPair/matK.blast.gz</code></pre>
<pre><code>## zcat work/swarmPair/rbcL.fa.gz|parallel --block 1m --recstart &#39;&gt;&#39; -L2 -j 14 --pipe blastn -db work/rbcl -num_threads 5 -culling_limit 30 -outfmt 6 -query -|gzip &gt; work/swarmPair/rbcL.blast.gz</code></pre>
<h2 id="load-libraries-1">Load libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(taxonomizr)
<span class="kw">packageVersion</span>(<span class="st">&#39;taxonomizr&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.3.0&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(parallel)
<span class="kw">packageVersion</span>(<span class="st">&#39;parallel&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;3.4.1&#39;</code></pre>
<h2 id="download-ncbi-accession-numbers-and-taxonomy">Download NCBI accession numbers and taxonomy</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sqlFile&lt;-<span class="st">&#39;work/accessionTaxa.sql&#39;</span>
<span class="kw">getNamesAndNodes</span>(<span class="st">&#39;work&#39;</span>)</code></pre></div>
<pre><code>## work/names.dmp, work/nodes.dmp already exist. Delete to redownload</code></pre>
<pre><code>## [1] &quot;work/names.dmp&quot; &quot;work/nodes.dmp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#this is a big download</span>
<span class="kw">getAccession2taxid</span>(<span class="st">&#39;work&#39;</span>)</code></pre></div>
<pre><code>## This can be a big (several gigabytes) download. Please be patient and use a fast connection.</code></pre>
<pre><code>## [1] &quot;work/nucl_gb.accession2taxid.gz&quot;  &quot;work/nucl_est.accession2taxid.gz&quot;
## [3] &quot;work/nucl_gss.accession2taxid.gz&quot; &quot;work/nucl_wgs.accession2taxid.gz&quot;</code></pre>
<h2 id="prepare-database-of-accession-numbers-and-taxonomy">Prepare database of accession numbers and taxonomy</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">read.nodes2</span>(<span class="st">&#39;work/nodes.dmp&#39;</span>,sqlFile)
<span class="kw">read.names2</span>(<span class="st">&#39;work/names.dmp&#39;</span>,sqlFile)
<span class="kw">read.accession2taxid</span>(<span class="kw">list.files</span>(<span class="st">&#39;work&#39;</span>,<span class="st">&#39;accession2taxid.gz$&#39;</span>,<span class="dt">full.names=</span><span class="ot">TRUE</span>),sqlFile,<span class="dt">overwrite=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Reading work/nucl_est.accession2taxid.gz.</code></pre>
<pre><code>## Reading work/nucl_gb.accession2taxid.gz.</code></pre>
<pre><code>## Reading work/nucl_gss.accession2taxid.gz.</code></pre>
<pre><code>## Reading work/nucl_wgs.accession2taxid.gz.</code></pre>
<pre><code>## Reading in values. This may take a while.</code></pre>
<pre><code>## Adding index. This may also take a while.</code></pre>
<h2 id="assign-taxonomy-to-blast-hits">Assign taxonomy to blast hits</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">blastFiles&lt;-<span class="kw">list.files</span>(<span class="st">&#39;work/swarmPair&#39;</span>,<span class="st">&#39;</span><span class="ch">\\</span><span class="st">.blast</span><span class="ch">\\</span><span class="st">.gz$&#39;</span>,<span class="dt">full.names=</span><span class="ot">TRUE</span>)

<span class="kw">mclapply</span>(blastFiles,function(ii,sqlFile,...){
  <span class="kw">message</span>(<span class="st">&#39;Parsing blast hits in &#39;</span>,ii)
  outFile&lt;-<span class="kw">sub</span>(<span class="st">&#39;.blast.gz$&#39;</span>,<span class="st">&#39;_taxa.csv&#39;</span>,ii)
  outFile2&lt;-<span class="kw">sub</span>(<span class="st">&#39;.blast.gz$&#39;</span>,<span class="st">&#39;_allHits.csv&#39;</span>,ii)
  <span class="kw">message</span>(<span class="st">&#39; Creating &#39;</span>,outFile)
  <span class="kw">message</span>(<span class="st">&#39;  Reading blast&#39;</span>)
  x&lt;-<span class="kw">read.blast</span>(ii)
  x&lt;-x[x$tName!=<span class="st">&#39;ref&#39;</span>,]
  x$accession&lt;-<span class="kw">sapply</span>(<span class="kw">strsplit</span>(x$tName,<span class="st">&#39;</span><span class="ch">\\</span><span class="st">|&#39;</span>),<span class="st">&#39;[[&#39;</span>,<span class="dv">3</span>)
  <span class="kw">message</span>(<span class="st">&#39;  Accession to taxonomy&#39;</span>)
  x$taxa&lt;-<span class="kw">accessionToTaxa</span>(x$accession,sqlFile)
  x$sumScore&lt;-<span class="kw">ave</span>(x$bit,<span class="kw">paste</span>(x$tName,x$qName,<span class="dt">sep=</span><span class="st">&#39;_-_&#39;</span>),<span class="dt">FUN=</span>sum)
  x$maxScore&lt;-<span class="kw">ave</span>(x$sumScore,x$qName,<span class="dt">FUN=</span>max)
  x&lt;-x[x$sumScore&gt;x$maxScore*.<span class="dv">98</span>&amp;!<span class="kw">is.na</span>(x$taxa),]
  <span class="kw">message</span>(<span class="st">&#39;  Getting upstream taxonomy&#39;</span>)
  taxonomy&lt;-<span class="kw">getTaxonomy2</span>(x$taxa,sqlFile)
  taxonomy&lt;-<span class="kw">as.data.frame</span>(taxonomy,<span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
  <span class="kw">message</span>(<span class="st">&#39;  Condensing taxonomy&#39;</span>)
  taxaAssigns&lt;-<span class="kw">condenseTaxa2</span>(taxonomy,x$qName)
  taxaAssigns&lt;-<span class="kw">as.data.frame</span>(taxaAssigns,<span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
  <span class="kw">rownames</span>(taxaAssigns)&lt;-taxaAssigns$id
  taxaAssigns&lt;-taxaAssigns[,<span class="kw">colnames</span>(taxaAssigns)!=<span class="st">&#39;id&#39;</span>]
  taxonomy$qName&lt;-x$qName
  <span class="kw">write.csv</span>(taxonomy,outFile2)
  taxaAssigns$best&lt;-<span class="kw">apply</span>(taxaAssigns,<span class="dv">1</span>,lastNotNa)
  bestScore&lt;-x[x$sumScore==x$maxScore,<span class="kw">c</span>(<span class="st">&#39;qName&#39;</span>,<span class="st">&#39;alignLength&#39;</span>,<span class="st">&#39;percID&#39;</span>,<span class="st">&#39;sumScore&#39;</span>)]
  bestScore&lt;-bestScore[!<span class="kw">duplicated</span>(bestScore$qName),]
  <span class="kw">rownames</span>(bestScore)&lt;-bestScore$qName
  taxaAssigns&lt;-<span class="kw">cbind</span>(taxaAssigns,<span class="st">&#39;bestScore&#39;</span>=bestScore[<span class="kw">rownames</span>(taxaAssigns),<span class="kw">c</span>(<span class="st">&#39;sumScore&#39;</span>)])
  <span class="kw">write.csv</span>(taxaAssigns,outFile)
  <span class="kw">return</span>(<span class="kw">c</span>(<span class="st">&#39;taxa&#39;</span>=outFile,<span class="st">&#39;allHits&#39;</span>=outFile2))
},sqlFile,<span class="dt">mc.cores=</span><span class="dv">2</span>)</code></pre></div>
<pre><code>## [[1]]
##                              taxa                           allHits 
##    &quot;work/swarmPair/matK_taxa.csv&quot; &quot;work/swarmPair/matK_allHits.csv&quot; 
## 
## [[2]]
##                              taxa                           allHits 
##    &quot;work/swarmPair/rbcL_taxa.csv&quot; &quot;work/swarmPair/rbcL_allHits.csv&quot;</code></pre>
<h2 id="load-libraries-2">Load libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phyloseq)
<span class="kw">packageVersion</span>(<span class="st">&#39;phyloseq&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;1.20.0&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ape)
<span class="kw">packageVersion</span>(<span class="st">&#39;ape&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;4.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Rtsne)
<span class="kw">packageVersion</span>(<span class="st">&#39;Rtsne&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.13&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vegan)</code></pre></div>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.4-3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">packageVersion</span>(<span class="st">&#39;vegan&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;2.4.3&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ade4)</code></pre></div>
<pre><code>## 
## Attaching package: &#39;ade4&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:vegan&#39;:
## 
##     cca</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">packageVersion</span>(<span class="st">&#39;ade4&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;1.7.6&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;functions.R&#39;</span>)</code></pre></div>
<h2 id="load-data">Load data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;loadData.R&quot;</span>)</code></pre></div>
<pre><code>## matK</code></pre>
<pre><code>## Cache work/matK_rarefyOtus.Rdat does exist. Loading data</code></pre>
<pre><code>## rbcL</code></pre>
<pre><code>## Cache work/rbcL_rarefyOtus.Rdat does exist. Loading data</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tlAdonis&lt;-interactAdonis&lt;-plantAdonis&lt;-chimpAdonis&lt;-bonoboAdonis&lt;-<span class="kw">list</span>()
mantels&lt;-<span class="kw">list</span>()
for(ii in <span class="kw">names</span>(swarmData)){
  plotProp&lt;-swarmData[[ii]][[<span class="st">&#39;props&#39;</span>]][swarmData[[ii]][[<span class="st">&#39;isEnough&#39;</span>]]&amp;<span class="kw">rownames</span>(swarmData[[ii]][[<span class="st">&#39;props&#39;</span>]]) %in%<span class="st"> </span><span class="kw">rownames</span>(samples),]
  plotProp2&lt;-swarmData[[ii]][[<span class="st">&#39;rare&#39;</span>]][swarmData[[ii]][[<span class="st">&#39;isEnough&#39;</span>]]&amp;<span class="kw">rownames</span>(swarmData[[ii]][[<span class="st">&#39;rare&#39;</span>]]) %in%<span class="st"> </span><span class="kw">rownames</span>(samples),]
  phyOtuW&lt;-<span class="kw">otu_table</span>(plotProp,<span class="dt">taxa_are_rows=</span><span class="ot">FALSE</span>)
  phyOtuU&lt;-<span class="kw">otu_table</span>(plotProp2,<span class="dt">taxa_are_rows=</span><span class="ot">FALSE</span>)
  qiimeDataW&lt;-<span class="kw">phyloseq</span>(<span class="dt">otu_table=</span>phyOtuW,<span class="dt">phy_tree=</span>swarmData[[ii]][[<span class="st">&#39;tree&#39;</span>]])
  qiimeDataU&lt;-<span class="kw">phyloseq</span>(<span class="dt">otu_table=</span>phyOtuU,<span class="dt">phy_tree=</span>swarmData[[ii]][[<span class="st">&#39;tree&#39;</span>]])
  brayDist&lt;-<span class="kw">distance</span>(qiimeDataU,<span class="st">&#39;bray&#39;</span>,<span class="dt">binary=</span><span class="ot">TRUE</span>)
  brayDistW&lt;-<span class="kw">distance</span>(qiimeDataW,<span class="st">&#39;bray&#39;</span>,<span class="dt">binary=</span><span class="ot">FALSE</span>)
  uniDist&lt;-<span class="kw">UniFrac</span>(qiimeDataU,<span class="dt">weighted=</span><span class="ot">FALSE</span>)
  uniDistW&lt;-<span class="kw">UniFrac</span>(qiimeDataW,<span class="dt">weighted=</span><span class="ot">TRUE</span>)
  mantels[[ii]]&lt;-<span class="kw">list</span>(
    <span class="st">&#39;uniW&#39;</span>=ade4::<span class="kw">mantel.rtest</span>(uniDist,uniDistW,<span class="dt">nrepet=</span><span class="fl">1e4</span>),
    <span class="st">&#39;brayW&#39;</span>=ade4::<span class="kw">mantel.rtest</span>(uniDist,brayDistW,<span class="dt">nrepet=</span><span class="fl">1e4</span>),
    <span class="st">&#39;brayUW&#39;</span>=ade4::<span class="kw">mantel.rtest</span>(uniDist,brayDist,<span class="dt">nrepet=</span><span class="fl">1e4</span>)
  )
  uniPca&lt;-<span class="kw">pcoa</span>(uniDist)
  tsneUni&lt;-<span class="kw">Rtsne</span>(uniDist,<span class="dt">is_distance=</span><span class="ot">TRUE</span>,<span class="dt">verbose=</span><span class="ot">TRUE</span>,<span class="dt">perplexity=</span><span class="dv">15</span>,<span class="dt">max_iter=</span><span class="dv">3000</span>)
  selectDist&lt;-uniDist

  importance&lt;-uniPca$values$Relative_eig
  <span class="kw">colnames</span>(uniPca$vectors)&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;Principal coordinate %d (%d%% of variance)&#39;</span>,<span class="dv">1</span>:<span class="kw">length</span>(importance),<span class="kw">round</span>(importance*<span class="dv">100</span>))[<span class="dv">1</span>:<span class="kw">ncol</span>(uniPca$vectors)]
  selectSamples&lt;-samples[<span class="kw">rownames</span>(plotProp),]
  selectSamples&lt;-selectSamples[<span class="kw">order</span>(selectSamples$bonobo,selectSamples$area2,selectSamples$malaria),]
  colorBrew&lt;-<span class="kw">c</span>(<span class="st">&#39;#e41a1cBB&#39;</span>,<span class="st">&#39;#377eb8BB&#39;</span>,<span class="st">&#39;#4daf4aBB&#39;</span>,<span class="st">&#39;#984ea3BB&#39;</span>,<span class="st">&#39;#ff7f00BB&#39;</span>,<span class="st">&#39;#ffff33BB&#39;</span>,<span class="st">&#39;#a65628BB&#39;</span>,<span class="st">&#39;#f781bfBB&#39;</span>,<span class="st">&#39;#999999BB&#39;</span>,<span class="st">&#39;#88ddffBB&#39;</span>)
  nArea&lt;-<span class="kw">length</span>(<span class="kw">unique</span>(selectSamples$area2))
  if(nArea&gt;<span class="kw">length</span>(colorBrew))<span class="kw">stop</span>(<span class="st">&#39;Need to adjust colors for more areas&#39;</span>)
  areaCols&lt;-colorBrew[<span class="dv">1</span>:nArea]
  <span class="kw">names</span>(areaCols)&lt;-<span class="kw">unique</span>(selectSamples$area2[<span class="kw">order</span>(selectSamples$chimpBonobo)])
  areaPch&lt;-<span class="kw">sapply</span>(<span class="kw">names</span>(areaCols),function(x)<span class="kw">mostAbundant</span>(selectSamples$chimpBonobo[selectSamples$area2==x]))
  malariaCols3&lt;-<span class="kw">c</span>(<span class="st">&#39;#00000022&#39;</span>,<span class="st">&#39;#000000CC&#39;</span>)
  mediumMalariaCol&lt;-<span class="st">&#39;#00000077&#39;</span>
  malariaCols&lt;-<span class="kw">c</span>(<span class="st">&#39;#00000022&#39;</span>,<span class="st">&#39;#000000CC&#39;</span>)
  malariaCols2&lt;-<span class="kw">rainbow.lab</span>(<span class="dv">2</span>,<span class="dt">alpha=</span>.<span class="dv">9</span>,<span class="dt">lightMultiple=</span>.<span class="dv">7</span>)
  <span class="kw">names</span>(malariaCols3)&lt;-<span class="kw">names</span>(malariaCols2)&lt;-<span class="kw">names</span>(malariaCols)&lt;-<span class="kw">c</span>(<span class="st">&#39;Laverania negative&#39;</span>,<span class="st">&#39;Laverania positive&#39;</span>)
  speciesPch&lt;-<span class="dv">20+1</span>:<span class="kw">length</span>(<span class="kw">unique</span>(selectSamples$Species))
  speciesCols&lt;-<span class="kw">rainbow.lab</span>(<span class="kw">length</span>(<span class="kw">unique</span>(selectSamples$Species)),<span class="dt">start=</span>-<span class="dv">2</span>,<span class="dt">end=</span><span class="dv">1</span>,<span class="dt">alpha=</span>.<span class="dv">8</span>,<span class="dt">lightMultiple=</span>.<span class="dv">8</span>)
  <span class="kw">names</span>(speciesCols)&lt;-<span class="kw">names</span>(speciesPch)&lt;-<span class="kw">sort</span>(<span class="kw">unique</span>(selectSamples$chimpBonobo))

  predictors&lt;-<span class="kw">model.matrix</span>(~<span class="dv">0</span>+Species+malaria+SIV+area,selectSamples)

  <span class="co">#pcoa</span>
    pos&lt;-<span class="kw">my.biplot.pcoa</span>(uniPca,predictors,<span class="dt">plot.axes=</span><span class="dv">1</span>:<span class="dv">2</span>,<span class="dt">pch=</span><span class="dv">21</span>,<span class="dt">bg=</span>speciesCols[selectSamples$chimpBonobo],<span class="dt">col=</span>malariaCols3[selectSamples$malaria<span class="dv">+1</span>],<span class="dt">cex=</span><span class="fl">2.25</span>,<span class="dt">lwd=</span><span class="dv">4</span>,<span class="dt">arrowsFilter=</span><span class="ot">Inf</span>,<span class="dt">las=</span><span class="dv">1</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="fl">2.75</span>,.<span class="dv">75</span>,<span class="dv">0</span>),<span class="dt">sameAxis=</span><span class="ot">FALSE</span>,<span class="dt">bty=</span><span class="st">&#39;l&#39;</span>,<span class="dt">type=</span><span class="st">&#39;n&#39;</span>)
    <span class="kw">points</span>(pos[!selectSamples$malaria,],<span class="dt">col=</span>malariaCols3[<span class="dv">1</span>],<span class="dt">cex=</span><span class="fl">2.25</span>,<span class="dt">lwd=</span><span class="dv">4</span>,<span class="dt">bg=</span>speciesCols[selectSamples[!selectSamples$malaria,<span class="st">&#39;chimpBonobo&#39;</span>]],<span class="dt">pch=</span><span class="dv">21</span>)
    <span class="kw">points</span>(pos[selectSamples$malaria,],<span class="dt">col=</span>malariaCols3[<span class="dv">2</span>],<span class="dt">cex=</span><span class="fl">2.25</span>,<span class="dt">lwd=</span><span class="dv">4</span>,<span class="dt">bg=</span>speciesCols[selectSamples[selectSamples$malaria,<span class="st">&#39;chimpBonobo&#39;</span>]],<span class="dt">pch=</span><span class="dv">21</span>)
    <span class="kw">title</span>(<span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&#39;%s&#39;</span>,ii,<span class="dv">1</span>,<span class="dv">2</span>))

  <span class="co">#tsne</span>
    <span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="fl">1.5</span>,<span class="dv">10</span>))
    <span class="kw">plot</span>(tsneUni$Y,<span class="dt">pch=</span>speciesPch[selectSamples$chimpBonobo],<span class="dt">bg=</span>areaCols[selectSamples$area2],<span class="dt">col=</span>malariaCols[selectSamples$malaria<span class="dv">+1</span>],<span class="dt">cex=</span><span class="fl">2.5</span>,<span class="dt">lwd=</span><span class="dv">3</span>,<span class="dt">ylab=</span><span class="st">&#39;t-SNE 2&#39;</span>,<span class="dt">xlab=</span><span class="st">&#39;t-SNE 1&#39;</span>,<span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&#39;%s&#39;</span>,ii),<span class="dt">bty=</span><span class="st">&#39;l&#39;</span>,<span class="dt">las=</span><span class="dv">1</span>)
    <span class="kw">legend</span>(
      <span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">2</span>]+.<span class="dv">01</span>*<span class="kw">diff</span>(<span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">1</span>:<span class="dv">2</span>]), 
      <span class="kw">mean</span>(<span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">3</span>:<span class="dv">4</span>]),
      <span class="kw">c</span>(<span class="kw">names</span>(malariaCols),<span class="kw">names</span>(areaCols),<span class="kw">names</span>(speciesPch)),
      <span class="dt">col=</span><span class="kw">c</span>(malariaCols,<span class="kw">rep</span>(<span class="kw">c</span>(malariaCols[<span class="dv">1</span>],mediumMalariaCol),<span class="kw">c</span>(<span class="kw">length</span>(areaCols),<span class="kw">length</span>(speciesPch)))),
      <span class="dt">pch=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">21</span>,<span class="kw">length</span>(malariaCols)),speciesPch[areaPch],speciesPch),
      <span class="dt">pt.bg=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="ot">NA</span>,<span class="kw">length</span>(malariaCols)),areaCols,<span class="kw">rep</span>(<span class="ot">NA</span>,<span class="kw">length</span>(speciesPch))),
      <span class="dt">inset=</span>.<span class="dv">01</span>,<span class="dt">pt.lwd=</span><span class="dv">3</span>,<span class="dt">pt.cex=</span><span class="fl">2.5</span>,
      <span class="dt">xjust=</span><span class="dv">0</span>,<span class="dt">xpd=</span><span class="ot">NA</span>,<span class="dt">bty=</span><span class="st">&#39;n&#39;</span>
    )

  ss&lt;-samples[<span class="kw">labels</span>(selectDist),]
  plantAdonis[[ii]]&lt;-<span class="kw">cacheOperation</span>(<span class="kw">sprintf</span>(<span class="st">&#39;work/adonis_%s.Rdat&#39;</span>,ii),adonis,selectDist~bonobo+area2+malaria,<span class="dt">data=</span>ss,<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">10</span>)
  interactAdonis[[ii]]&lt;-<span class="kw">cacheOperation</span>(<span class="kw">sprintf</span>(<span class="st">&#39;work/interactAdonis_%s.Rdat&#39;</span>,ii),adonis,selectDist~bonobo+area2+malaria*bonobo,<span class="dt">data=</span>ss,<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">10</span>)
  chimpDist&lt;-<span class="kw">as.matrix</span>(selectDist)
  chimpDist&lt;-<span class="kw">as.dist</span>(chimpDist[!ss$bonobo,!ss$bonobo])
  chimpAdonis[[ii]]&lt;-<span class="kw">cacheOperation</span>(<span class="kw">sprintf</span>(<span class="st">&#39;work/adonisChimp_%s.Rdat&#39;</span>,ii),adonis,chimpDist~area2+malaria,<span class="dt">data=</span>ss[!ss$bonobo,],<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">10</span>)
  bonoboDist&lt;-<span class="kw">as.matrix</span>(selectDist)
  bonoboDist&lt;-<span class="kw">as.dist</span>(bonoboDist[ss$bonobo,ss$bonobo])
  bonoboAdonis[[ii]]&lt;-<span class="kw">cacheOperation</span>(<span class="kw">sprintf</span>(<span class="st">&#39;work/adonisBonobo_%s.Rdat&#39;</span>,ii),adonis,bonoboDist~area2+malaria,<span class="dt">data=</span>ss[ss$bonobo,],<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">10</span>)
  tlDist&lt;-<span class="kw">as.dist</span>(<span class="kw">as.matrix</span>(selectDist)[ss$isTL,ss$isTL])
  tlAdonis[[ii]]&lt;-<span class="kw">cacheOperation</span>(<span class="kw">sprintf</span>(<span class="st">&#39;work/adonisTL_%s.Rdat&#39;</span>,ii),adonis,tlDist~malaria,<span class="dt">data=</span>ss[ss$isTL,],<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">5</span>)

}</code></pre></div>
<pre><code>## Read the 95 x 95 data matrix successfully!
## Using no_dims = 2, perplexity = 15.000000, and theta = 0.500000
## Computing input similarities...
## Building tree...
##  - point 0 of 95
## Done in 0.01 seconds (sparsity = 0.571967)!
## Learning embedding...
## Iteration 50: error is 55.965421 (50 iterations in 0.02 seconds)
## Iteration 100: error is 57.031884 (50 iterations in 0.02 seconds)
## Iteration 150: error is 54.508975 (50 iterations in 0.02 seconds)
## Iteration 200: error is 54.161525 (50 iterations in 0.02 seconds)
## Iteration 250: error is 54.119804 (50 iterations in 0.02 seconds)
## Iteration 300: error is 1.762057 (50 iterations in 0.02 seconds)
## Iteration 350: error is 1.040244 (50 iterations in 0.01 seconds)
## Iteration 400: error is 0.499174 (50 iterations in 0.02 seconds)
## Iteration 450: error is 0.378671 (50 iterations in 0.02 seconds)
## Iteration 500: error is 0.352965 (50 iterations in 0.02 seconds)
## Iteration 550: error is 0.337048 (50 iterations in 0.02 seconds)
## Iteration 600: error is 0.333652 (50 iterations in 0.02 seconds)
## Iteration 650: error is 0.331043 (50 iterations in 0.02 seconds)
## Iteration 700: error is 0.330688 (50 iterations in 0.02 seconds)
## Iteration 750: error is 0.327358 (50 iterations in 0.02 seconds)
## Iteration 800: error is 0.320485 (50 iterations in 0.02 seconds)
## Iteration 850: error is 0.314633 (50 iterations in 0.02 seconds)
## Iteration 900: error is 0.311386 (50 iterations in 0.02 seconds)
## Iteration 950: error is 0.311551 (50 iterations in 0.02 seconds)
## Iteration 1000: error is 0.313876 (50 iterations in 0.02 seconds)
## Iteration 1050: error is 0.313800 (50 iterations in 0.02 seconds)
## Iteration 1100: error is 0.312004 (50 iterations in 0.02 seconds)
## Iteration 1150: error is 0.313866 (50 iterations in 0.02 seconds)
## Iteration 1200: error is 0.302103 (50 iterations in 0.02 seconds)
## Iteration 1250: error is 0.300235 (50 iterations in 0.02 seconds)
## Iteration 1300: error is 0.301186 (50 iterations in 0.02 seconds)
## Iteration 1350: error is 0.300139 (50 iterations in 0.02 seconds)
## Iteration 1400: error is 0.301235 (50 iterations in 0.02 seconds)
## Iteration 1450: error is 0.301189 (50 iterations in 0.02 seconds)
## Iteration 1500: error is 0.300247 (50 iterations in 0.02 seconds)
## Iteration 1550: error is 0.302570 (50 iterations in 0.02 seconds)
## Iteration 1600: error is 0.301800 (50 iterations in 0.02 seconds)
## Iteration 1650: error is 0.300891 (50 iterations in 0.02 seconds)
## Iteration 1700: error is 0.302369 (50 iterations in 0.02 seconds)
## Iteration 1750: error is 0.300749 (50 iterations in 0.02 seconds)
## Iteration 1800: error is 0.300972 (50 iterations in 0.02 seconds)
## Iteration 1850: error is 0.301457 (50 iterations in 0.02 seconds)
## Iteration 1900: error is 0.301347 (50 iterations in 0.02 seconds)
## Iteration 1950: error is 0.299806 (50 iterations in 0.02 seconds)
## Iteration 2000: error is 0.300216 (50 iterations in 0.02 seconds)
## Iteration 2050: error is 0.301588 (50 iterations in 0.02 seconds)
## Iteration 2100: error is 0.300813 (50 iterations in 0.02 seconds)
## Iteration 2150: error is 0.300991 (50 iterations in 0.02 seconds)
## Iteration 2200: error is 0.300815 (50 iterations in 0.02 seconds)
## Iteration 2250: error is 0.301033 (50 iterations in 0.02 seconds)
## Iteration 2300: error is 0.301184 (50 iterations in 0.02 seconds)
## Iteration 2350: error is 0.301484 (50 iterations in 0.02 seconds)
## Iteration 2400: error is 0.301240 (50 iterations in 0.02 seconds)
## Iteration 2450: error is 0.300869 (50 iterations in 0.02 seconds)
## Iteration 2500: error is 0.301710 (50 iterations in 0.02 seconds)
## Iteration 2550: error is 0.301230 (50 iterations in 0.02 seconds)
## Iteration 2600: error is 0.301458 (50 iterations in 0.02 seconds)
## Iteration 2650: error is 0.300874 (50 iterations in 0.02 seconds)
## Iteration 2700: error is 0.300956 (50 iterations in 0.02 seconds)
## Iteration 2750: error is 0.300689 (50 iterations in 0.02 seconds)
## Iteration 2800: error is 0.301102 (50 iterations in 0.02 seconds)
## Iteration 2850: error is 0.301753 (50 iterations in 0.02 seconds)
## Iteration 2900: error is 0.300921 (50 iterations in 0.02 seconds)
## Iteration 2950: error is 0.300417 (50 iterations in 0.02 seconds)
## Iteration 3000: error is 0.301098 (50 iterations in 0.02 seconds)
## Fitting performed in 1.21 seconds.</code></pre>
<div class="figure">
<img src="figure/plantPcoaTsne-1.png" alt="plot of chunk plantPcoaTsne" />
<p class="caption">plot of chunk plantPcoaTsne</p>
</div>
<pre><code>## Cache work/adonis_matK.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/interactAdonis_matK.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/adonisChimp_matK.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/adonisBonobo_matK.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/adonisTL_matK.Rdat does not exist. Running operation</code></pre>
<div class="figure">
<img src="figure/plantPcoaTsne-2.png" alt="plot of chunk plantPcoaTsne" />
<p class="caption">plot of chunk plantPcoaTsne</p>
</div>
<pre><code>## Read the 93 x 93 data matrix successfully!
## Using no_dims = 2, perplexity = 15.000000, and theta = 0.500000
## Computing input similarities...
## Building tree...
##  - point 0 of 93
## Done in 0.00 seconds (sparsity = 0.614175)!
## Learning embedding...
## Iteration 50: error is 59.351777 (50 iterations in 0.02 seconds)
## Iteration 100: error is 58.373725 (50 iterations in 0.02 seconds)
## Iteration 150: error is 58.582589 (50 iterations in 0.02 seconds)
## Iteration 200: error is 61.438773 (50 iterations in 0.02 seconds)
## Iteration 250: error is 59.135099 (50 iterations in 0.02 seconds)
## Iteration 300: error is 2.009035 (50 iterations in 0.02 seconds)
## Iteration 350: error is 1.228906 (50 iterations in 0.01 seconds)
## Iteration 400: error is 0.810851 (50 iterations in 0.02 seconds)
## Iteration 450: error is 0.664075 (50 iterations in 0.02 seconds)
## Iteration 500: error is 0.477107 (50 iterations in 0.02 seconds)
## Iteration 550: error is 0.366709 (50 iterations in 0.02 seconds)
## Iteration 600: error is 0.343803 (50 iterations in 0.02 seconds)
## Iteration 650: error is 0.340801 (50 iterations in 0.02 seconds)
## Iteration 700: error is 0.337100 (50 iterations in 0.02 seconds)
## Iteration 750: error is 0.337315 (50 iterations in 0.02 seconds)
## Iteration 800: error is 0.335431 (50 iterations in 0.02 seconds)
## Iteration 850: error is 0.336453 (50 iterations in 0.02 seconds)
## Iteration 900: error is 0.338145 (50 iterations in 0.02 seconds)
## Iteration 950: error is 0.338037 (50 iterations in 0.02 seconds)
## Iteration 1000: error is 0.336819 (50 iterations in 0.02 seconds)
## Iteration 1050: error is 0.335821 (50 iterations in 0.02 seconds)
## Iteration 1100: error is 0.335631 (50 iterations in 0.02 seconds)
## Iteration 1150: error is 0.333037 (50 iterations in 0.02 seconds)
## Iteration 1200: error is 0.335055 (50 iterations in 0.02 seconds)
## Iteration 1250: error is 0.333108 (50 iterations in 0.02 seconds)
## Iteration 1300: error is 0.331196 (50 iterations in 0.02 seconds)
## Iteration 1350: error is 0.333172 (50 iterations in 0.02 seconds)
## Iteration 1400: error is 0.330903 (50 iterations in 0.02 seconds)
## Iteration 1450: error is 0.330390 (50 iterations in 0.02 seconds)
## Iteration 1500: error is 0.332706 (50 iterations in 0.02 seconds)
## Iteration 1550: error is 0.333011 (50 iterations in 0.02 seconds)
## Iteration 1600: error is 0.333683 (50 iterations in 0.02 seconds)
## Iteration 1650: error is 0.332345 (50 iterations in 0.02 seconds)
## Iteration 1700: error is 0.332225 (50 iterations in 0.02 seconds)
## Iteration 1750: error is 0.332751 (50 iterations in 0.02 seconds)
## Iteration 1800: error is 0.329998 (50 iterations in 0.02 seconds)
## Iteration 1850: error is 0.330948 (50 iterations in 0.02 seconds)
## Iteration 1900: error is 0.330979 (50 iterations in 0.02 seconds)
## Iteration 1950: error is 0.328637 (50 iterations in 0.02 seconds)
## Iteration 2000: error is 0.328912 (50 iterations in 0.02 seconds)
## Iteration 2050: error is 0.329687 (50 iterations in 0.02 seconds)
## Iteration 2100: error is 0.331638 (50 iterations in 0.02 seconds)
## Iteration 2150: error is 0.327180 (50 iterations in 0.02 seconds)
## Iteration 2200: error is 0.327002 (50 iterations in 0.02 seconds)
## Iteration 2250: error is 0.327801 (50 iterations in 0.02 seconds)
## Iteration 2300: error is 0.326809 (50 iterations in 0.02 seconds)
## Iteration 2350: error is 0.328673 (50 iterations in 0.02 seconds)
## Iteration 2400: error is 0.328505 (50 iterations in 0.02 seconds)
## Iteration 2450: error is 0.329269 (50 iterations in 0.02 seconds)
## Iteration 2500: error is 0.330925 (50 iterations in 0.02 seconds)
## Iteration 2550: error is 0.328937 (50 iterations in 0.02 seconds)
## Iteration 2600: error is 0.328903 (50 iterations in 0.02 seconds)
## Iteration 2650: error is 0.330823 (50 iterations in 0.02 seconds)
## Iteration 2700: error is 0.329287 (50 iterations in 0.02 seconds)
## Iteration 2750: error is 0.329791 (50 iterations in 0.02 seconds)
## Iteration 2800: error is 0.331089 (50 iterations in 0.02 seconds)
## Iteration 2850: error is 0.328553 (50 iterations in 0.02 seconds)
## Iteration 2900: error is 0.330655 (50 iterations in 0.02 seconds)
## Iteration 2950: error is 0.330349 (50 iterations in 0.02 seconds)
## Iteration 3000: error is 0.329454 (50 iterations in 0.02 seconds)
## Fitting performed in 1.12 seconds.</code></pre>
<div class="figure">
<img src="figure/plantPcoaTsne-3.png" alt="plot of chunk plantPcoaTsne" />
<p class="caption">plot of chunk plantPcoaTsne</p>
</div>
<pre><code>## Cache work/adonis_rbcL.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/interactAdonis_rbcL.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/adonisChimp_rbcL.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/adonisBonobo_rbcL.Rdat does not exist. Running operation</code></pre>
<pre><code>## Cache work/adonisTL_rbcL.Rdat does not exist. Running operation</code></pre>
<div class="figure">
<img src="figure/plantPcoaTsne-4.png" alt="plot of chunk plantPcoaTsne" />
<p class="caption">plot of chunk plantPcoaTsne</p>
</div>
<h2 id="load-libraries-3">Load libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;functions.R&#39;</span>)</code></pre></div>
<h2 id="load-data-1">Load data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;loadData.R&quot;</span>)</code></pre></div>
<pre><code>## matK</code></pre>
<pre><code>## Cache work/matK_rarefyOtus.Rdat does exist. Loading data</code></pre>
<pre><code>## rbcL</code></pre>
<pre><code>## Cache work/rbcL_rarefyOtus.Rdat does exist. Loading data</code></pre>
<h3 id="make-heatmap-of-otu-abundances">Make heatmap of OTU abundances</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sortSamples&lt;-samples[<span class="kw">order</span>(samples$Species,samples$area2,samples$malaria,samples$Code),]
cols&lt;-<span class="kw">c</span>(<span class="st">&#39;white&#39;</span>,<span class="kw">tail</span>(<span class="kw">rev</span>(<span class="kw">heat.colors</span>(<span class="dv">110</span>)),<span class="dv">99</span>)) 

minProp&lt;-.<span class="dv">02</span>
for(ii in <span class="kw">names</span>(swarmData)){
  if(ii==<span class="st">&#39;matK&#39;</span>)subLetters&lt;-<span class="kw">c</span>(<span class="st">&#39;A&#39;</span>,<span class="st">&#39;B&#39;</span>)
  else subLetters&lt;-<span class="kw">c</span>(<span class="st">&#39;C&#39;</span>,<span class="st">&#39;D&#39;</span>)
  <span class="kw">message</span>(ii)
  plotProp&lt;-swarmData[[ii]][[<span class="st">&#39;props&#39;</span>]][swarmData[[ii]][[<span class="st">&#39;isEnough&#39;</span>]],]
  plotProp&lt;-plotProp[,<span class="kw">apply</span>(plotProp,<span class="dv">2</span>,max)&gt;minProp]
  <span class="kw">colnames</span>(plotProp)&lt;-swarmData[[ii]][[<span class="st">&#39;taxa&#39;</span>]][<span class="kw">colnames</span>(plotProp),<span class="st">&#39;bestId&#39;</span>]
  plotProp&lt;-plotProp[,<span class="kw">order</span>(<span class="kw">apply</span>(plotProp,<span class="dv">2</span>,mean),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)]
  plotProp&lt;-plotProp[<span class="kw">orderIn</span>(<span class="kw">rownames</span>(plotProp),<span class="kw">rownames</span>(sortSamples)),]
  maxProp&lt;-<span class="kw">apply</span>(plotProp,<span class="dv">2</span>,function(x)x/<span class="kw">max</span>(x))
  maxTree&lt;-<span class="kw">hclust</span>(<span class="kw">dist</span>(<span class="kw">t</span>(maxProp[,])))
  maxProp&lt;-maxProp[,<span class="kw">rev</span>(maxTree$labels[maxTree$order])]
  plotProp&lt;-plotProp[,<span class="kw">rev</span>(maxTree$labels[maxTree$order])]
  breaks&lt;-<span class="kw">c</span>(-<span class="fl">1e-6</span>,<span class="kw">seq</span>(<span class="kw">min</span>(plotProp[plotProp&gt;<span class="dv">0</span>])-<span class="fl">1e-10</span>,<span class="kw">max</span>(plotProp)+<span class="fl">1e-10</span>,<span class="dt">length.out=</span><span class="dv">100</span>))
  breaks2&lt;-<span class="kw">c</span>(-<span class="fl">1e-6</span>,<span class="kw">seq</span>(<span class="kw">min</span>(maxProp[maxProp&gt;<span class="dv">0</span>])-<span class="fl">1e-10</span>,<span class="kw">max</span>(maxProp)+<span class="fl">1e-10</span>,<span class="dt">length.out=</span><span class="dv">100</span>))
  <span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>),<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">12</span>,.<span class="dv">1</span>,<span class="fl">3.5</span>,<span class="dv">15</span>))
  <span class="kw">image</span>(<span class="dv">1</span>:<span class="kw">ncol</span>(plotProp),<span class="dv">1</span>:<span class="kw">nrow</span>(plotProp),<span class="kw">t</span>(plotProp),<span class="dt">col=</span>cols,<span class="dt">breaks=</span>breaks,<span class="dt">xlab=</span><span class="st">&#39;&#39;</span>,<span class="dt">ylab=</span><span class="st">&#39;&#39;</span>,<span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">main=</span>ii)
  <span class="kw">text</span>(<span class="kw">grconvertX</span>(.<span class="dv">005</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),<span class="kw">grconvertY</span>(.<span class="dv">995</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),subLetters[<span class="dv">1</span>],<span class="dt">xpd=</span><span class="ot">NA</span>,<span class="dt">cex=</span><span class="dv">3</span>,<span class="dt">adj=</span><span class="dv">0</span>:<span class="dv">1</span>)
  <span class="kw">insetScale</span>(<span class="kw">round</span>(breaks2,<span class="dv">6</span>),cols,<span class="kw">c</span>(.<span class="dv">97</span>,.<span class="dv">58</span>,.<span class="dv">98</span>,.<span class="dv">83</span>),<span class="dt">label=</span><span class="st">&#39;Proportion of OTU within each sample&#39;</span>)
  <span class="kw">box</span>()
  <span class="kw">axis</span>(<span class="dv">1</span>,<span class="dv">1</span>:<span class="kw">ncol</span>(plotProp),<span class="kw">colnames</span>(plotProp),<span class="dt">cex.axis=</span>.<span class="dv">7</span>,<span class="dt">las=</span><span class="dv">2</span>,<span class="dt">tcl=</span>-.<span class="dv">1</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">3</span>,<span class="dv">0</span>))
  metadata&lt;-sortSamples[<span class="kw">rownames</span>(plotProp),<span class="kw">c</span>(<span class="st">&#39;chimpBonobo&#39;</span>,<span class="st">&#39;area2&#39;</span>,<span class="st">&#39;plasmoPM&#39;</span>,<span class="st">&#39;Code&#39;</span>)]
  <span class="kw">colnames</span>(metadata)&lt;-<span class="kw">c</span>(<span class="st">&#39;Species&#39;</span>,<span class="st">&#39;Site&#39;</span>,<span class="st">&#39;Laverania&#39;</span>,<span class="st">&#39;Sample&#39;</span>)
  <span class="kw">addMetaData</span>(metadata,<span class="dt">cex=</span>.<span class="dv">75</span>)
  <span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">1</span>:<span class="kw">nrow</span>(plotProp)-.<span class="dv">5</span>,<span class="dt">v=</span><span class="dv">1</span>:<span class="kw">ncol</span>(plotProp)+.<span class="dv">5</span>,<span class="dt">col=</span><span class="st">&#39;#00000011&#39;</span>)
  <span class="kw">image</span>(<span class="dv">1</span>:<span class="kw">ncol</span>(maxProp),<span class="dv">1</span>:<span class="kw">nrow</span>(maxProp),<span class="kw">t</span>(maxProp),<span class="dt">col=</span>cols,<span class="dt">breaks=</span>breaks2,<span class="dt">xlab=</span><span class="st">&#39;&#39;</span>,<span class="dt">ylab=</span><span class="st">&#39;&#39;</span>,<span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">main=</span>ii)
  <span class="kw">text</span>(<span class="kw">grconvertX</span>(.<span class="dv">005</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),<span class="kw">grconvertY</span>(.<span class="dv">995</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),subLetters[<span class="dv">2</span>],<span class="dt">xpd=</span><span class="ot">NA</span>,<span class="dt">cex=</span><span class="dv">3</span>,<span class="dt">adj=</span><span class="dv">0</span>:<span class="dv">1</span>)
  <span class="kw">box</span>()
  <span class="kw">insetScale</span>(<span class="kw">round</span>(breaks2,<span class="dv">6</span>),cols,<span class="kw">c</span>(.<span class="dv">97</span>,.<span class="dv">58</span>,.<span class="dv">98</span>,.<span class="dv">83</span>),<span class="dt">label=</span><span class="st">&#39;Proportion of OTU maximum&#39;</span>)
  <span class="kw">axis</span>(<span class="dv">1</span>,<span class="dv">1</span>:<span class="kw">ncol</span>(maxProp),<span class="kw">colnames</span>(maxProp),<span class="dt">cex.axis=</span>.<span class="dv">7</span>,<span class="dt">las=</span><span class="dv">2</span>,<span class="dt">tcl=</span>-.<span class="dv">1</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">3</span>,<span class="dv">0</span>))
  <span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">1</span>:<span class="kw">nrow</span>(maxProp)-.<span class="dv">5</span>,<span class="dt">v=</span><span class="dv">1</span>:<span class="kw">ncol</span>(maxProp)+.<span class="dv">5</span>,<span class="dt">col=</span><span class="st">&#39;#00000011&#39;</span>)
  metadata&lt;-sortSamples[<span class="kw">rownames</span>(maxProp),<span class="kw">c</span>(<span class="st">&#39;chimpBonobo&#39;</span>,<span class="st">&#39;area2&#39;</span>,<span class="st">&#39;plasmoPM&#39;</span>,<span class="st">&#39;Code&#39;</span>)]
  <span class="kw">colnames</span>(metadata)&lt;-<span class="kw">c</span>(<span class="st">&#39;Species&#39;</span>,<span class="st">&#39;Site&#39;</span>,<span class="st">&#39;Laverania&#39;</span>,<span class="st">&#39;Sample&#39;</span>)
  <span class="kw">addMetaData</span>(metadata,<span class="dt">cex=</span>.<span class="dv">75</span>)
}</code></pre></div>
<pre><code>## matK</code></pre>
<pre><code>## rbcL</code></pre>
<p><img src="figure/plantHeatmap-1.png" alt="plot of chunk plantHeatmap" /><img src="figure/plantHeatmap-2.png" alt="plot of chunk plantHeatmap" /> ## Load libraries</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># installed from https://github.com/sherrillmix/dnar </span>
<span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(parallel)
<span class="kw">packageVersion</span>(<span class="st">&#39;parallel&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;3.4.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;../functions.R&#39;</span>)</code></pre></div>
<h2 id="software-versions-2">Software versions</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;echo &quot;source activate qiime1;print_qiime_config.py -t&quot;|bash 2&gt;&amp;1&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>##  [1] &quot;&quot;                                                                                                                                                                            
##  [2] &quot;System information&quot;                                                                                                                                                          
##  [3] &quot;==================&quot;                                                                                                                                                          
##  [4] &quot;         Platform:\tlinux2&quot;                                                                                                                                                  
##  [5] &quot;   Python version:\t2.7.12 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:42:40)  [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]&quot;                                             
##  [6] &quot;Python executable:\t/home/scott/miniconda3/envs/qiime1/bin/python&quot;                                                                                                           
##  [7] &quot;&quot;                                                                                                                                                                            
##  [8] &quot;QIIME default reference information&quot;                                                                                                                                         
##  [9] &quot;===================================&quot;                                                                                                                                         
## [10] &quot;For details on what files are used as QIIME&#39;s default references, see here:&quot;                                                                                                 
## [11] &quot; https://github.com/biocore/qiime-default-reference/releases/tag/0.1.3&quot;                                                                                                      
## [12] &quot;&quot;                                                                                                                                                                            
## [13] &quot;Dependency versions&quot;                                                                                                                                                         
## [14] &quot;===================&quot;                                                                                                                                                         
## [15] &quot;          QIIME library version:\t1.9.1&quot;                                                                                                                                     
## [16] &quot;           QIIME script version:\t1.9.1&quot;                                                                                                                                     
## [17] &quot;qiime-default-reference version:\t0.1.3&quot;                                                                                                                                     
## [18] &quot;                  NumPy version:\t1.10.4&quot;                                                                                                                                    
## [19] &quot;                  SciPy version:\t0.17.1&quot;                                                                                                                                    
## [20] &quot;                 pandas version:\t0.18.1&quot;                                                                                                                                    
## [21] &quot;             matplotlib version:\t1.4.3&quot;                                                                                                                                     
## [22] &quot;            biom-format version:\t2.1.5&quot;                                                                                                                                     
## [23] &quot;                   h5py version:\t2.6.0 (HDF5 version: 1.8.16)&quot;                                                                                                              
## [24] &quot;                   qcli version:\t0.1.1&quot;                                                                                                                                     
## [25] &quot;                   pyqi version:\t0.3.2&quot;                                                                                                                                     
## [26] &quot;             scikit-bio version:\t0.2.3&quot;                                                                                                                                     
## [27] &quot;                 PyNAST version:\t1.2.2&quot;                                                                                                                                     
## [28] &quot;                Emperor version:\t0.9.51&quot;                                                                                                                                    
## [29] &quot;                burrito version:\t0.9.1&quot;                                                                                                                                     
## [30] &quot;       burrito-fillings version:\t0.1.1&quot;                                                                                                                                     
## [31] &quot;              sortmerna version:\tSortMeRNA version 2.0, 29/11/2014&quot;                                                                                                         
## [32] &quot;              sumaclust version:\tSUMACLUST Version 1.0.00&quot;                                                                                                                  
## [33] &quot;                  swarm version:\tSwarm 1.2.19 [Mar  1 2016 23:41:10]&quot;                                                                                                       
## [34] &quot;                          gdata:\tInstalled.&quot;                                                                                                                                
## [35] &quot;&quot;                                                                                                                                                                            
## [36] &quot;QIIME config values&quot;                                                                                                                                                         
## [37] &quot;===================&quot;                                                                                                                                                         
## [38] &quot;For definitions of these settings and to learn how to configure QIIME, see here:&quot;                                                                                            
## [39] &quot; http://qiime.org/install/qiime_config.html&quot;                                                                                                                                 
## [40] &quot; http://qiime.org/tutorials/parallel_qiime.html&quot;                                                                                                                             
## [41] &quot;&quot;                                                                                                                                                                            
## [42] &quot;                     blastmat_dir:\tNone&quot;                                                                                                                                    
## [43] &quot;      pick_otus_reference_seqs_fp:\t/home/scott/miniconda3/envs/qiime1/lib/python2.7/site-packages/qiime_default_reference/gg_13_8_otus/rep_set/97_otus.fasta&quot;               
## [44] &quot;                         sc_queue:\tall.q&quot;                                                                                                                                   
## [45] &quot;      topiaryexplorer_project_dir:\tNone&quot;                                                                                                                                    
## [46] &quot;     pynast_template_alignment_fp:\t/home/scott/miniconda3/envs/qiime1/lib/python2.7/site-packages/qiime_default_reference/gg_13_8_otus/rep_set_aligned/85_otus.pynast.fasta&quot;
## [47] &quot;                  cluster_jobs_fp:\tstart_parallel_jobs.py&quot;                                                                                                                  
## [48] &quot;pynast_template_alignment_blastdb:\tNone&quot;                                                                                                                                    
## [49] &quot;assign_taxonomy_reference_seqs_fp:\t/home/scott/miniconda3/envs/qiime1/lib/python2.7/site-packages/qiime_default_reference/gg_13_8_otus/rep_set/97_otus.fasta&quot;               
## [50] &quot;                     torque_queue:\tfriendlyq&quot;                                                                                                                               
## [51] &quot;                    jobs_to_start:\t1&quot;                                                                                                                                       
## [52] &quot;                       slurm_time:\tNone&quot;                                                                                                                                    
## [53] &quot;            denoiser_min_per_core:\t50&quot;                                                                                                                                      
## [54] &quot;assign_taxonomy_id_to_taxonomy_fp:\t/home/scott/miniconda3/envs/qiime1/lib/python2.7/site-packages/qiime_default_reference/gg_13_8_otus/taxonomy/97_otu_taxonomy.txt&quot;        
## [55] &quot;                         temp_dir:\t/tmp/&quot;                                                                                                                                   
## [56] &quot;                     slurm_memory:\tNone&quot;                                                                                                                                    
## [57] &quot;                      slurm_queue:\tNone&quot;                                                                                                                                    
## [58] &quot;                      blastall_fp:\tblastall&quot;                                                                                                                                
## [59] &quot;                 seconds_to_sleep:\t1&quot;                                                                                                                                       
## [60] &quot;&quot;                                                                                                                                                                            
## [61] &quot;QIIME base install test results&quot;                                                                                                                                             
## [62] &quot;===============================&quot;                                                                                                                                             
## [63] &quot;.........&quot;                                                                                                                                                                   
## [64] &quot;----------------------------------------------------------------------&quot;                                                                                                      
## [65] &quot;Ran 9 tests in 0.017s&quot;                                                                                                                                                       
## [66] &quot;&quot;                                                                                                                                                                            
## [67] &quot;OK&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">system</span>(<span class="st">&#39;bbmerge.sh --version 2&gt;&amp;1&#39;</span>,<span class="dt">intern=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## [1] &quot;java -Djava.library.path=/home/scott/installs/bbmap/jni/ -ea -Xmx1000m -Xmx1000m -cp /home/scott/installs/bbmap/current/ jgi.BBMerge --version&quot;
## [2] &quot;BBMap version 36.77&quot;                                                                                                                           
## [3] &quot;For help, please run the shellscript with no parameters, or look in /docs/.&quot;</code></pre>
<h2 id="merge-reads">Merge reads</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  if(!<span class="kw">dir.exists</span>(<span class="st">&#39;data/joined&#39;</span>))<span class="kw">dir.create</span>(<span class="st">&#39;data/joined&#39;</span>)
  fastq1s&lt;-<span class="kw">list.files</span>(<span class="st">&#39;data&#39;</span>,<span class="st">&#39;_R1_.*</span><span class="ch">\\</span><span class="st">.fastq.gz&#39;</span>,<span class="dt">full.name=</span><span class="ot">TRUE</span>)
  <span class="kw">mclapply</span>(fastq1s,function(ii){
    out&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;data/joined/%s.fastq&#39;</span>,<span class="kw">sub</span>(<span class="st">&#39;_R1_.*$&#39;</span>,<span class="st">&#39;&#39;</span>,<span class="kw">basename</span>(ii)))
    cmd&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;bbmerge.sh in1=%s in2=%s out=%s t=10 2&gt;%s&#39;</span>,ii,<span class="kw">sub</span>(<span class="st">&#39;_R1_&#39;</span>,<span class="st">&#39;_R2_&#39;</span>,ii),out,<span class="kw">sub</span>(<span class="st">&#39;fastq$&#39;</span>,<span class="st">&#39;out&#39;</span>,out))
    exit&lt;-<span class="kw">system</span>(cmd)
    if(exit!=<span class="dv">0</span>)<span class="kw">stop</span>(<span class="st">&quot;Problem running pairing in&quot;</span>,ii)
    <span class="kw">system</span>(<span class="kw">sprintf</span>(<span class="st">&#39;gzip %s&#39;</span>,out))
    <span class="kw">return</span>(cmd)
  },<span class="dt">mc.cores=</span><span class="dv">10</span>)</code></pre></div>
<pre><code>## [[1]]
## [1] &quot;bbmerge.sh in1=data/BI0054_16s_R1_.fastq.gz in2=data/BI0054_16s_R2_.fastq.gz out=data/joined/BI0054_16s.fastq t=10 2&gt;data/joined/BI0054_16s.out&quot;
## 
## [[2]]
## [1] &quot;bbmerge.sh in1=data/BI0055_16s_R1_.fastq.gz in2=data/BI0055_16s_R2_.fastq.gz out=data/joined/BI0055_16s.fastq t=10 2&gt;data/joined/BI0055_16s.out&quot;
## 
## [[3]]
## [1] &quot;bbmerge.sh in1=data/BI0093_16s_R1_.fastq.gz in2=data/BI0093_16s_R2_.fastq.gz out=data/joined/BI0093_16s.fastq t=10 2&gt;data/joined/BI0093_16s.out&quot;
## 
## [[4]]
## [1] &quot;bbmerge.sh in1=data/BI0097_16s_R1_.fastq.gz in2=data/BI0097_16s_R2_.fastq.gz out=data/joined/BI0097_16s.fastq t=10 2&gt;data/joined/BI0097_16s.out&quot;
## 
## [[5]]
## [1] &quot;bbmerge.sh in1=data/BI0246_16s_R1_.fastq.gz in2=data/BI0246_16s_R2_.fastq.gz out=data/joined/BI0246_16s.fastq t=10 2&gt;data/joined/BI0246_16s.out&quot;
## 
## [[6]]
## [1] &quot;bbmerge.sh in1=data/BI0248_16s_R1_.fastq.gz in2=data/BI0248_16s_R2_.fastq.gz out=data/joined/BI0248_16s.fastq t=10 2&gt;data/joined/BI0248_16s.out&quot;
## 
## [[7]]
## [1] &quot;bbmerge.sh in1=data/BI0257_16s_R1_.fastq.gz in2=data/BI0257_16s_R2_.fastq.gz out=data/joined/BI0257_16s.fastq t=10 2&gt;data/joined/BI0257_16s.out&quot;
## 
## [[8]]
## [1] &quot;bbmerge.sh in1=data/BI0260_16s_R1_.fastq.gz in2=data/BI0260_16s_R2_.fastq.gz out=data/joined/BI0260_16s.fastq t=10 2&gt;data/joined/BI0260_16s.out&quot;
## 
## [[9]]
## [1] &quot;bbmerge.sh in1=data/BI2414_16s_R1_.fastq.gz in2=data/BI2414_16s_R2_.fastq.gz out=data/joined/BI2414_16s.fastq t=10 2&gt;data/joined/BI2414_16s.out&quot;
## 
## [[10]]
## [1] &quot;bbmerge.sh in1=data/BI2415_16s_R1_.fastq.gz in2=data/BI2415_16s_R2_.fastq.gz out=data/joined/BI2415_16s.fastq t=10 2&gt;data/joined/BI2415_16s.out&quot;
## 
## [[11]]
## [1] &quot;bbmerge.sh in1=data/IK3158_16s_R1_.fastq.gz in2=data/IK3158_16s_R2_.fastq.gz out=data/joined/IK3158_16s.fastq t=10 2&gt;data/joined/IK3158_16s.out&quot;
## 
## [[12]]
## [1] &quot;bbmerge.sh in1=data/IK3276_16s_R1_.fastq.gz in2=data/IK3276_16s_R2_.fastq.gz out=data/joined/IK3276_16s.fastq t=10 2&gt;data/joined/IK3276_16s.out&quot;
## 
## [[13]]
## [1] &quot;bbmerge.sh in1=data/IK3358_16s_R1_.fastq.gz in2=data/IK3358_16s_R2_.fastq.gz out=data/joined/IK3358_16s.fastq t=10 2&gt;data/joined/IK3358_16s.out&quot;
## 
## [[14]]
## [1] &quot;bbmerge.sh in1=data/IK3469_16s_R1_.fastq.gz in2=data/IK3469_16s_R2_.fastq.gz out=data/joined/IK3469_16s.fastq t=10 2&gt;data/joined/IK3469_16s.out&quot;
## 
## [[15]]
## [1] &quot;bbmerge.sh in1=data/IK3513_16s_R1_.fastq.gz in2=data/IK3513_16s_R2_.fastq.gz out=data/joined/IK3513_16s.fastq t=10 2&gt;data/joined/IK3513_16s.out&quot;
## 
## [[16]]
## [1] &quot;bbmerge.sh in1=data/IK3650_16s_R1_.fastq.gz in2=data/IK3650_16s_R2_.fastq.gz out=data/joined/IK3650_16s.fastq t=10 2&gt;data/joined/IK3650_16s.out&quot;
## 
## [[17]]
## [1] &quot;bbmerge.sh in1=data/IK3701_16s_R1_.fastq.gz in2=data/IK3701_16s_R2_.fastq.gz out=data/joined/IK3701_16s.fastq t=10 2&gt;data/joined/IK3701_16s.out&quot;
## 
## [[18]]
## [1] &quot;bbmerge.sh in1=data/IK3777_16s_R1_.fastq.gz in2=data/IK3777_16s_R2_.fastq.gz out=data/joined/IK3777_16s.fastq t=10 2&gt;data/joined/IK3777_16s.out&quot;
## 
## [[19]]
## [1] &quot;bbmerge.sh in1=data/IK4184_16s_R1_.fastq.gz in2=data/IK4184_16s_R2_.fastq.gz out=data/joined/IK4184_16s.fastq t=10 2&gt;data/joined/IK4184_16s.out&quot;
## 
## [[20]]
## [1] &quot;bbmerge.sh in1=data/IK4214_16s_R1_.fastq.gz in2=data/IK4214_16s_R2_.fastq.gz out=data/joined/IK4214_16s.fastq t=10 2&gt;data/joined/IK4214_16s.out&quot;
## 
## [[21]]
## [1] &quot;bbmerge.sh in1=data/KR02_16s_R1_.fastq.gz in2=data/KR02_16s_R2_.fastq.gz out=data/joined/KR02_16s.fastq t=10 2&gt;data/joined/KR02_16s.out&quot;
## 
## [[22]]
## [1] &quot;bbmerge.sh in1=data/KR05_16s_R1_.fastq.gz in2=data/KR05_16s_R2_.fastq.gz out=data/joined/KR05_16s.fastq t=10 2&gt;data/joined/KR05_16s.out&quot;
## 
## [[23]]
## [1] &quot;bbmerge.sh in1=data/KR07_16s_R1_.fastq.gz in2=data/KR07_16s_R2_.fastq.gz out=data/joined/KR07_16s.fastq t=10 2&gt;data/joined/KR07_16s.out&quot;
## 
## [[24]]
## [1] &quot;bbmerge.sh in1=data/KR10_16s_R1_.fastq.gz in2=data/KR10_16s_R2_.fastq.gz out=data/joined/KR10_16s.fastq t=10 2&gt;data/joined/KR10_16s.out&quot;
## 
## [[25]]
## [1] &quot;bbmerge.sh in1=data/KR12_16s_R1_.fastq.gz in2=data/KR12_16s_R2_.fastq.gz out=data/joined/KR12_16s.fastq t=10 2&gt;data/joined/KR12_16s.out&quot;
## 
## [[26]]
## [1] &quot;bbmerge.sh in1=data/KR21_16s_R1_.fastq.gz in2=data/KR21_16s_R2_.fastq.gz out=data/joined/KR21_16s.fastq t=10 2&gt;data/joined/KR21_16s.out&quot;
## 
## [[27]]
## [1] &quot;bbmerge.sh in1=data/KR33_16s_R1_.fastq.gz in2=data/KR33_16s_R2_.fastq.gz out=data/joined/KR33_16s.fastq t=10 2&gt;data/joined/KR33_16s.out&quot;
## 
## [[28]]
## [1] &quot;bbmerge.sh in1=data/KR35_16s_R1_.fastq.gz in2=data/KR35_16s_R2_.fastq.gz out=data/joined/KR35_16s.fastq t=10 2&gt;data/joined/KR35_16s.out&quot;
## 
## [[29]]
## [1] &quot;bbmerge.sh in1=data/KR52_16s_R1_.fastq.gz in2=data/KR52_16s_R2_.fastq.gz out=data/joined/KR52_16s.fastq t=10 2&gt;data/joined/KR52_16s.out&quot;
## 
## [[30]]
## [1] &quot;bbmerge.sh in1=data/KR57_16s_R1_.fastq.gz in2=data/KR57_16s_R2_.fastq.gz out=data/joined/KR57_16s.fastq t=10 2&gt;data/joined/KR57_16s.out&quot;
## 
## [[31]]
## [1] &quot;bbmerge.sh in1=data/KR67_16s_R1_.fastq.gz in2=data/KR67_16s_R2_.fastq.gz out=data/joined/KR67_16s.fastq t=10 2&gt;data/joined/KR67_16s.out&quot;
## 
## [[32]]
## [1] &quot;bbmerge.sh in1=data/LG4300_16s_R1_.fastq.gz in2=data/LG4300_16s_R2_.fastq.gz out=data/joined/LG4300_16s.fastq t=10 2&gt;data/joined/LG4300_16s.out&quot;
## 
## [[33]]
## [1] &quot;bbmerge.sh in1=data/LG4314_16s_R1_.fastq.gz in2=data/LG4314_16s_R2_.fastq.gz out=data/joined/LG4314_16s.fastq t=10 2&gt;data/joined/LG4314_16s.out&quot;
## 
## [[34]]
## [1] &quot;bbmerge.sh in1=data/LG4322_16s_R1_.fastq.gz in2=data/LG4322_16s_R2_.fastq.gz out=data/joined/LG4322_16s.fastq t=10 2&gt;data/joined/LG4322_16s.out&quot;
## 
## [[35]]
## [1] &quot;bbmerge.sh in1=data/LG4327_16s_R1_.fastq.gz in2=data/LG4327_16s_R2_.fastq.gz out=data/joined/LG4327_16s.fastq t=10 2&gt;data/joined/LG4327_16s.out&quot;
## 
## [[36]]
## [1] &quot;bbmerge.sh in1=data/LK645_16s_R1_.fastq.gz in2=data/LK645_16s_R2_.fastq.gz out=data/joined/LK645_16s.fastq t=10 2&gt;data/joined/LK645_16s.out&quot;
## 
## [[37]]
## [1] &quot;bbmerge.sh in1=data/LK647_16s_R1_.fastq.gz in2=data/LK647_16s_R2_.fastq.gz out=data/joined/LK647_16s.fastq t=10 2&gt;data/joined/LK647_16s.out&quot;
## 
## [[38]]
## [1] &quot;bbmerge.sh in1=data/LK653_16s_R1_.fastq.gz in2=data/LK653_16s_R2_.fastq.gz out=data/joined/LK653_16s.fastq t=10 2&gt;data/joined/LK653_16s.out&quot;
## 
## [[39]]
## [1] &quot;bbmerge.sh in1=data/LK661_16s_R1_.fastq.gz in2=data/LK661_16s_R2_.fastq.gz out=data/joined/LK661_16s.fastq t=10 2&gt;data/joined/LK661_16s.out&quot;
## 
## [[40]]
## [1] &quot;bbmerge.sh in1=data/LK665_16s_R1_.fastq.gz in2=data/LK665_16s_R2_.fastq.gz out=data/joined/LK665_16s.fastq t=10 2&gt;data/joined/LK665_16s.out&quot;
## 
## [[41]]
## [1] &quot;bbmerge.sh in1=data/LK668_16s_R1_.fastq.gz in2=data/LK668_16s_R2_.fastq.gz out=data/joined/LK668_16s.fastq t=10 2&gt;data/joined/LK668_16s.out&quot;
## 
## [[42]]
## [1] &quot;bbmerge.sh in1=data/LK670_16s_R1_.fastq.gz in2=data/LK670_16s_R2_.fastq.gz out=data/joined/LK670_16s.fastq t=10 2&gt;data/joined/LK670_16s.out&quot;
## 
## [[43]]
## [1] &quot;bbmerge.sh in1=data/LK682_16s_R1_.fastq.gz in2=data/LK682_16s_R2_.fastq.gz out=data/joined/LK682_16s.fastq t=10 2&gt;data/joined/LK682_16s.out&quot;
## 
## [[44]]
## [1] &quot;bbmerge.sh in1=data/LK685_16s_R1_.fastq.gz in2=data/LK685_16s_R2_.fastq.gz out=data/joined/LK685_16s.fastq t=10 2&gt;data/joined/LK685_16s.out&quot;
## 
## [[45]]
## [1] &quot;bbmerge.sh in1=data/LK686_16s_R1_.fastq.gz in2=data/LK686_16s_R2_.fastq.gz out=data/joined/LK686_16s.fastq t=10 2&gt;data/joined/LK686_16s.out&quot;
## 
## [[46]]
## [1] &quot;bbmerge.sh in1=data/PA0367_16s_R1_.fastq.gz in2=data/PA0367_16s_R2_.fastq.gz out=data/joined/PA0367_16s.fastq t=10 2&gt;data/joined/PA0367_16s.out&quot;
## 
## [[47]]
## [1] &quot;bbmerge.sh in1=data/PA0368_16s_R1_.fastq.gz in2=data/PA0368_16s_R2_.fastq.gz out=data/joined/PA0368_16s.fastq t=10 2&gt;data/joined/PA0368_16s.out&quot;
## 
## [[48]]
## [1] &quot;bbmerge.sh in1=data/PA0370_16s_R1_.fastq.gz in2=data/PA0370_16s_R2_.fastq.gz out=data/joined/PA0370_16s.fastq t=10 2&gt;data/joined/PA0370_16s.out&quot;
## 
## [[49]]
## [1] &quot;bbmerge.sh in1=data/PA0456_16s_R1_.fastq.gz in2=data/PA0456_16s_R2_.fastq.gz out=data/joined/PA0456_16s.fastq t=10 2&gt;data/joined/PA0456_16s.out&quot;
## 
## [[50]]
## [1] &quot;bbmerge.sh in1=data/PA1038_16s_R1_.fastq.gz in2=data/PA1038_16s_R2_.fastq.gz out=data/joined/PA1038_16s.fastq t=10 2&gt;data/joined/PA1038_16s.out&quot;
## 
## [[51]]
## [1] &quot;bbmerge.sh in1=data/PA1039_16s_R1_.fastq.gz in2=data/PA1039_16s_R2_.fastq.gz out=data/joined/PA1039_16s.fastq t=10 2&gt;data/joined/PA1039_16s.out&quot;
## 
## [[52]]
## [1] &quot;bbmerge.sh in1=data/PA1044_16s_R1_.fastq.gz in2=data/PA1044_16s_R2_.fastq.gz out=data/joined/PA1044_16s.fastq t=10 2&gt;data/joined/PA1044_16s.out&quot;
## 
## [[53]]
## [1] &quot;bbmerge.sh in1=data/PA1049_16s_R1_.fastq.gz in2=data/PA1049_16s_R2_.fastq.gz out=data/joined/PA1049_16s.fastq t=10 2&gt;data/joined/PA1049_16s.out&quot;
## 
## [[54]]
## [1] &quot;bbmerge.sh in1=data/PA1059_16s_R1_.fastq.gz in2=data/PA1059_16s_R2_.fastq.gz out=data/joined/PA1059_16s.fastq t=10 2&gt;data/joined/PA1059_16s.out&quot;
## 
## [[55]]
## [1] &quot;bbmerge.sh in1=data/PA1065_16s_R1_.fastq.gz in2=data/PA1065_16s_R2_.fastq.gz out=data/joined/PA1065_16s.fastq t=10 2&gt;data/joined/PA1065_16s.out&quot;
## 
## [[56]]
## [1] &quot;bbmerge.sh in1=data/TL3793_16s_R1_.fastq.gz in2=data/TL3793_16s_R2_.fastq.gz out=data/joined/TL3793_16s.fastq t=10 2&gt;data/joined/TL3793_16s.out&quot;
## 
## [[57]]
## [1] &quot;bbmerge.sh in1=data/TL3797_16s_R1_.fastq.gz in2=data/TL3797_16s_R2_.fastq.gz out=data/joined/TL3797_16s.fastq t=10 2&gt;data/joined/TL3797_16s.out&quot;
## 
## [[58]]
## [1] &quot;bbmerge.sh in1=data/TL3814_16s_R1_.fastq.gz in2=data/TL3814_16s_R2_.fastq.gz out=data/joined/TL3814_16s.fastq t=10 2&gt;data/joined/TL3814_16s.out&quot;
## 
## [[59]]
## [1] &quot;bbmerge.sh in1=data/TL3816_16s_R1_.fastq.gz in2=data/TL3816_16s_R2_.fastq.gz out=data/joined/TL3816_16s.fastq t=10 2&gt;data/joined/TL3816_16s.out&quot;
## 
## [[60]]
## [1] &quot;bbmerge.sh in1=data/TL3820_16s_R1_.fastq.gz in2=data/TL3820_16s_R2_.fastq.gz out=data/joined/TL3820_16s.fastq t=10 2&gt;data/joined/TL3820_16s.out&quot;
## 
## [[61]]
## [1] &quot;bbmerge.sh in1=data/TL3821_16s_R1_.fastq.gz in2=data/TL3821_16s_R2_.fastq.gz out=data/joined/TL3821_16s.fastq t=10 2&gt;data/joined/TL3821_16s.out&quot;
## 
## [[62]]
## [1] &quot;bbmerge.sh in1=data/TL3824_16s_R1_.fastq.gz in2=data/TL3824_16s_R2_.fastq.gz out=data/joined/TL3824_16s.fastq t=10 2&gt;data/joined/TL3824_16s.out&quot;
## 
## [[63]]
## [1] &quot;bbmerge.sh in1=data/TL3826_16s_R1_.fastq.gz in2=data/TL3826_16s_R2_.fastq.gz out=data/joined/TL3826_16s.fastq t=10 2&gt;data/joined/TL3826_16s.out&quot;
## 
## [[64]]
## [1] &quot;bbmerge.sh in1=data/TL3838_16s_R1_.fastq.gz in2=data/TL3838_16s_R2_.fastq.gz out=data/joined/TL3838_16s.fastq t=10 2&gt;data/joined/TL3838_16s.out&quot;
## 
## [[65]]
## [1] &quot;bbmerge.sh in1=data/TL3842_16s_R1_.fastq.gz in2=data/TL3842_16s_R2_.fastq.gz out=data/joined/TL3842_16s.fastq t=10 2&gt;data/joined/TL3842_16s.out&quot;
## 
## [[66]]
## [1] &quot;bbmerge.sh in1=data/TL3856_16s_R1_.fastq.gz in2=data/TL3856_16s_R2_.fastq.gz out=data/joined/TL3856_16s.fastq t=10 2&gt;data/joined/TL3856_16s.out&quot;
## 
## [[67]]
## [1] &quot;bbmerge.sh in1=data/TL3862_16s_R1_.fastq.gz in2=data/TL3862_16s_R2_.fastq.gz out=data/joined/TL3862_16s.fastq t=10 2&gt;data/joined/TL3862_16s.out&quot;
## 
## [[68]]
## [1] &quot;bbmerge.sh in1=data/TL3882_16s_R1_.fastq.gz in2=data/TL3882_16s_R2_.fastq.gz out=data/joined/TL3882_16s.fastq t=10 2&gt;data/joined/TL3882_16s.out&quot;
## 
## [[69]]
## [1] &quot;bbmerge.sh in1=data/TL3889_16s_R1_.fastq.gz in2=data/TL3889_16s_R2_.fastq.gz out=data/joined/TL3889_16s.fastq t=10 2&gt;data/joined/TL3889_16s.out&quot;
## 
## [[70]]
## [1] &quot;bbmerge.sh in1=data/TL3905_16s_R1_.fastq.gz in2=data/TL3905_16s_R2_.fastq.gz out=data/joined/TL3905_16s.fastq t=10 2&gt;data/joined/TL3905_16s.out&quot;
## 
## [[71]]
## [1] &quot;bbmerge.sh in1=data/TL3910_16s_R1_.fastq.gz in2=data/TL3910_16s_R2_.fastq.gz out=data/joined/TL3910_16s.fastq t=10 2&gt;data/joined/TL3910_16s.out&quot;
## 
## [[72]]
## [1] &quot;bbmerge.sh in1=data/TL3911_16s_R1_.fastq.gz in2=data/TL3911_16s_R2_.fastq.gz out=data/joined/TL3911_16s.fastq t=10 2&gt;data/joined/TL3911_16s.out&quot;
## 
## [[73]]
## [1] &quot;bbmerge.sh in1=data/TL3915_16s_R1_.fastq.gz in2=data/TL3915_16s_R2_.fastq.gz out=data/joined/TL3915_16s.fastq t=10 2&gt;data/joined/TL3915_16s.out&quot;
## 
## [[74]]
## [1] &quot;bbmerge.sh in1=data/TL3916_16s_R1_.fastq.gz in2=data/TL3916_16s_R2_.fastq.gz out=data/joined/TL3916_16s.fastq t=10 2&gt;data/joined/TL3916_16s.out&quot;
## 
## [[75]]
## [1] &quot;bbmerge.sh in1=data/TL3918_16s_R1_.fastq.gz in2=data/TL3918_16s_R2_.fastq.gz out=data/joined/TL3918_16s.fastq t=10 2&gt;data/joined/TL3918_16s.out&quot;
## 
## [[76]]
## [1] &quot;bbmerge.sh in1=data/TL3925_16s_R1_.fastq.gz in2=data/TL3925_16s_R2_.fastq.gz out=data/joined/TL3925_16s.fastq t=10 2&gt;data/joined/TL3925_16s.out&quot;
## 
## [[77]]
## [1] &quot;bbmerge.sh in1=data/TL3926_16s_R1_.fastq.gz in2=data/TL3926_16s_R2_.fastq.gz out=data/joined/TL3926_16s.fastq t=10 2&gt;data/joined/TL3926_16s.out&quot;
## 
## [[78]]
## [1] &quot;bbmerge.sh in1=data/TL3927_16s_R1_.fastq.gz in2=data/TL3927_16s_R2_.fastq.gz out=data/joined/TL3927_16s.fastq t=10 2&gt;data/joined/TL3927_16s.out&quot;
## 
## [[79]]
## [1] &quot;bbmerge.sh in1=data/TL3929_16s_R1_.fastq.gz in2=data/TL3929_16s_R2_.fastq.gz out=data/joined/TL3929_16s.fastq t=10 2&gt;data/joined/TL3929_16s.out&quot;
## 
## [[80]]
## [1] &quot;bbmerge.sh in1=data/TL3932_16s_R1_.fastq.gz in2=data/TL3932_16s_R2_.fastq.gz out=data/joined/TL3932_16s.fastq t=10 2&gt;data/joined/TL3932_16s.out&quot;
## 
## [[81]]
## [1] &quot;bbmerge.sh in1=data/TL3936_16s_R1_.fastq.gz in2=data/TL3936_16s_R2_.fastq.gz out=data/joined/TL3936_16s.fastq t=10 2&gt;data/joined/TL3936_16s.out&quot;
## 
## [[82]]
## [1] &quot;bbmerge.sh in1=data/TL3939_16s_R1_.fastq.gz in2=data/TL3939_16s_R2_.fastq.gz out=data/joined/TL3939_16s.fastq t=10 2&gt;data/joined/TL3939_16s.out&quot;
## 
## [[83]]
## [1] &quot;bbmerge.sh in1=data/TL3940_16s_R1_.fastq.gz in2=data/TL3940_16s_R2_.fastq.gz out=data/joined/TL3940_16s.fastq t=10 2&gt;data/joined/TL3940_16s.out&quot;
## 
## [[84]]
## [1] &quot;bbmerge.sh in1=data/TL3942_16s_R1_.fastq.gz in2=data/TL3942_16s_R2_.fastq.gz out=data/joined/TL3942_16s.fastq t=10 2&gt;data/joined/TL3942_16s.out&quot;
## 
## [[85]]
## [1] &quot;bbmerge.sh in1=data/TL3943_16s_R1_.fastq.gz in2=data/TL3943_16s_R2_.fastq.gz out=data/joined/TL3943_16s.fastq t=10 2&gt;data/joined/TL3943_16s.out&quot;
## 
## [[86]]
## [1] &quot;bbmerge.sh in1=data/TL3944_16s_R1_.fastq.gz in2=data/TL3944_16s_R2_.fastq.gz out=data/joined/TL3944_16s.fastq t=10 2&gt;data/joined/TL3944_16s.out&quot;
## 
## [[87]]
## [1] &quot;bbmerge.sh in1=data/TL3945_16s_R1_.fastq.gz in2=data/TL3945_16s_R2_.fastq.gz out=data/joined/TL3945_16s.fastq t=10 2&gt;data/joined/TL3945_16s.out&quot;
## 
## [[88]]
## [1] &quot;bbmerge.sh in1=data/TL3946_16s_R1_.fastq.gz in2=data/TL3946_16s_R2_.fastq.gz out=data/joined/TL3946_16s.fastq t=10 2&gt;data/joined/TL3946_16s.out&quot;
## 
## [[89]]
## [1] &quot;bbmerge.sh in1=data/TL3948_16s_R1_.fastq.gz in2=data/TL3948_16s_R2_.fastq.gz out=data/joined/TL3948_16s.fastq t=10 2&gt;data/joined/TL3948_16s.out&quot;
## 
## [[90]]
## [1] &quot;bbmerge.sh in1=data/UB0439_16s_R1_.fastq.gz in2=data/UB0439_16s_R2_.fastq.gz out=data/joined/UB0439_16s.fastq t=10 2&gt;data/joined/UB0439_16s.out&quot;
## 
## [[91]]
## [1] &quot;bbmerge.sh in1=data/UB0445_16s_R1_.fastq.gz in2=data/UB0445_16s_R2_.fastq.gz out=data/joined/UB0445_16s.fastq t=10 2&gt;data/joined/UB0445_16s.out&quot;
## 
## [[92]]
## [1] &quot;bbmerge.sh in1=data/UB0599_16s_R1_.fastq.gz in2=data/UB0599_16s_R2_.fastq.gz out=data/joined/UB0599_16s.fastq t=10 2&gt;data/joined/UB0599_16s.out&quot;
## 
## [[93]]
## [1] &quot;bbmerge.sh in1=data/UB1430_16s_R1_.fastq.gz in2=data/UB1430_16s_R2_.fastq.gz out=data/joined/UB1430_16s.fastq t=10 2&gt;data/joined/UB1430_16s.out&quot;
## 
## [[94]]
## [1] &quot;bbmerge.sh in1=data/UB1435_16s_R1_.fastq.gz in2=data/UB1435_16s_R2_.fastq.gz out=data/joined/UB1435_16s.fastq t=10 2&gt;data/joined/UB1435_16s.out&quot;
## 
## [[95]]
## [1] &quot;bbmerge.sh in1=data/UB1446_16s_R1_.fastq.gz in2=data/UB1446_16s_R2_.fastq.gz out=data/joined/UB1446_16s.fastq t=10 2&gt;data/joined/UB1446_16s.out&quot;
## 
## [[96]]
## [1] &quot;bbmerge.sh in1=data/UB1452_16s_R1_.fastq.gz in2=data/UB1452_16s_R2_.fastq.gz out=data/joined/UB1452_16s.fastq t=10 2&gt;data/joined/UB1452_16s.out&quot;
## 
## [[97]]
## [1] &quot;bbmerge.sh in1=data/UB1454_16s_R1_.fastq.gz in2=data/UB1454_16s_R2_.fastq.gz out=data/joined/UB1454_16s.fastq t=10 2&gt;data/joined/UB1454_16s.out&quot;
## 
## [[98]]
## [1] &quot;bbmerge.sh in1=data/UB2037_16s_R1_.fastq.gz in2=data/UB2037_16s_R2_.fastq.gz out=data/joined/UB2037_16s.fastq t=10 2&gt;data/joined/UB2037_16s.out&quot;</code></pre>
<h2 id="read-in-all-sequences">Read in all sequences</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fastqs&lt;-<span class="kw">list.files</span>(<span class="st">&#39;data/joined&#39;</span>,<span class="st">&#39;.fastq.gz&#39;</span>,<span class="dt">full.name=</span><span class="ot">TRUE</span>)
<span class="co">#just get sequences to reduce memory</span>
allSeq&lt;-<span class="kw">mclapply</span>(fastqs,function(xx)<span class="kw">read.fastq</span>(xx)$seq,<span class="dt">mc.cores=</span><span class="dv">12</span>)
<span class="kw">message</span>(<span class="st">&#39;Read &#39;</span>,<span class="kw">length</span>(<span class="kw">unlist</span>(allSeq)),<span class="st">&#39; sequences in &#39;</span>,<span class="kw">length</span>(allSeq),<span class="st">&#39; fastqs&#39;</span>)</code></pre></div>
<pre><code>## Read 6574907 sequences in 98 fastqs</code></pre>
<h2 id="run-qiime">Run qiime</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">if(!<span class="kw">dir.exists</span>(<span class="st">&#39;work/&#39;</span>))<span class="kw">dir.create</span>(<span class="st">&#39;work&#39;</span>)
out&lt;-<span class="kw">runQiime</span>(<span class="kw">unlist</span>(allSeq),<span class="dt">storeDir=</span><span class="st">&#39;work/qiime&#39;</span>)</code></pre></div>
<pre><code>## echo &quot;source activate qiime1; pick_de_novo_otus.py --input /tmp/RtmpTfT7gW/filea38447171fb9/XXX.fa --output /tmp/RtmpTfT7gW/filea384cfe426 --parallel --jobs_to_start 32 --force&quot;|bash</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">outDf&lt;-<span class="kw">data.frame</span>(
  <span class="st">&#39;file&#39;</span>=<span class="kw">rep</span>(<span class="kw">sub</span>(<span class="st">&#39;_16s.fastq.gz&#39;</span>,<span class="st">&#39;&#39;</span>,<span class="kw">basename</span>(fastqs)),<span class="kw">sapply</span>(allSeq,length)),
  <span class="st">&#39;otu&#39;</span>=out[[<span class="st">&#39;otus&#39;</span>]],
  <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>
)
<span class="kw">withAs</span>(<span class="dt">outFile=</span><span class="kw">gzfile</span>(<span class="st">&#39;work/qiimeOtuIds.csv.gz&#39;</span>),<span class="kw">write.csv</span>(outDf,outFile,<span class="dt">row.names=</span><span class="ot">FALSE</span>))
<span class="kw">write.fa</span>(<span class="kw">names</span>(out[[<span class="st">&#39;seqs&#39;</span>]]),out[[<span class="st">&#39;seqs&#39;</span>]],<span class="st">&#39;work/qiimeOtus.fa.gz&#39;</span>)
outTaxa&lt;-<span class="kw">data.frame</span>(
  <span class="st">&#39;name&#39;</span>=<span class="kw">names</span>(out[[<span class="st">&#39;taxa&#39;</span>]]),
  <span class="st">&#39;taxa&#39;</span>=out[[<span class="st">&#39;taxa&#39;</span>]],
  <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>
)
<span class="kw">write.csv</span>(outTaxa,<span class="st">&#39;work/qiimeOtus.taxa&#39;</span>,<span class="dt">row.names=</span><span class="ot">FALSE</span>)</code></pre></div>
<h2 id="load-libraries-4">Load libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vipor)
<span class="kw">packageVersion</span>(<span class="st">&#39;vipor&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.4.5&#39;</code></pre>
<h2 id="load-the-data">Load the data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;readData.R&#39;</span>)</code></pre></div>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dnar&#39;:
## 
##     read.fastq</code></pre>
<pre><code>## ape package version 4.1</code></pre>
<pre><code>## phyloseq package version 1.20.0</code></pre>
<pre><code>## Requiring samples to have 15000 reads</code></pre>
<h2 id="set-up-grouping-variables">Set up grouping variables</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">groupings&lt;-<span class="kw">paste</span>(<span class="kw">ifelse</span>(samples$bonobo[isEnough[samples$Code]],<span class="st">&#39;Bonobo&#39;</span>,<span class="st">&#39;Chimpanzee&#39;</span>),<span class="kw">ifelse</span>(samples$malaria[isEnough[samples$Code]],<span class="st">&#39;Laverania positive&#39;</span>,<span class="st">&#39;Laverania negative&#39;</span>))
groupings2&lt;-<span class="kw">paste</span>(<span class="kw">ifelse</span>(samples$bonobo[isEnough[samples$Code]],<span class="st">&#39;Bonobo &#39;</span>,<span class="st">&#39;Chimpanzee &#39;</span>),<span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&#39;^TL&#39;</span>,samples$area[isEnough[samples$Code]]),<span class="st">&#39;TL2 site&#39;</span>,<span class="kw">ifelse</span>(samples$bonobo[isEnough[samples$Code]],<span class="st">&#39;non-endemic sites&#39;</span>,<span class="st">&#39;all sites&#39;</span>)),<span class="dt">sep=</span><span class="st">&#39;&#39;</span>)</code></pre></div>
<h2 id="set-up-plotting-parameters">Set up plotting parameters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#mildly magic number throwing out 3rd color to leave gap in coloring between two groups</span>
nGroupings&lt;-<span class="kw">length</span>(<span class="kw">unique</span>(groupings))
nGroupings2&lt;-<span class="kw">length</span>(<span class="kw">unique</span>(groupings2))
groupCols&lt;-<span class="kw">rainbow.lab</span>(nGroupings+<span class="kw">length</span>(<span class="kw">unique</span>(groupings2))+<span class="dv">1</span>,<span class="dt">alpha=</span>.<span class="dv">9</span>)[<span class="dv">1</span>:(nGroupings<span class="dv">+1</span>)][-<span class="dv">3</span>]
groupCols2&lt;-<span class="kw">rainbow.lab</span>(nGroupings+nGroupings2<span class="dv">+1</span>,<span class="dt">alpha=</span>.<span class="dv">25</span>)[<span class="dv">1</span>:(nGroupings<span class="dv">+1</span>)][-<span class="dv">3</span>]
<span class="kw">names</span>(groupCols)&lt;-<span class="kw">names</span>(groupCols2)&lt;-<span class="kw">unique</span>(groupings)
group2Cols&lt;-<span class="kw">rainbow.lab</span>(nGroupings+nGroupings2<span class="dv">+1</span>,<span class="dt">alpha=</span>.<span class="dv">9</span>)[<span class="dv">1</span>+nGroupings<span class="dv">+1</span>:nGroupings2]
group2Cols2&lt;-<span class="kw">rainbow.lab</span>(nGroupings+nGroupings2<span class="dv">+1</span>,<span class="dt">alpha=</span>.<span class="dv">25</span>)[<span class="dv">1</span>+nGroupings<span class="dv">+1</span>:nGroupings2]
<span class="kw">names</span>(group2Cols)&lt;-<span class="kw">names</span>(group2Cols2)&lt;-<span class="kw">unique</span>(groupings2)</code></pre></div>
<h2 id="calculate-shannon-diversities">Calculate Shannon diversities</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shannons&lt;-<span class="kw">apply</span>(otuTab[,samples$Code[isEnough[samples$Code]]],<span class="dv">2</span>,shannon)
groupShannon&lt;-<span class="kw">sapply</span>(<span class="kw">unique</span>(groupings),function(xx)<span class="kw">median</span>(shannons[groupings==xx]))
shannonCI&lt;-<span class="kw">sapply</span>(<span class="kw">unique</span>(groupings),function(xx)<span class="kw">medianCI</span>(shannons[groupings==xx]))
group2Shannon&lt;-<span class="kw">sapply</span>(<span class="kw">unique</span>(groupings2),function(xx)<span class="kw">median</span>(shannons[groupings2==xx]))
shannon2CI&lt;-<span class="kw">sapply</span>(<span class="kw">unique</span>(groupings2),function(xx)<span class="kw">medianCI</span>(shannons[groupings2==xx]))</code></pre></div>
<h2 id="calculate-wilcoxon-test-p-values">Calculate Wilcoxon test p-values</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">outer</span>(<span class="kw">split</span>(shannons,groupings),<span class="kw">split</span>(shannons,groupings),function(xx,yy)<span class="kw">mapply</span>(function(xxx,yyy)<span class="kw">wilcox.test</span>(xxx,yyy)$p.value,xx,yy))</code></pre></div>
<pre><code>## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties

## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties

## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties</code></pre>
<pre><code>##                               Bonobo Laverania negative
## Bonobo Laverania negative                    1.00000000
## Bonobo Laverania positive                    0.80779871
## Chimpanzee Laverania negative                0.08170533
## Chimpanzee Laverania positive                0.09433924
##                               Bonobo Laverania positive
## Bonobo Laverania negative                    0.80779871
## Bonobo Laverania positive                    1.00000000
## Chimpanzee Laverania negative                0.09707015
## Chimpanzee Laverania positive                0.18265586
##                               Chimpanzee Laverania negative
## Bonobo Laverania negative                        0.08170533
## Bonobo Laverania positive                        0.09707015
## Chimpanzee Laverania negative                    1.00000000
## Chimpanzee Laverania positive                    0.18568852
##                               Chimpanzee Laverania positive
## Bonobo Laverania negative                        0.09433924
## Bonobo Laverania positive                        0.18265586
## Chimpanzee Laverania negative                    0.18568852
## Chimpanzee Laverania positive                    1.00000000</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">outer</span>(<span class="kw">split</span>(shannons,groupings2),<span class="kw">split</span>(shannons,groupings2),function(xx,yy)<span class="kw">mapply</span>(function(xxx,yyy)<span class="kw">wilcox.test</span>(xxx,yyy)$p.value,xx,yy))</code></pre></div>
<pre><code>## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties

## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties

## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties</code></pre>
<pre><code>##                          Bonobo non-endemic sites Bonobo TL2 site
## Bonobo non-endemic sites                1.0000000       0.5958866
## Bonobo TL2 site                         0.5958866       1.0000000
## Chimpanzee all sites                    0.9940949       0.8531955
##                          Chimpanzee all sites
## Bonobo non-endemic sites            0.9940949
## Bonobo TL2 site                     0.8531955
## Chimpanzee all sites                1.0000000</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">outer</span>(<span class="kw">split</span>(shannons,<span class="kw">sub</span>(<span class="st">&#39;Plasmodi.*&#39;</span>,<span class="st">&#39;&#39;</span>,groupings)),<span class="kw">split</span>(shannons,<span class="kw">sub</span>(<span class="st">&#39;Plasmodi.*&#39;</span>,<span class="st">&#39;&#39;</span>,groupings)),function(xx,yy)<span class="kw">mapply</span>(function(xxx,yyy)<span class="kw">wilcox.test</span>(xxx,yyy)$p.value,xx,yy))</code></pre></div>
<pre><code>## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties

## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties

## Warning in wilcox.test.default(xxx, yyy): cannot compute exact p-value with
## ties</code></pre>
<pre><code>##                               Bonobo Laverania negative
## Bonobo Laverania negative                    1.00000000
## Bonobo Laverania positive                    0.80779871
## Chimpanzee Laverania negative                0.08170533
## Chimpanzee Laverania positive                0.09433924
##                               Bonobo Laverania positive
## Bonobo Laverania negative                    0.80779871
## Bonobo Laverania positive                    1.00000000
## Chimpanzee Laverania negative                0.09707015
## Chimpanzee Laverania positive                0.18265586
##                               Chimpanzee Laverania negative
## Bonobo Laverania negative                        0.08170533
## Bonobo Laverania positive                        0.09707015
## Chimpanzee Laverania negative                    1.00000000
## Chimpanzee Laverania positive                    0.18568852
##                               Chimpanzee Laverania positive
## Bonobo Laverania negative                        0.09433924
## Bonobo Laverania positive                        0.18265586
## Chimpanzee Laverania negative                    0.18568852
## Chimpanzee Laverania positive                    1.00000000</code></pre>
<h2 id="generate-distance-comparison-figure">Generate distance comparison figure</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">spacer&lt;-.<span class="dv">5</span>
<span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="fl">6.6</span>,<span class="dv">4</span>,.<span class="dv">3</span>,<span class="dv">5</span>),<span class="dt">lheight=</span>.<span class="dv">85</span>)
<span class="kw">plot</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">type=</span><span class="st">&#39;n&#39;</span>,<span class="dt">ylab=</span><span class="st">&#39;Shannon diversity&#39;</span>,<span class="dt">las=</span><span class="dv">2</span>,<span class="dt">xlim=</span><span class="kw">c</span>(.<span class="dv">5</span>,<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">c</span>(groupings,groupings2)))+.<span class="dv">5</span>+spacer),<span class="dt">ylim=</span><span class="kw">range</span>(shannons),<span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">xlab=</span><span class="st">&#39;&#39;</span>,<span class="dt">bty=</span><span class="st">&#39;l&#39;</span>)
groupFac&lt;-<span class="kw">factor</span>(<span class="kw">sub</span>(<span class="st">&#39; Plas&#39;</span>,<span class="st">&#39;</span><span class="ch">\n</span><span class="st">Plas&#39;</span>,groupings))
xPos&lt;-<span class="kw">as.numeric</span>(groupFac)+<span class="kw">offsetX</span>(shannons,groupFac,<span class="dt">width=</span>.<span class="dv">3</span>)
width&lt;-.<span class="dv">45</span>
<span class="kw">segments</span>(<span class="dv">1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac))-width,groupShannon[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac))],<span class="dv">1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac))+width,groupShannon[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac))],<span class="dt">lwd=</span><span class="dv">3</span>,<span class="dt">col=</span>groupCols[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac))])
<span class="kw">rect</span>(<span class="dv">1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac))-width,shannonCI[<span class="dv">1</span>,<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac))],<span class="dv">1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac))+width,shannonCI[<span class="dv">2</span>,<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac))],<span class="dt">lwd=</span><span class="dv">2</span>,<span class="dt">border=</span><span class="ot">NA</span>,<span class="dt">col=</span>groupCols2[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac))])
<span class="kw">points</span>(xPos,shannons,<span class="dt">pch=</span><span class="dv">21</span>,<span class="dt">bg=</span>groupCols[groupings],<span class="dt">cex=</span><span class="fl">1.7</span>)
<span class="kw">slantAxis</span>(<span class="dv">1</span>,<span class="dv">1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac)),<span class="kw">sub</span>(<span class="st">&#39;(Chimpanzee|Bonobo)&#39;</span>,<span class="st">&#39;</span><span class="ch">\\</span><span class="st">1 samples&#39;</span>,<span class="kw">sub</span>(<span class="st">&#39; Laverania&#39;</span>,<span class="st">&#39;</span><span class="ch">\n</span><span class="st">Laverania&#39;</span>,<span class="kw">levels</span>(groupFac))),<span class="dt">srt=</span>-<span class="dv">30</span>)
groupFac2&lt;-<span class="kw">factor</span>(groupings2,<span class="dt">levels=</span><span class="kw">unique</span>(groupings2))
offset&lt;-<span class="kw">max</span>(<span class="kw">as.numeric</span>(groupFac))+spacer
<span class="kw">abline</span>(<span class="dt">v=</span>offset<span class="fl">+.5</span>-spacer/<span class="dv">2</span>,<span class="dt">lty=</span><span class="dv">2</span>)
xPos&lt;-offset+<span class="kw">as.numeric</span>(groupFac2)+<span class="kw">offsetX</span>(shannons,groupFac2,<span class="dt">width=</span>.<span class="dv">3</span>)
width&lt;-.<span class="dv">45</span>
<span class="kw">segments</span>(offset<span class="dv">+1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac2))-width,group2Shannon[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac2))],offset<span class="dv">+1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac2))+width,group2Shannon[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac2))],<span class="dt">lwd=</span><span class="dv">3</span>,<span class="dt">col=</span>group2Cols[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac2))])
<span class="kw">rect</span>(offset<span class="dv">+1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac2))-width,shannon2CI[<span class="dv">1</span>,<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac2))],offset<span class="dv">+1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac2))+width,shannon2CI[<span class="dv">2</span>,<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac2))],<span class="dt">lwd=</span><span class="dv">2</span>,<span class="dt">border=</span><span class="ot">NA</span>,<span class="dt">col=</span>group2Cols2[<span class="kw">sub</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="st">&#39; &#39;</span>,<span class="kw">levels</span>(groupFac2))])
<span class="kw">points</span>(xPos,shannons,<span class="dt">pch=</span><span class="dv">21</span>,<span class="dt">bg=</span>group2Cols[groupings2],<span class="dt">cex=</span><span class="fl">1.7</span>)
<span class="co">#replacing first space with \n</span>
<span class="kw">slantAxis</span>(<span class="dv">1</span>,offset<span class="dv">+1</span>:<span class="kw">length</span>(<span class="kw">levels</span>(groupFac2)),<span class="kw">sub</span>(<span class="st">&#39;(Chimpanzee|Bonobo)&#39;</span>,<span class="st">&#39;</span><span class="ch">\\</span><span class="st">1 samples&#39;</span>,<span class="kw">sub</span>(<span class="st">&#39; &#39;</span>,<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="kw">levels</span>(groupFac2))),<span class="dt">srt=</span>-<span class="dv">30</span>)</code></pre></div>
<p><img src="figure/shannonDiversity-1.png" alt="plot of chunk shannonDiversity" /> ## Load libraries</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<h2 id="load-the-data-1">Load the data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;readData.R&#39;</span>)</code></pre></div>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dnar&#39;:
## 
##     read.fastq</code></pre>
<pre><code>## ape package version 4.1</code></pre>
<pre><code>## phyloseq package version 1.20.0</code></pre>
<pre><code>## Requiring samples to have 15000 reads</code></pre>
<h2 id="setup-plotting-parameters">Setup plotting parameters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samples&lt;-samples[<span class="kw">order</span>(samples$Species,samples$area2,samples$malaria,samples$Code),]
moreThanProp&lt;-<span class="kw">apply</span>(otuProp[,samples$Code[isEnough[samples$Code]]],<span class="dv">1</span>,max,<span class="dt">na.rm=</span><span class="ot">TRUE</span>)&gt;.<span class="dv">02</span>
cols&lt;-<span class="kw">c</span>(<span class="st">&#39;white&#39;</span>,<span class="kw">tail</span>(<span class="kw">rev</span>(<span class="kw">heat.colors</span>(<span class="dv">110</span>)),<span class="dv">99</span>)) 
plotProp&lt;-<span class="kw">t</span>(otuProp[moreThanProp,samples$Code[isEnough[samples$Code]]])
maxProp&lt;-<span class="kw">apply</span>(otuProp[moreThanProp,samples$Code[isEnough[samples$Code]]],<span class="dv">1</span>,function(x)x/<span class="kw">max</span>(x))
<span class="kw">colnames</span>(plotProp)&lt;-<span class="kw">colnames</span>(maxProp)&lt;-taxa[<span class="kw">colnames</span>(plotProp),<span class="st">&#39;bestId&#39;</span>]
plotProp&lt;-plotProp[,<span class="kw">order</span>(<span class="kw">apply</span>(plotProp,<span class="dv">2</span>,mean),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)]
maxTree&lt;-<span class="kw">hclust</span>(<span class="kw">dist</span>(<span class="kw">t</span>(maxProp[,])))
maxProp&lt;-maxProp[,<span class="kw">rev</span>(maxTree$labels[maxTree$order])]
plotProp&lt;-plotProp[,<span class="kw">rev</span>(maxTree$labels[maxTree$order])]
breaks&lt;-<span class="kw">c</span>(-<span class="fl">1e-6</span>,<span class="kw">seq</span>(<span class="kw">min</span>(plotProp[plotProp&gt;<span class="dv">0</span>])-<span class="fl">1e-10</span>,<span class="kw">max</span>(plotProp)+<span class="fl">1e-10</span>,<span class="dt">length.out=</span><span class="dv">100</span>))
breaks2&lt;-<span class="kw">c</span>(-<span class="fl">1e-6</span>,<span class="kw">seq</span>(<span class="kw">min</span>(maxProp[maxProp&gt;<span class="dv">0</span>])-<span class="fl">1e-10</span>,<span class="kw">max</span>(maxProp)+<span class="fl">1e-10</span>,<span class="dt">length.out=</span><span class="dv">100</span>))</code></pre></div>
<h2 id="plot-heatmap-of-bacterial-proportions">Plot heatmap of bacterial proportions</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>),<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">12</span>,.<span class="dv">1</span>,<span class="fl">3.5</span>,<span class="dv">14</span>))
<span class="kw">image</span>(<span class="dv">1</span>:<span class="kw">ncol</span>(plotProp),<span class="dv">1</span>:<span class="kw">nrow</span>(plotProp),<span class="kw">t</span>(plotProp),<span class="dt">col=</span>cols,<span class="dt">breaks=</span>breaks,<span class="dt">xlab=</span><span class="st">&#39;&#39;</span>,<span class="dt">ylab=</span><span class="st">&#39;&#39;</span>,<span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)
<span class="kw">text</span>(<span class="kw">grconvertX</span>(.<span class="dv">005</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),<span class="kw">grconvertY</span>(.<span class="dv">995</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),<span class="st">&#39;A)&#39;</span>,<span class="dt">xpd=</span><span class="ot">NA</span>,<span class="dt">cex=</span><span class="dv">3</span>,<span class="dt">adj=</span><span class="dv">0</span>:<span class="dv">1</span>)
<span class="kw">box</span>()
<span class="kw">insetScale</span>(<span class="kw">round</span>(breaks2,<span class="dv">6</span>),cols,<span class="kw">c</span>(.<span class="dv">97</span>,.<span class="dv">58</span>,.<span class="dv">98</span>,.<span class="dv">83</span>),<span class="dt">label=</span><span class="st">&#39;Proportion of OTU within each sample&#39;</span>)
<span class="kw">axis</span>(<span class="dv">1</span>,<span class="dv">1</span>:<span class="kw">ncol</span>(plotProp),<span class="kw">colnames</span>(plotProp),<span class="dt">cex.axis=</span>.<span class="dv">7</span>,<span class="dt">las=</span><span class="dv">2</span>,<span class="dt">tcl=</span>-.<span class="dv">1</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">3</span>,<span class="dv">0</span>))
metadata&lt;-samples[<span class="kw">rownames</span>(plotProp),<span class="kw">c</span>(<span class="st">&#39;chimpBonobo&#39;</span>,<span class="st">&#39;area2&#39;</span>,<span class="st">&#39;plasmoPM&#39;</span>,<span class="st">&#39;Code&#39;</span>)]
<span class="kw">colnames</span>(metadata)&lt;-<span class="kw">c</span>(<span class="st">&#39;Species&#39;</span>,<span class="st">&#39;Site&#39;</span>,<span class="st">&#39;Laverania&#39;</span>,<span class="st">&#39;Sample&#39;</span>)
<span class="kw">addMetaData</span>(metadata,<span class="dt">cex=</span>.<span class="dv">75</span>)
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">1</span>:<span class="kw">nrow</span>(plotProp)-.<span class="dv">5</span>,<span class="dt">v=</span><span class="dv">1</span>:<span class="kw">ncol</span>(plotProp)+.<span class="dv">5</span>,<span class="dt">col=</span><span class="st">&#39;#00000011&#39;</span>)
<span class="kw">image</span>(<span class="dv">1</span>:<span class="kw">ncol</span>(maxProp),<span class="dv">1</span>:<span class="kw">nrow</span>(maxProp),<span class="kw">t</span>(maxProp),<span class="dt">col=</span>cols,<span class="dt">breaks=</span>breaks2,<span class="dt">xlab=</span><span class="st">&#39;&#39;</span>,<span class="dt">ylab=</span><span class="st">&#39;&#39;</span>,<span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)
<span class="kw">text</span>(<span class="kw">grconvertX</span>(.<span class="dv">005</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),<span class="kw">grconvertY</span>(.<span class="dv">995</span>, <span class="st">&quot;nfc&quot;</span>, <span class="st">&quot;user&quot;</span>),<span class="st">&#39;B)&#39;</span>,<span class="dt">xpd=</span><span class="ot">NA</span>,<span class="dt">cex=</span><span class="dv">3</span>,<span class="dt">adj=</span><span class="dv">0</span>:<span class="dv">1</span>)
<span class="kw">box</span>()
<span class="kw">insetScale</span>(<span class="kw">round</span>(breaks2,<span class="dv">6</span>),cols,<span class="kw">c</span>(.<span class="dv">97</span>,.<span class="dv">58</span>,.<span class="dv">98</span>,.<span class="dv">83</span>),<span class="dt">label=</span><span class="st">&#39;Proportion of OTU maximum&#39;</span>)
<span class="kw">axis</span>(<span class="dv">1</span>,<span class="dv">1</span>:<span class="kw">ncol</span>(maxProp),<span class="kw">colnames</span>(maxProp),<span class="dt">cex.axis=</span>.<span class="dv">7</span>,<span class="dt">las=</span><span class="dv">2</span>,<span class="dt">tcl=</span>-.<span class="dv">1</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">3</span>,<span class="dv">0</span>))
<span class="kw">abline</span>(<span class="dt">h=</span><span class="dv">1</span>:<span class="kw">nrow</span>(maxProp)-.<span class="dv">5</span>,<span class="dt">v=</span><span class="dv">1</span>:<span class="kw">ncol</span>(maxProp)+.<span class="dv">5</span>,<span class="dt">col=</span><span class="st">&#39;#00000011&#39;</span>)
metadata&lt;-samples[<span class="kw">rownames</span>(maxProp),<span class="kw">c</span>(<span class="st">&#39;chimpBonobo&#39;</span>,<span class="st">&#39;area2&#39;</span>,<span class="st">&#39;plasmoPM&#39;</span>,<span class="st">&#39;Code&#39;</span>)]
<span class="kw">colnames</span>(metadata)&lt;-<span class="kw">c</span>(<span class="st">&#39;Species&#39;</span>,<span class="st">&#39;Site&#39;</span>,<span class="st">&#39;Laverania&#39;</span>,<span class="st">&#39;Sample&#39;</span>)
<span class="kw">addMetaData</span>(metadata,<span class="dt">cex=</span>.<span class="dv">75</span>)</code></pre></div>
<div class="figure">
<img src="figure/bacteriaHeatmap-1.png" alt="plot of chunk bacteriaHeatmap" />
<p class="caption">plot of chunk bacteriaHeatmap</p>
</div>
<h2 id="calculate-p-values">Calculate p-values</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sampleOrder&lt;-<span class="kw">withAs</span>(<span class="dt">s=</span>samples[isEnough[samples$Code],],s$Code[<span class="kw">order</span>(!s$bonobo,<span class="kw">ifelse</span>(s$area==<span class="st">&#39;KR&#39;</span>,<span class="ot">FALSE</span>,s$malaria),s$area2,s$malaria)])
tlPos&lt;-samples$Code[isEnough[samples$Code]&amp;samples$isTL&amp;samples$malaria]
tlNeg&lt;-samples$Code[isEnough[samples$Code]&amp;samples$isTL&amp;!samples$malaria]
<span class="co">#from functions.R</span>
tmp&lt;-<span class="kw">setupHeat</span>(tlPos,tlNeg,otuProp[,sampleOrder],taxa,<span class="dt">pCut=</span>.<span class="dv">05</span>)
selectProp&lt;-tmp[[<span class="dv">1</span>]]
effectSplit&lt;-tmp[[<span class="dv">2</span>]]</code></pre></div>
<h2 id="plot-otus-that-different-between-tl-2-plasmodium-positive-and-negative-bonobos">Plot OTUs that different between TL-2 plasmodium positive and negative bonobos</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="fl">11.5</span>,.<span class="dv">1</span>,<span class="dv">3</span>,<span class="fl">13.5</span>),<span class="dt">lheight=</span>.<span class="dv">7</span>)
<span class="kw">plotHeat</span>(selectProp,breaks2,cols,<span class="dt">yaxt=</span><span class="st">&#39;n&#39;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span>effectSplit<span class="fl">-.5</span>)
metadata&lt;-samples[<span class="kw">rownames</span>(selectProp),<span class="kw">c</span>(<span class="st">&#39;chimpBonobo&#39;</span>,<span class="st">&#39;area2&#39;</span>,<span class="st">&#39;plasmoPM&#39;</span>,<span class="st">&#39;Code&#39;</span>)]
<span class="kw">colnames</span>(metadata)&lt;-<span class="kw">c</span>(<span class="st">&#39;Species&#39;</span>,<span class="st">&#39;Site&#39;</span>,<span class="st">&#39;Laverania&#39;</span>,<span class="st">&#39;Sample&#39;</span>)
<span class="kw">addMetaData</span>(metadata,<span class="dt">cex=</span>.<span class="dv">75</span>)</code></pre></div>
<div class="figure">
<img src="figure/differingOtus-1.png" alt="plot of chunk differingOtus" />
<p class="caption">plot of chunk differingOtus</p>
</div>
<h2 id="load-libraries-5">Load libraries</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phyloseq)
<span class="kw">packageVersion</span>(<span class="st">&#39;phyloseq&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;1.20.0&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Rtsne)
<span class="kw">packageVersion</span>(<span class="st">&#39;Rtsne&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.13&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ape)
<span class="kw">packageVersion</span>(<span class="st">&#39;ape&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;4.1&#39;</code></pre>
<h2 id="load-the-data-2">Load the data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;readData.R&#39;</span>)</code></pre></div>
<pre><code>## ape package version 4.1</code></pre>
<pre><code>## phyloseq package version 1.20.0</code></pre>
<pre><code>## Requiring samples to have 15000 reads</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;../functions.R&#39;</span>)</code></pre></div>
<h2 id="calculate-distance-matrices-from-otu-table">Calculate distance matrices from OTU table</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  selectSamples&lt;-samples[isEnough[samples$Code],]
  subsampledOtus&lt;-<span class="kw">apply</span>(otuTab[<span class="kw">rownames</span>(otuTab) %in%<span class="st"> </span>tree$tip.label,selectSamples$Code],<span class="dv">2</span>,rarefyCounts,nRequiredReads)
  phyOtu&lt;-<span class="kw">otu_table</span>(subsampledOtus,<span class="dt">taxa_are_rows=</span><span class="ot">TRUE</span>)
  qiimeData&lt;-<span class="kw">phyloseq</span>(<span class="dt">otu_table=</span>phyOtu,<span class="dt">phy_tree=</span>tree)
  uniDist&lt;-<span class="kw">UniFrac</span>(qiimeData,<span class="dt">weighted=</span><span class="ot">FALSE</span>)
  <span class="kw">save</span>(uniDist,<span class="dt">file=</span><span class="st">&#39;work/uniDist.csv&#39;</span>)
  phyOtuW&lt;-<span class="kw">otu_table</span>(otuProp[<span class="kw">rownames</span>(otuTab) %in%<span class="st"> </span>tree$tip.label,selectSamples$Code],<span class="dt">taxa_are_rows=</span><span class="ot">TRUE</span>)
  qiimeDataW&lt;-<span class="kw">phyloseq</span>(<span class="dt">otu_table=</span>phyOtuW,<span class="dt">phy_tree=</span>tree)
  uniDistW&lt;-<span class="kw">UniFrac</span>(qiimeDataW,<span class="dt">weighted=</span><span class="ot">TRUE</span>)
  brayDist&lt;-<span class="kw">distance</span>(qiimeDataW,<span class="st">&#39;bray&#39;</span>)
  brayDistUW&lt;-<span class="kw">distance</span>(qiimeData,<span class="st">&#39;bray&#39;</span>,<span class="dt">binary=</span><span class="ot">TRUE</span>)
  uniPca&lt;-<span class="kw">pcoa</span>(uniDist)
  mantels&lt;-<span class="kw">list</span>(
    <span class="st">&#39;uniW&#39;</span>=ade4::<span class="kw">mantel.rtest</span>(uniDist,uniDistW,<span class="dt">nrepet=</span><span class="fl">1e4</span>),
    <span class="st">&#39;brayW&#39;</span>=ade4::<span class="kw">mantel.rtest</span>(uniDist,brayDist,<span class="dt">nrepet=</span><span class="fl">1e4</span>),
    <span class="st">&#39;brayUW&#39;</span>=ade4::<span class="kw">mantel.rtest</span>(uniDist,brayDistUW,<span class="dt">nrepet=</span><span class="fl">1e4</span>)
  )
  predictors&lt;-<span class="kw">model.matrix</span>(~<span class="dv">0</span>+Species+malaria+SIV+area,selectSamples)
  <span class="kw">colnames</span>(predictors)&lt;-<span class="kw">sub</span>(<span class="st">&#39;^Species&#39;</span>,<span class="st">&#39;&#39;</span>,<span class="kw">colnames</span>(predictors))
  <span class="kw">colnames</span>(predictors)[<span class="kw">colnames</span>(predictors)==<span class="st">&#39;malariaTRUE&#39;</span>]&lt;-<span class="st">&#39;malariaPos&#39;</span></code></pre></div>
<h2 id="set-up-plotting-parameters-1">Set up plotting parameters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colorBrew&lt;-<span class="kw">c</span>(<span class="st">&#39;#e41a1cBB&#39;</span>,<span class="st">&#39;#377eb8BB&#39;</span>,<span class="st">&#39;#4daf4aBB&#39;</span>,<span class="st">&#39;#984ea3BB&#39;</span>,<span class="st">&#39;#ff7f00BB&#39;</span>,<span class="st">&#39;#ffff33BB&#39;</span>,<span class="st">&#39;#a65628BB&#39;</span>,<span class="st">&#39;#f781bfBB&#39;</span>,<span class="st">&#39;#999999BB&#39;</span>,<span class="st">&#39;#88ddffBB&#39;</span>)
nArea&lt;-<span class="kw">length</span>(<span class="kw">unique</span>(selectSamples$area2))
if(nArea&gt;<span class="kw">length</span>(colorBrew))<span class="kw">stop</span>(<span class="st">&#39;Need to adjust colors for more areas&#39;</span>)
areaCols&lt;-colorBrew[<span class="dv">1</span>:nArea]
<span class="kw">names</span>(areaCols)&lt;-<span class="kw">unique</span>(selectSamples$area2)
areaPch&lt;-<span class="kw">sapply</span>(<span class="kw">names</span>(areaCols),function(x)<span class="kw">mostAbundant</span>(selectSamples$chimpBonobo[selectSamples$area2==x]))
malariaCols&lt;-<span class="kw">c</span>(<span class="st">&#39;#00000022&#39;</span>,<span class="st">&#39;#000000CC&#39;</span>)
mediumMalariaCol&lt;-<span class="st">&#39;#00000077&#39;</span>
malariaCols3&lt;-<span class="kw">c</span>(<span class="st">&#39;#00000022&#39;</span>,<span class="st">&#39;#000000CC&#39;</span>)
<span class="kw">names</span>(malariaCols3)&lt;-<span class="kw">names</span>(malariaCols)&lt;-<span class="kw">c</span>(<span class="st">&#39;Laverania negative&#39;</span>,<span class="st">&#39;Laverania positive&#39;</span>)
speciesPch&lt;-<span class="dv">20+1</span>:<span class="kw">length</span>(<span class="kw">unique</span>(selectSamples$chimpBonobo))
speciesCols&lt;-<span class="kw">rainbow.lab</span>(<span class="kw">length</span>(<span class="kw">unique</span>(selectSamples$chimpBonobo)),<span class="dt">start=</span>-<span class="dv">2</span>,<span class="dt">end=</span><span class="dv">1</span>,<span class="dt">alpha=</span>.<span class="dv">8</span>,<span class="dt">lightMultiple=</span>.<span class="dv">8</span>)
<span class="kw">names</span>(speciesCols)&lt;-<span class="kw">names</span>(speciesPch)&lt;-<span class="kw">sort</span>(<span class="kw">unique</span>(selectSamples$chimpBonobo))
importance&lt;-uniPca$values$Relative_eig
<span class="kw">colnames</span>(uniPca$vectors)&lt;-<span class="kw">sprintf</span>(<span class="st">&#39;Principal coordinate %d (%d%% of variance)&#39;</span>,<span class="dv">1</span>:<span class="kw">length</span>(importance),<span class="kw">round</span>(importance*<span class="dv">100</span>))[<span class="dv">1</span>:<span class="kw">ncol</span>(targetPca$vectors)]</code></pre></div>
<pre><code>## Error in ncol(targetPca$vectors): object &#39;targetPca&#39; not found</code></pre>
<h2 id="plot-16s-pcoa">Plot 16s PCoA</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  axes&lt;-<span class="dv">1</span>:<span class="dv">2</span>
  <span class="co">#malaria/species</span>
  pos&lt;-<span class="kw">my.biplot.pcoa</span>(uniPca,predictors,<span class="dt">plot.axes=</span>axes,<span class="dt">pch=</span><span class="dv">21</span>,<span class="dt">bg=</span>speciesCols[selectSamples$chimpBonobo],<span class="dt">col=</span>malariaCols3[selectSamples$malaria<span class="dv">+1</span>],<span class="dt">cex=</span><span class="fl">2.25</span>,<span class="dt">lwd=</span><span class="dv">4</span>,<span class="dt">arrowsFilter=</span><span class="ot">Inf</span>,<span class="dt">las=</span><span class="dv">1</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="fl">2.75</span>,.<span class="dv">75</span>,<span class="dv">0</span>),<span class="dt">sameAxis=</span><span class="ot">FALSE</span>,<span class="dt">bty=</span><span class="st">&#39;l&#39;</span>,<span class="dt">type=</span><span class="st">&#39;n&#39;</span>)
  <span class="kw">points</span>(pos[!selectSamples$malaria,],<span class="dt">col=</span>malariaCols3[<span class="dv">1</span>],<span class="dt">cex=</span><span class="fl">2.25</span>,<span class="dt">lwd=</span><span class="dv">4</span>,<span class="dt">bg=</span>speciesCols[selectSamples[!selectSamples$malaria,<span class="st">&#39;chimpBonobo&#39;</span>]],<span class="dt">pch=</span><span class="dv">21</span>)
  <span class="kw">points</span>(pos[selectSamples$malaria,],<span class="dt">col=</span>malariaCols3[<span class="dv">2</span>],<span class="dt">cex=</span><span class="fl">2.25</span>,<span class="dt">lwd=</span><span class="dv">4</span>,<span class="dt">bg=</span>speciesCols[selectSamples[selectSamples$malaria,<span class="st">&#39;chimpBonobo&#39;</span>]],<span class="dt">pch=</span><span class="dv">21</span>)
  <span class="kw">title</span>(<span class="dt">main=</span><span class="kw">sprintf</span>(<span class="st">&#39;16S rRNA&#39;</span>,axes[<span class="dv">1</span>],axes[<span class="dv">2</span>]))</code></pre></div>
<div class="figure">
<img src="figure/bacteriaPCoA-1.png" alt="plot of chunk bacteriaPCoA" />
<p class="caption">plot of chunk bacteriaPCoA</p>
</div>
<h2 id="plot-16s-t-sne">Plot 16s t-SNE</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tsne&lt;-<span class="kw">Rtsne</span>(uniDist,<span class="dt">is_distance=</span><span class="ot">TRUE</span>,<span class="dt">verbose=</span><span class="ot">TRUE</span>,<span class="dt">perplexity=</span><span class="dv">15</span>,<span class="dt">max_iter=</span><span class="dv">5000</span>)</code></pre></div>
<pre><code>## Read the 94 x 94 data matrix successfully!
## Using no_dims = 2, perplexity = 15.000000, and theta = 0.500000
## Computing input similarities...
## Building tree...
##  - point 0 of 94
## Done in 0.01 seconds (sparsity = 0.616342)!
## Learning embedding...
## Iteration 50: error is 58.289865 (50 iterations in 0.04 seconds)
## Iteration 100: error is 59.650702 (50 iterations in 0.03 seconds)
## Iteration 150: error is 58.595945 (50 iterations in 0.03 seconds)
## Iteration 200: error is 59.967717 (50 iterations in 0.02 seconds)
## Iteration 250: error is 56.462338 (50 iterations in 0.02 seconds)
## Iteration 300: error is 1.886983 (50 iterations in 0.01 seconds)
## Iteration 350: error is 1.326446 (50 iterations in 0.01 seconds)
## Iteration 400: error is 0.967031 (50 iterations in 0.01 seconds)
## Iteration 450: error is 0.666075 (50 iterations in 0.01 seconds)
## Iteration 500: error is 0.443368 (50 iterations in 0.01 seconds)
## Iteration 550: error is 0.392551 (50 iterations in 0.01 seconds)
## Iteration 600: error is 0.382026 (50 iterations in 0.01 seconds)
## Iteration 650: error is 0.378080 (50 iterations in 0.01 seconds)
## Iteration 700: error is 0.332201 (50 iterations in 0.01 seconds)
## Iteration 750: error is 0.321300 (50 iterations in 0.01 seconds)
## Iteration 800: error is 0.315114 (50 iterations in 0.01 seconds)
## Iteration 850: error is 0.281121 (50 iterations in 0.01 seconds)
## Iteration 900: error is 0.280487 (50 iterations in 0.01 seconds)
## Iteration 950: error is 0.278743 (50 iterations in 0.01 seconds)
## Iteration 1000: error is 0.278635 (50 iterations in 0.01 seconds)
## Iteration 1050: error is 0.279910 (50 iterations in 0.01 seconds)
## Iteration 1100: error is 0.279977 (50 iterations in 0.01 seconds)
## Iteration 1150: error is 0.278770 (50 iterations in 0.01 seconds)
## Iteration 1200: error is 0.280171 (50 iterations in 0.01 seconds)
## Iteration 1250: error is 0.280643 (50 iterations in 0.01 seconds)
## Iteration 1300: error is 0.280017 (50 iterations in 0.01 seconds)
## Iteration 1350: error is 0.281352 (50 iterations in 0.01 seconds)
## Iteration 1400: error is 0.280629 (50 iterations in 0.01 seconds)
## Iteration 1450: error is 0.278825 (50 iterations in 0.01 seconds)
## Iteration 1500: error is 0.280534 (50 iterations in 0.01 seconds)
## Iteration 1550: error is 0.281185 (50 iterations in 0.01 seconds)
## Iteration 1600: error is 0.283002 (50 iterations in 0.01 seconds)
## Iteration 1650: error is 0.279119 (50 iterations in 0.01 seconds)
## Iteration 1700: error is 0.282295 (50 iterations in 0.01 seconds)
## Iteration 1750: error is 0.278931 (50 iterations in 0.01 seconds)
## Iteration 1800: error is 0.280701 (50 iterations in 0.01 seconds)
## Iteration 1850: error is 0.279879 (50 iterations in 0.01 seconds)
## Iteration 1900: error is 0.279457 (50 iterations in 0.01 seconds)
## Iteration 1950: error is 0.279548 (50 iterations in 0.01 seconds)
## Iteration 2000: error is 0.282116 (50 iterations in 0.01 seconds)
## Iteration 2050: error is 0.281311 (50 iterations in 0.01 seconds)
## Iteration 2100: error is 0.279824 (50 iterations in 0.01 seconds)
## Iteration 2150: error is 0.278998 (50 iterations in 0.01 seconds)
## Iteration 2200: error is 0.280140 (50 iterations in 0.01 seconds)
## Iteration 2250: error is 0.280266 (50 iterations in 0.01 seconds)
## Iteration 2300: error is 0.282527 (50 iterations in 0.01 seconds)
## Iteration 2350: error is 0.276199 (50 iterations in 0.02 seconds)
## Iteration 2400: error is 0.278470 (50 iterations in 0.02 seconds)
## Iteration 2450: error is 0.278977 (50 iterations in 0.02 seconds)
## Iteration 2500: error is 0.279900 (50 iterations in 0.02 seconds)
## Iteration 2550: error is 0.279521 (50 iterations in 0.02 seconds)
## Iteration 2600: error is 0.280701 (50 iterations in 0.02 seconds)
## Iteration 2650: error is 0.280156 (50 iterations in 0.02 seconds)
## Iteration 2700: error is 0.279635 (50 iterations in 0.02 seconds)
## Iteration 2750: error is 0.281502 (50 iterations in 0.02 seconds)
## Iteration 2800: error is 0.281417 (50 iterations in 0.02 seconds)
## Iteration 2850: error is 0.280907 (50 iterations in 0.02 seconds)
## Iteration 2900: error is 0.280152 (50 iterations in 0.02 seconds)
## Iteration 2950: error is 0.280882 (50 iterations in 0.02 seconds)
## Iteration 3000: error is 0.281527 (50 iterations in 0.02 seconds)
## Iteration 3050: error is 0.282565 (50 iterations in 0.02 seconds)
## Iteration 3100: error is 0.279673 (50 iterations in 0.02 seconds)
## Iteration 3150: error is 0.279873 (50 iterations in 0.02 seconds)
## Iteration 3200: error is 0.279581 (50 iterations in 0.02 seconds)
## Iteration 3250: error is 0.282249 (50 iterations in 0.02 seconds)
## Iteration 3300: error is 0.281236 (50 iterations in 0.02 seconds)
## Iteration 3350: error is 0.279163 (50 iterations in 0.02 seconds)
## Iteration 3400: error is 0.281716 (50 iterations in 0.02 seconds)
## Iteration 3450: error is 0.282428 (50 iterations in 0.02 seconds)
## Iteration 3500: error is 0.280189 (50 iterations in 0.02 seconds)
## Iteration 3550: error is 0.280365 (50 iterations in 0.02 seconds)
## Iteration 3600: error is 0.279144 (50 iterations in 0.02 seconds)
## Iteration 3650: error is 0.280452 (50 iterations in 0.02 seconds)
## Iteration 3700: error is 0.280388 (50 iterations in 0.02 seconds)
## Iteration 3750: error is 0.279483 (50 iterations in 0.02 seconds)
## Iteration 3800: error is 0.281393 (50 iterations in 0.02 seconds)
## Iteration 3850: error is 0.282006 (50 iterations in 0.02 seconds)
## Iteration 3900: error is 0.278511 (50 iterations in 0.02 seconds)
## Iteration 3950: error is 0.282289 (50 iterations in 0.02 seconds)
## Iteration 4000: error is 0.281744 (50 iterations in 0.02 seconds)
## Iteration 4050: error is 0.280475 (50 iterations in 0.02 seconds)
## Iteration 4100: error is 0.280037 (50 iterations in 0.02 seconds)
## Iteration 4150: error is 0.280059 (50 iterations in 0.02 seconds)
## Iteration 4200: error is 0.280619 (50 iterations in 0.02 seconds)
## Iteration 4250: error is 0.280058 (50 iterations in 0.02 seconds)
## Iteration 4300: error is 0.281159 (50 iterations in 0.02 seconds)
## Iteration 4350: error is 0.280558 (50 iterations in 0.02 seconds)
## Iteration 4400: error is 0.281114 (50 iterations in 0.02 seconds)
## Iteration 4450: error is 0.279915 (50 iterations in 0.02 seconds)
## Iteration 4500: error is 0.280025 (50 iterations in 0.02 seconds)
## Iteration 4550: error is 0.279443 (50 iterations in 0.02 seconds)
## Iteration 4600: error is 0.279553 (50 iterations in 0.02 seconds)
## Iteration 4650: error is 0.279743 (50 iterations in 0.02 seconds)
## Iteration 4700: error is 0.281385 (50 iterations in 0.02 seconds)
## Iteration 4750: error is 0.279032 (50 iterations in 0.02 seconds)
## Iteration 4800: error is 0.278856 (50 iterations in 0.02 seconds)
## Iteration 4850: error is 0.281935 (50 iterations in 0.02 seconds)
## Iteration 4900: error is 0.280185 (50 iterations in 0.02 seconds)
## Iteration 4950: error is 0.280052 (50 iterations in 0.02 seconds)
## Iteration 5000: error is 0.278943 (50 iterations in 0.02 seconds)
## Fitting performed in 1.55 seconds.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="fl">1.5</span>,<span class="fl">9.5</span>))
<span class="kw">plot</span>(tsne$Y,<span class="dt">pch=</span>speciesPch[selectSamples$chimpBonobo],<span class="dt">bg=</span>areaCols[selectSamples$area2],<span class="dt">col=</span>malariaCols[selectSamples$malaria<span class="dv">+1</span>],<span class="dt">cex=</span><span class="fl">2.2</span>,<span class="dt">lwd=</span><span class="fl">2.5</span>,<span class="dt">ylab=</span><span class="st">&#39;t-SNE 2&#39;</span>,<span class="dt">xlab=</span><span class="st">&#39;t-SNE 1&#39;</span>,<span class="dt">main=</span><span class="st">&#39;&#39;</span>,<span class="dt">bty=</span><span class="st">&#39;l&#39;</span>)
<span class="kw">legend</span>(
  <span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">2</span>]+.<span class="dv">01</span>*<span class="kw">diff</span>(<span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">1</span>:<span class="dv">2</span>]), 
  <span class="kw">mean</span>(<span class="kw">par</span>(<span class="st">&#39;usr&#39;</span>)[<span class="dv">3</span>:<span class="dv">4</span>]),
  <span class="kw">c</span>(<span class="kw">names</span>(malariaCols),<span class="kw">names</span>(areaCols),<span class="kw">names</span>(speciesPch)),
  <span class="dt">col=</span><span class="kw">c</span>(malariaCols,<span class="kw">rep</span>(<span class="kw">c</span>(malariaCols[<span class="dv">1</span>],mediumMalariaCol),<span class="kw">c</span>(<span class="kw">length</span>(areaCols),<span class="kw">length</span>(speciesPch)))),
  <span class="dt">pch=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">21</span>,<span class="kw">length</span>(malariaCols)),speciesPch[areaPch],speciesPch),
  <span class="dt">pt.bg=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="ot">NA</span>,<span class="kw">length</span>(malariaCols)),areaCols,<span class="kw">rep</span>(<span class="ot">NA</span>,<span class="kw">length</span>(speciesPch))),
  <span class="dt">inset=</span>.<span class="dv">01</span>,<span class="dt">pt.lwd=</span><span class="dv">3</span>,<span class="dt">pt.cex=</span><span class="fl">2.5</span>,
  <span class="dt">xjust=</span><span class="dv">0</span>,<span class="dt">xpd=</span><span class="ot">NA</span>,<span class="dt">bty=</span><span class="st">&#39;n&#39;</span>
)</code></pre></div>
<p><img src="figure/bacteria_tSNE-1.png" alt="plot of chunk bacteria_tSNE" /> ## Load libraries</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dnar)
<span class="kw">packageVersion</span>(<span class="st">&#39;dnar&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;0.1&#39;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(vegan)</code></pre></div>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.4-3</code></pre>
<pre><code>## 
## Attaching package: &#39;vegan&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dnar&#39;:
## 
##     rarefy</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">packageVersion</span>(<span class="st">&#39;vegan&#39;</span>)</code></pre></div>
<pre><code>## [1] &#39;2.4.3&#39;</code></pre>
<h2 id="load-data-2">Load data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;readData.R&#39;</span>)</code></pre></div>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dnar&#39;:
## 
##     read.fastq</code></pre>
<pre><code>## ape package version 4.1</code></pre>
<pre><code>## phyloseq package version 1.20.0</code></pre>
<pre><code>## Requiring samples to have 15000 reads</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&#39;../functions.R&#39;</span>)
<span class="co">#from plotPcoa.Rmd</span>
<span class="kw">load</span>(<span class="st">&#39;work/uniDist.Rdat&#39;</span>)</code></pre></div>
<h2 id="convenience-function-for-getting-distances-from-matrix">Convenience function for getting distances from matrix</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pullDists&lt;-function(xx,distMat){
  isIdentical&lt;-<span class="kw">length</span>(xx[[<span class="dv">1</span>]])==<span class="kw">length</span>(xx[[<span class="dv">2</span>]]) &amp;&amp;<span class="st"> </span><span class="kw">all</span>(xx[[<span class="dv">1</span>]]==xx[[<span class="dv">2</span>]])
  select&lt;-distMat[xx[[<span class="dv">1</span>]],xx[[<span class="dv">2</span>]]]
  if(isIdentical)dists&lt;-select[<span class="kw">upper.tri</span>(select)]
  else dists&lt;-<span class="kw">as.vector</span>(select)
  <span class="kw">return</span>(dists)
}</code></pre></div>
<h2 id="pull-out-distances-for-comparisons">Pull out distances for comparisons</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">comparisons&lt;-<span class="kw">withAs</span>(<span class="dt">s=</span>samples[isEnough[samples$Code],],<span class="kw">list</span>(
  <span class="kw">list</span>(
    <span class="st">&#39;Within bonobo samples&#39;</span>=<span class="kw">list</span>(s[s$bonobo,<span class="st">&#39;Code&#39;</span>],s[s$bonobo,<span class="st">&#39;Code&#39;</span>]),
    <span class="st">&#39;Within chimpanzee samples&#39;</span>=<span class="kw">list</span>(s[!s$bonobo,<span class="st">&#39;Code&#39;</span>],s[!s$bonobo,<span class="st">&#39;Code&#39;</span>]),
    <span class="st">&#39;Between bonobo and</span><span class="ch">\n</span><span class="st">chimpanzee samples&#39;</span>=<span class="kw">list</span>(s[s$bonobo,<span class="st">&#39;Code&#39;</span>],s[!s$bonobo,<span class="st">&#39;Code&#39;</span>])
  ),<span class="kw">list</span>(
    <span class="st">&#39;Within non-endemic bonobo</span><span class="ch">\n</span><span class="st">field site samples&#39;</span>=<span class="kw">list</span>(<span class="dv">0</span>,<span class="dv">0</span>),
    <span class="st">&#39;Between non-endemic bonobo</span><span class="ch">\n</span><span class="st">field site samples&#39;</span>=<span class="kw">list</span>(<span class="dv">0</span>,<span class="dv">0</span>),
    <span class="st">&#39;Between TL2 and non-endemic</span><span class="ch">\n</span><span class="st">bonobo field site samples&#39;</span>=<span class="kw">list</span>(s[s$isTL&amp;s$bonobo,<span class="st">&#39;Code&#39;</span>],s[!s$isTL&amp;s$bonobo,<span class="st">&#39;Code&#39;</span>])
  ),<span class="kw">list</span>(
    <span class="st">&#39;Within TL2 Laverania negative samples&#39;</span>=<span class="kw">list</span>(s[s$isTL&amp;!s$malaria,<span class="st">&#39;Code&#39;</span>],s[s$isTL&amp;!s$malaria,<span class="st">&#39;Code&#39;</span>]),
    <span class="st">&#39;Within TL2 Laverania positive samples&#39;</span>=<span class="kw">list</span>(s[s$isTL&amp;s$malaria,<span class="st">&#39;Code&#39;</span>],s[s$isTL&amp;s$malaria,<span class="st">&#39;Code&#39;</span>]),
    <span class="st">&#39;Between TL2 Laverania</span><span class="ch">\n</span><span class="st">positive and negative samples&#39;</span>=<span class="kw">list</span>(s[s$isTL&amp;s$malaria,<span class="st">&#39;Code&#39;</span>],s[s$isTL&amp;!s$malaria,<span class="st">&#39;Code&#39;</span>])
  ),<span class="kw">list</span>(
    <span class="st">&#39;Within Laverania negative</span><span class="ch">\n</span><span class="st">chimpanzee samples&#39;</span>=<span class="kw">list</span>(s[!s$bonobo&amp;!s$malaria,<span class="st">&#39;Code&#39;</span>],s[!s$bonobo&amp;!s$malaria,<span class="st">&#39;Code&#39;</span>]),
    <span class="st">&#39;Within Laverania positive</span><span class="ch">\n</span><span class="st">chimpanzee samples&#39;</span>=<span class="kw">list</span>(s[!s$bonobo&amp;s$malaria,<span class="st">&#39;Code&#39;</span>],s[!s$bonobo&amp;s$malaria,<span class="st">&#39;Code&#39;</span>]),
    <span class="st">&#39;Between Laverania negative</span><span class="ch">\n</span><span class="st"> and positive chimpanzee samples&#39;</span>=<span class="kw">list</span>(s[!s$bonobo&amp;s$malaria,<span class="st">&#39;Code&#39;</span>],s[!s$bonobo&amp;!s$malaria,<span class="st">&#39;Code&#39;</span>])
  )
))
nonTL&lt;-<span class="kw">unique</span>(samples[samples$bonobo&amp;!samples$isTL,<span class="st">&#39;area&#39;</span>])
<span class="kw">names</span>(nonTL)&lt;-nonTL
allCombo&lt;-<span class="kw">unique</span>(<span class="kw">t</span>(<span class="kw">apply</span>(<span class="kw">expand.grid</span>(nonTL,nonTL),<span class="dv">1</span>,sort)))
allCombo&lt;-allCombo[allCombo[,<span class="dv">1</span>]!=allCombo[,<span class="dv">2</span>],]
betweenSites&lt;-<span class="kw">apply</span>(allCombo,<span class="dv">1</span>,function(xx,distMat)<span class="kw">pullDists</span>(<span class="kw">list</span>(samples[isEnough[samples$Code]&amp;samples$area==xx[<span class="dv">1</span>],<span class="st">&#39;Code&#39;</span>],samples[isEnough[samples$Code]&amp;samples$area==xx[<span class="dv">2</span>],<span class="st">&#39;Code&#39;</span>]),distMat),<span class="kw">as.matrix</span>(uniDist))
withinSites&lt;-<span class="kw">lapply</span>(nonTL,function(xx,distMat)<span class="kw">pullDists</span>(<span class="kw">list</span>(samples[isEnough[samples$Code]&amp;samples$area==xx,<span class="st">&#39;Code&#39;</span>],samples[isEnough[samples$Code]&amp;samples$area==xx,<span class="st">&#39;Code&#39;</span>]),distMat),<span class="kw">as.matrix</span>(uniDist))
distList&lt;-<span class="kw">lapply</span>(comparisons,function(xx)<span class="kw">lapply</span>(xx,pullDists,<span class="kw">as.matrix</span>(uniDist)))
distList&lt;-<span class="kw">lapply</span>(distList,function(xx){
  <span class="kw">names</span>(xx)&lt;-<span class="kw">ifelse</span>(<span class="kw">nchar</span>(<span class="kw">names</span>(xx))&gt;<span class="dv">20</span>,<span class="kw">sub</span>(<span class="st">&#39; vs &#39;</span>,<span class="st">&#39; vs</span><span class="ch">\n</span><span class="st">&#39;</span>,<span class="kw">names</span>(xx)),<span class="kw">names</span>(xx))
  if(<span class="kw">any</span>(<span class="kw">names</span>(xx)==<span class="st">&#39;Between non-endemic bonobo</span><span class="ch">\n</span><span class="st">field site samples&#39;</span>))xx[[<span class="st">&#39;Between non-endemic bonobo</span><span class="ch">\n</span><span class="st">field site samples&#39;</span>]]&lt;-<span class="kw">unlist</span>(betweenSites)
  if(<span class="kw">any</span>(<span class="kw">names</span>(xx)==<span class="st">&#39;Within non-endemic bonobo</span><span class="ch">\n</span><span class="st">field site samples&#39;</span>))xx[[<span class="st">&#39;Within non-endemic bonobo</span><span class="ch">\n</span><span class="st">field site samples&#39;</span>]]&lt;-<span class="kw">unlist</span>(withinSites)
  <span class="kw">return</span>(xx)
})</code></pre></div>
<h2 id="calculate-wilcoxon-test-p-values-between-groups-of-distances">Calculate Wilcoxon test p-values between groups of distances</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pVals&lt;-<span class="kw">lapply</span>(distList,function(dists)<span class="kw">outer</span>(dists,dists,function(xx,yy)<span class="kw">mapply</span>(function(xxx,yyy){<span class="kw">wilcox.test</span>(xxx,yyy)$p.value},xx,yy)))
pVals2&lt;-<span class="kw">lapply</span>(distList,function(dists)<span class="kw">outer</span>(dists,dists,function(xx,yy)<span class="kw">mapply</span>(function(xxx,yyy){<span class="kw">wilcox.test</span>(xxx,yyy,<span class="dt">alternative=</span><span class="st">&#39;less&#39;</span>)$p.value},xx,yy)))
pVals&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">mapply</span>(function(xx,yy){
  n&lt;-<span class="kw">nrow</span>(xx)
  <span class="co">#use one sided for comparisons to between</span>
  xx[,<span class="kw">ncol</span>(xx)]&lt;-yy[,<span class="kw">ncol</span>(xx)]
  cols&lt;-<span class="kw">colnames</span>(xx)[<span class="kw">matrix</span>(<span class="dv">1</span>:n,<span class="dt">nrow=</span>n,<span class="dt">ncol=</span>n,<span class="dt">byrow=</span><span class="ot">TRUE</span>)[<span class="kw">upper.tri</span>(xx)]]
  rows&lt;-<span class="kw">rownames</span>(xx)[<span class="kw">matrix</span>(<span class="dv">1</span>:n,<span class="dt">nrow=</span>n,<span class="dt">ncol=</span>n,<span class="dt">byrow=</span><span class="ot">FALSE</span>)[<span class="kw">upper.tri</span>(xx)]]
  ps&lt;-xx[<span class="kw">upper.tri</span>(xx)]
  <span class="kw">data.frame</span>(<span class="st">&#39;x&#39;</span>=rows,<span class="st">&#39;y&#39;</span>=cols,<span class="st">&#39;p&#39;</span>=ps,<span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
},pVals,pVals2,<span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>))
pVals$sig&lt;-<span class="st"> </span><span class="kw">symnum</span>(pVals$p, <span class="dt">corr =</span> <span class="ot">FALSE</span>, <span class="dt">na =</span> <span class="ot">FALSE</span>, <span class="dt">cutpoints =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">1e-6</span>, <span class="fl">1e-4</span>, <span class="fl">1e-2</span>,<span class="dv">1</span>), <span class="dt">symbols =</span> <span class="kw">c</span>(<span class="st">&quot;***&quot;</span>, <span class="st">&quot;**&quot;</span>, <span class="st">&quot;*&quot;</span>,<span class="st">&#39;&#39;</span>))
pVals&lt;-pVals[pVals$p&lt;.<span class="dv">01</span>,]</code></pre></div>
<h2 id="setup-plotting-parameters-1">Setup plotting parameters</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">spacer&lt;-.<span class="dv">6</span>
cols&lt;-<span class="kw">rainbow.lab</span>(<span class="kw">length</span>(distList),<span class="dt">start=</span><span class="dv">1</span>,<span class="dt">end=</span>-<span class="dv">2</span>)
groupId&lt;-<span class="kw">rep</span>(<span class="dv">1</span>:<span class="kw">length</span>(distList),<span class="kw">sapply</span>(distList,length))</code></pre></div>
<h2 id="plot-comparison-of-distances-between-groups">Plot comparison of distances between groups</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">11</span>,<span class="fl">2.75</span>,.<span class="dv">3</span>,<span class="dv">4</span>),<span class="dt">lheight=</span>.<span class="dv">8</span>)
compareFactor&lt;-<span class="kw">factor</span>(<span class="kw">rep</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(distList,names)),<span class="kw">unlist</span>(<span class="kw">lapply</span>(distList,sapply,length))),<span class="dt">levels=</span><span class="kw">unlist</span>(<span class="kw">lapply</span>(distList,function(xx)<span class="kw">rev</span>(<span class="kw">names</span>(xx)))))
stats&lt;-<span class="kw">boxplot</span>(<span class="kw">unlist</span>(distList)~compareFactor,<span class="dt">range=</span><span class="ot">Inf</span>,<span class="dt">notch=</span><span class="ot">TRUE</span>,<span class="dt">plot=</span><span class="ot">FALSE</span>)
betaCI&lt;-<span class="kw">tapply</span>(<span class="kw">unlist</span>(distList),compareFactor,function(xx)<span class="kw">medianCI</span>(xx))
pos&lt;-<span class="kw">sum</span>(<span class="kw">sapply</span>(distList,length)):<span class="dv">1</span>-<span class="kw">rep</span>((<span class="dv">1</span>:<span class="kw">length</span>(distList)-<span class="dv">1</span>)*spacer,<span class="kw">sapply</span>(distList,length))
<span class="kw">names</span>(pos)&lt;-<span class="kw">levels</span>(compareFactor)
pVals$top&lt;-pos[pVals$x]
pVals$bottom&lt;-pos[pVals$y]
pVals$row&lt;-<span class="kw">stackRegions</span>(pVals$top,pVals$bottom)
pVals$middle&lt;-<span class="kw">apply</span>(pVals[,<span class="kw">c</span>(<span class="st">&#39;bottom&#39;</span>,<span class="st">&#39;top&#39;</span>)],<span class="dv">1</span>,mean)
pVals$xPos&lt;-.<span class="dv">95</span><span class="fl">+.03</span>*(pVals$row<span class="dv">-1</span>)
<span class="kw">plot</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">type=</span><span class="st">&#39;n&#39;</span>,<span class="dt">xlim=</span><span class="kw">range</span>(pos)+<span class="kw">c</span>(-.<span class="dv">5</span>-spacer,.<span class="dv">5</span>+spacer),<span class="dt">ylim=</span><span class="kw">c</span>(<span class="kw">min</span>(<span class="kw">unlist</span>(distList)),<span class="dv">1</span>),<span class="dt">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="dt">xlab=</span><span class="st">&#39;&#39;</span>,<span class="dt">ylab=</span><span class="st">&#39;UniFrac distance&#39;</span>,<span class="dt">mgp=</span><span class="kw">c</span>(<span class="fl">1.75</span>,.<span class="dv">4</span>,<span class="dv">0</span>),<span class="dt">tcl=</span>-.<span class="dv">3</span>,<span class="dt">xaxs=</span><span class="st">&#39;i&#39;</span>,<span class="dt">las=</span><span class="dv">1</span>,<span class="dt">bty=</span><span class="st">&#39;l&#39;</span>)
for(ii in <span class="kw">ncol</span>(stats$stats):<span class="dv">1</span>){
  rawDists&lt;-distList[[groupId[ii]]][[stats$names[ii]]]
  thisCI&lt;-betaCI[[stats$names[ii]]]
  xCoords&lt;-<span class="kw">c</span>(stats$stats[<span class="dv">2</span>,ii],thisCI[<span class="dv">1</span>],stats$stats[<span class="dv">3</span>,ii],thisCI[<span class="dv">2</span>],stats$stats[<span class="dv">4</span>,ii])
  yCoords&lt;-<span class="kw">c</span>(.<span class="dv">4</span>,.<span class="dv">4</span>,.<span class="dv">1</span>,.<span class="dv">4</span>,.<span class="dv">4</span>)
  <span class="kw">segments</span>(pos[ii],stats$stats[<span class="dv">1</span>,ii],pos[ii],stats$stats[<span class="dv">5</span>,ii],<span class="dt">lwd=</span><span class="dv">3</span>,<span class="dt">col=</span>cols[groupId[ii]])
  <span class="kw">polygon</span>(<span class="kw">c</span>(yCoords,-<span class="kw">rev</span>(yCoords))+pos[ii],<span class="kw">c</span>(xCoords,<span class="kw">rev</span>(xCoords)),<span class="dt">col=</span>cols[groupId[ii]])
  <span class="kw">segments</span>(pos[ii]+yCoords[<span class="dv">3</span>],xCoords[<span class="dv">3</span>],pos[ii]-yCoords[<span class="dv">3</span>],xCoords[<span class="dv">3</span>])
}
<span class="kw">text</span>(pVals$middle,pVals$xPos<span class="fl">+.005</span>,pVals$sig,<span class="dt">adj=</span><span class="kw">c</span>(.<span class="dv">5</span>,<span class="fl">0.5</span>),<span class="dt">xpd=</span><span class="ot">NA</span>)
<span class="kw">segments</span>(pVals$bottom,pVals$xPos,pVals$top,pVals$xPos)
breaks&lt;-<span class="kw">which</span>(<span class="kw">c</span>(<span class="ot">FALSE</span>,pos[-<span class="dv">1</span>]-pos[-<span class="kw">length</span>(pos)]&lt;<span class="st"> </span>-<span class="dv">1</span>))
<span class="kw">slantAxis</span>(<span class="dv">1</span>,pos,<span class="kw">names</span>(pos),<span class="dt">srt=</span>-<span class="dv">40</span>)</code></pre></div>
<div class="figure">
<img src="figure/distanceComparison-1.png" alt="plot of chunk distanceComparison" />
<p class="caption">plot of chunk distanceComparison</p>
</div>
<h2 id="calculate-permanova-statistics-of-distances-and-covariates">Calculate PERMANOVA statistics of distances and covariates</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">adonis</span>(uniDist~bonobo+area2+malaria,<span class="dt">data=</span>samples[<span class="kw">labels</span>(uniDist),],<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">20</span>)</code></pre></div>
<pre><code>## 
## Call:
## adonis(formula = uniDist ~ bonobo + area2 + malaria, data = samples[labels(uniDist),      ], permutations = 1e+07, parallel = 20) 
## 
## Permutation: free
## Number of permutations: 1e+07
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2  Pr(&gt;F)    
## bonobo     1    1.5005 1.50050  8.6010 0.07490   1e-07 ***
## area2      8    3.8150 0.47687  2.7334 0.19043   1e-07 ***
## malaria    1    0.2382 0.23817  1.3652 0.01189 0.04587 *  
## Residuals 83   14.4799 0.17446         0.72278            
## Total     93   20.0335                 1.00000            
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tlDist&lt;-<span class="kw">as.dist</span>(<span class="kw">as.matrix</span>(uniDist)[samples[<span class="kw">labels</span>(uniDist),<span class="st">&#39;isTL&#39;</span>],samples[<span class="kw">labels</span>(uniDist),<span class="st">&#39;isTL&#39;</span>]])
<span class="kw">adonis</span>(tlDist~area2+malaria,<span class="dt">data=</span>samples[<span class="kw">labels</span>(tlDist),],<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">20</span>)</code></pre></div>
<pre><code>## 
## Call:
## adonis(formula = tlDist ~ area2 + malaria, data = samples[labels(tlDist),      ], permutations = 1e+07, parallel = 20) 
## 
## Permutation: free
## Number of permutations: 1e+07
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2   Pr(&gt;F)    
## area2      2    0.8622 0.43108  2.6072 0.14149    1e-07 ***
## malaria    1    0.2711 0.27115  1.6399 0.04450 0.002445 ** 
## Residuals 30    4.9603 0.16534         0.81402             
## Total     33    6.0936                 1.00000             
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">chimpDist&lt;-<span class="kw">as.dist</span>(<span class="kw">as.matrix</span>(uniDist)[!samples[<span class="kw">labels</span>(uniDist),<span class="st">&#39;bonobo&#39;</span>],!samples[<span class="kw">labels</span>(uniDist),<span class="st">&#39;bonobo&#39;</span>]])
<span class="kw">adonis</span>(chimpDist~area2+malaria,<span class="dt">data=</span>samples[<span class="kw">labels</span>(chimpDist),],<span class="dt">permutations=</span><span class="fl">1e7</span>,<span class="dt">parallel=</span><span class="dv">20</span>)</code></pre></div>
<pre><code>## 
## Call:
## adonis(formula = chimpDist ~ area2 + malaria, data = samples[labels(chimpDist),      ], permutations = 1e+07, parallel = 20) 
## 
## Permutation: free
## Number of permutations: 1e+07
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2   Pr(&gt;F)    
## area2      2    0.9274 0.46372  2.4584 0.17562 2.25e-05 ***
## malaria    1    0.2037 0.20367  1.0797 0.03857   0.2998    
## Residuals 22    4.1498 0.18863         0.78581             
## Total     25    5.2810                 1.00000             
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</body>
</html>
