## Load libraries
```{r}
# installed from https://github.com/sherrillmix/dnar 
library(dnar)
packageVersion('dnar')
library(parallel)
packageVersion('parallel')
library(ape)
packageVersion('ape')
library(phyloseq)
packageVersion('phyloseq')
source('../functions.R')
```

## Software versions
```{r,cache=TRUE}
system('echo "source activate qiime1;print_qiime_config.py -t"|bash 2>&1',intern=TRUE)
system('bbmerge.sh --version 2>&1',intern=TRUE)
```

## Require 15000 reads in a sample
```{r}
nRequiredReads<-15000
```

## Merge reads
```{r}
  fastqs<-list.files('data','.fastq.gz',full.name=TRUE)
  samples<-sub('_.*$','',basename(fastqs))
  for(ii in samples){
    cmd<-sprintf('~/installs/bbmap/bbmerge.sh in1=%s in2=%s out=%s t=10 2>%s',links[1],links[2],out,sub('fastq$','out',out))
    message(cmd)
    exit<-system(cmd)
    if(exit!=0)stop("Problem running pairing in",ii)
    system(sprintf('gzip %s',out))
  }
```

## Read in all sequences
```{r readFastqs,cache=TRUE}
fastqs<-list.files('data/joined','.fastq.gz',full.name=TRUE)
#just get sequence to reduce memory
allSeq<-lapply(fastqs,function(xx)read.fastq(xx)$seq)
```

## Run qiime
```{r runQiime,cache=TRUE}
if(!dir.exists('work/qiime')){
  out<-runQiime(unlist(allSeq),storeDir='work/qiime')
  withAs(outFile=gzfile('work/qiimeOtuIds.csv.gz'),write.csv(data.frame('file'=rep(sub('_Feces_1_1.fastq.gz','',basename(fastqs)),sapply(allSeq,length)),'otu'=out[['otus']],stringsAsFactors=FALSE),outFile,row.names=FALSE))
  write.fa(names(out[['seqs']]),out[['seqs']],'work/qiimeOtus.fa.gz')
  write.csv(data.frame('name'=names(out[['taxa']]),'taxa'=out[['taxa']],stringsAsFactors=FALSE),'work/qiimeOtus.taxa',row.names=FALSE)
}
```

## Read qiime otus
```{r readOtus,cache=TRUE}
otus<-read.csv('work/qiimeOtuIds.csv.gz',stringsAsFactors=FALSE)
otuTab<-as.data.frame.matrix(table(otus$otu,otus$file))
```

### Discard singletons
```{r}
otuTab<-otuTab[apply(otuTab,1,sum)>1,]
```

## Read qiime taxonomy
```{r readTaxa,cache=TRUE}
taxaRaw<-read.csv('work/qiimeOtus.taxa',stringsAsFactors=FALSE)
taxa<-parseQiimeTaxa(taxaRaw$taxa)
rownames(taxa)<-taxaRaw$name
taxa$best<-apply(taxa,1,function(x)ifelse(is.na(x),x,sprintf('%s_%s',names(x),x))[max(c(1,which(!is.na(x))))])
taxa$bestId<-ave(taxa$best,naReplace(taxa$best,'__NAFILLER__'),FUN=function(x){sprintf('%s #%d',ifelse(is.na(x),'Unknown',x),1:length(x))})
otuSeq<-read.fa('work/qiimeOtus.fa.gz')
rownames(otuSeq)<-otuSeq$name
taxa$seq<-otuSeq[rownames(taxa),'seq']
```

## Filter chloroplast reads
```{r}
isChloro<-taxa[rownames(otuTab),'c']=='Chloroplast'&!is.na(taxa[rownames(otuTab),'c'])
otuTab<-otuTab[!isChloro,]
```

## Filter samples without enough reads
```{r}
otuTab<-otuTab[,apply(otuTab,2,sum)>nRequiredReads]
```

## Convert to proportions
```{r}
otuProp<-apply(otuProp,2,function(x)x/sum(x))
```

## Read tree
```{r}
#make sure tree is bifurcating or breaks UniFrac without error
tree<-ape::multi2di(phyloseq::read_tree('work/qiime/rep_set.tre'))
```

## Save files
```{r}
save(otuTab,otuProp,taxa,tree,nRequiredReads,file='work/otus.Rdat')
```

## Read counts
```{r}
readCounts<-apply(otuTab,2,sum)
print(readCounts)
```
